{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 15 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene</th><th scope=col>ontology</th><th scope=col>num_cells</th><th scope=col>med_gene_spots</th><th scope=col>med_spots</th><th scope=col>med_score</th><th scope=col>z</th><th scope=col>bh_p</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Flt1</td><td>Endo      </td><td>3657</td><td>29.0</td><td>155.0</td><td>-0.03846154</td><td>-15.313786</td><td>2.257567e-52</td></tr>\n",
       "\t<tr><td>Flt1</td><td>SMC       </td><td>2245</td><td>14.0</td><td>329.0</td><td>-0.01265823</td><td>  2.097699</td><td>4.673136e-02</td></tr>\n",
       "\t<tr><td>Flt1</td><td>OPC       </td><td> 204</td><td>10.0</td><td>310.0</td><td> 0.13192302</td><td>  7.788495</td><td>1.904675e-14</td></tr>\n",
       "\t<tr><td>Flt1</td><td>Astro     </td><td> 559</td><td> 9.0</td><td>201.0</td><td> 0.16580311</td><td> 11.759899</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Flt1</td><td>L6 IT Car3</td><td>  61</td><td> 9.0</td><td>658.0</td><td> 0.34340223</td><td>  7.717276</td><td>3.298247e-14</td></tr>\n",
       "\t<tr><td>Flt1</td><td>Oligo     </td><td> 654</td><td> 9.0</td><td>266.5</td><td> 0.34366399</td><td> 24.619161</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Flt1</td><td>Vip       </td><td>  52</td><td> 8.0</td><td>529.0</td><td> 0.38169994</td><td>  7.327015</td><td>6.173839e-13</td></tr>\n",
       "\t<tr><td>Flt1</td><td>Sst       </td><td>  56</td><td> 7.5</td><td>774.5</td><td> 0.38746101</td><td>  9.600999</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Flt1</td><td>L5 IT     </td><td> 267</td><td> 7.0</td><td>731.0</td><td> 0.52532562</td><td> 20.429761</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Flt1</td><td>L5 ET     </td><td> 111</td><td> 7.0</td><td>956.0</td><td> 0.55937053</td><td> 13.782273</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Flt1</td><td>L2/3 IT   </td><td> 316</td><td> 6.0</td><td>678.0</td><td> 0.56336607</td><td> 24.159044</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Flt1</td><td>L6 IT     </td><td> 171</td><td> 6.0</td><td>671.0</td><td> 0.62674095</td><td> 19.267720</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Flt1</td><td>Pvalb     </td><td> 131</td><td> 6.0</td><td>789.0</td><td> 0.64690027</td><td> 17.159377</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Flt1</td><td>Lamp5     </td><td>  56</td><td> 7.0</td><td>662.0</td><td> 0.66324235</td><td> 13.237130</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Flt1</td><td>L6 CT     </td><td> 126</td><td> 6.0</td><td>479.0</td><td> 0.67299578</td><td> 16.154562</td><td>0.000000e+00</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 15 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " gene & ontology & num\\_cells & med\\_gene\\_spots & med\\_spots & med\\_score & z & bh\\_p\\\\\n",
       " <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t Flt1 & Endo       & 3657 & 29.0 & 155.0 & -0.03846154 & -15.313786 & 2.257567e-52\\\\\n",
       "\t Flt1 & SMC        & 2245 & 14.0 & 329.0 & -0.01265823 &   2.097699 & 4.673136e-02\\\\\n",
       "\t Flt1 & OPC        &  204 & 10.0 & 310.0 &  0.13192302 &   7.788495 & 1.904675e-14\\\\\n",
       "\t Flt1 & Astro      &  559 &  9.0 & 201.0 &  0.16580311 &  11.759899 & 0.000000e+00\\\\\n",
       "\t Flt1 & L6 IT Car3 &   61 &  9.0 & 658.0 &  0.34340223 &   7.717276 & 3.298247e-14\\\\\n",
       "\t Flt1 & Oligo      &  654 &  9.0 & 266.5 &  0.34366399 &  24.619161 & 0.000000e+00\\\\\n",
       "\t Flt1 & Vip        &   52 &  8.0 & 529.0 &  0.38169994 &   7.327015 & 6.173839e-13\\\\\n",
       "\t Flt1 & Sst        &   56 &  7.5 & 774.5 &  0.38746101 &   9.600999 & 0.000000e+00\\\\\n",
       "\t Flt1 & L5 IT      &  267 &  7.0 & 731.0 &  0.52532562 &  20.429761 & 0.000000e+00\\\\\n",
       "\t Flt1 & L5 ET      &  111 &  7.0 & 956.0 &  0.55937053 &  13.782273 & 0.000000e+00\\\\\n",
       "\t Flt1 & L2/3 IT    &  316 &  6.0 & 678.0 &  0.56336607 &  24.159044 & 0.000000e+00\\\\\n",
       "\t Flt1 & L6 IT      &  171 &  6.0 & 671.0 &  0.62674095 &  19.267720 & 0.000000e+00\\\\\n",
       "\t Flt1 & Pvalb      &  131 &  6.0 & 789.0 &  0.64690027 &  17.159377 & 0.000000e+00\\\\\n",
       "\t Flt1 & Lamp5      &   56 &  7.0 & 662.0 &  0.66324235 &  13.237130 & 0.000000e+00\\\\\n",
       "\t Flt1 & L6 CT      &  126 &  6.0 & 479.0 &  0.67299578 &  16.154562 & 0.000000e+00\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 15 × 8\n",
       "\n",
       "| gene &lt;chr&gt; | ontology &lt;chr&gt; | num_cells &lt;int&gt; | med_gene_spots &lt;dbl&gt; | med_spots &lt;dbl&gt; | med_score &lt;dbl&gt; | z &lt;dbl&gt; | bh_p &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| Flt1 | Endo       | 3657 | 29.0 | 155.0 | -0.03846154 | -15.313786 | 2.257567e-52 |\n",
       "| Flt1 | SMC        | 2245 | 14.0 | 329.0 | -0.01265823 |   2.097699 | 4.673136e-02 |\n",
       "| Flt1 | OPC        |  204 | 10.0 | 310.0 |  0.13192302 |   7.788495 | 1.904675e-14 |\n",
       "| Flt1 | Astro      |  559 |  9.0 | 201.0 |  0.16580311 |  11.759899 | 0.000000e+00 |\n",
       "| Flt1 | L6 IT Car3 |   61 |  9.0 | 658.0 |  0.34340223 |   7.717276 | 3.298247e-14 |\n",
       "| Flt1 | Oligo      |  654 |  9.0 | 266.5 |  0.34366399 |  24.619161 | 0.000000e+00 |\n",
       "| Flt1 | Vip        |   52 |  8.0 | 529.0 |  0.38169994 |   7.327015 | 6.173839e-13 |\n",
       "| Flt1 | Sst        |   56 |  7.5 | 774.5 |  0.38746101 |   9.600999 | 0.000000e+00 |\n",
       "| Flt1 | L5 IT      |  267 |  7.0 | 731.0 |  0.52532562 |  20.429761 | 0.000000e+00 |\n",
       "| Flt1 | L5 ET      |  111 |  7.0 | 956.0 |  0.55937053 |  13.782273 | 0.000000e+00 |\n",
       "| Flt1 | L2/3 IT    |  316 |  6.0 | 678.0 |  0.56336607 |  24.159044 | 0.000000e+00 |\n",
       "| Flt1 | L6 IT      |  171 |  6.0 | 671.0 |  0.62674095 |  19.267720 | 0.000000e+00 |\n",
       "| Flt1 | Pvalb      |  131 |  6.0 | 789.0 |  0.64690027 |  17.159377 | 0.000000e+00 |\n",
       "| Flt1 | Lamp5      |   56 |  7.0 | 662.0 |  0.66324235 |  13.237130 | 0.000000e+00 |\n",
       "| Flt1 | L6 CT      |  126 |  6.0 | 479.0 |  0.67299578 |  16.154562 | 0.000000e+00 |\n",
       "\n"
      ],
      "text/plain": [
       "   gene ontology   num_cells med_gene_spots med_spots med_score   z         \n",
       "1  Flt1 Endo       3657      29.0           155.0     -0.03846154 -15.313786\n",
       "2  Flt1 SMC        2245      14.0           329.0     -0.01265823   2.097699\n",
       "3  Flt1 OPC         204      10.0           310.0      0.13192302   7.788495\n",
       "4  Flt1 Astro       559       9.0           201.0      0.16580311  11.759899\n",
       "5  Flt1 L6 IT Car3   61       9.0           658.0      0.34340223   7.717276\n",
       "6  Flt1 Oligo       654       9.0           266.5      0.34366399  24.619161\n",
       "7  Flt1 Vip          52       8.0           529.0      0.38169994   7.327015\n",
       "8  Flt1 Sst          56       7.5           774.5      0.38746101   9.600999\n",
       "9  Flt1 L5 IT       267       7.0           731.0      0.52532562  20.429761\n",
       "10 Flt1 L5 ET       111       7.0           956.0      0.55937053  13.782273\n",
       "11 Flt1 L2/3 IT     316       6.0           678.0      0.56336607  24.159044\n",
       "12 Flt1 L6 IT       171       6.0           671.0      0.62674095  19.267720\n",
       "13 Flt1 Pvalb       131       6.0           789.0      0.64690027  17.159377\n",
       "14 Flt1 Lamp5        56       7.0           662.0      0.66324235  13.237130\n",
       "15 Flt1 L6 CT       126       6.0           479.0      0.67299578  16.154562\n",
       "   bh_p        \n",
       "1  2.257567e-52\n",
       "2  4.673136e-02\n",
       "3  1.904675e-14\n",
       "4  0.000000e+00\n",
       "5  3.298247e-14\n",
       "6  0.000000e+00\n",
       "7  6.173839e-13\n",
       "8  0.000000e+00\n",
       "9  0.000000e+00\n",
       "10 0.000000e+00\n",
       "11 0.000000e+00\n",
       "12 0.000000e+00\n",
       "13 0.000000e+00\n",
       "14 0.000000e+00\n",
       "15 0.000000e+00"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "srrs_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "#support for plus/minus strand bam plotting from:\n",
    "#https://support.bioconductor.org/p/56047/\n",
    "library(Rsamtools)\n",
    "strandedBamImport <- function (file, selection) {\n",
    "    if (!file.exists(paste(file, \"bai\", sep = \".\")))\n",
    "        stop(\"Unable to find index for BAM file '\", file, \"'. You can\n",
    "build an index using the following command:\\n\\t\",\n",
    "             \"library(Rsamtools)\\n\\tindexBam(\\\"\", file, \"\\\")\")\n",
    "    sinfo <- scanBamHeader(file)[[1]]\n",
    "    res <- if (!as.character(seqnames(selection)[1]) %in%\n",
    "               names(sinfo$targets)) {\n",
    "        mcols(selection) <- DataFrame(score = 0)\n",
    "        selection\n",
    "    }else {\n",
    "        param <- ScanBamParam(what = c(\"pos\", \"qwidth\", \"strand\"),\n",
    "                              which = selection, flag =\n",
    "scanBamFlag(isUnmappedQuery = FALSE))\n",
    "        x <- scanBam(file, param = param)[[1]]\n",
    "        gr <- GRanges(strand=x[[\"strand\"]], ranges=IRanges(x[[\"pos\"]],\n",
    "width = x[[\"qwidth\"]]), seqnames=seqnames(selection)[1])\n",
    "        grs <- split(gr, strand(gr))\n",
    "        cov <- lapply(grs[c(\"+\", \"-\")], function(y)\n",
    "coverage(ranges(y),\n",
    "width=end(selection)))\n",
    "        pos <- sort(unique(unlist(lapply(cov, function(y) c(start(y),\n",
    "end(y))))))\n",
    "        if(length(pos)==0){\n",
    "            mcols(selection) <- DataFrame(plus=0, minus=0)\n",
    "            selection\n",
    "        }else{\n",
    "            GRanges(seqnames = seqnames(selection)[1],\n",
    "ranges=IRanges(start=head(pos, -1), end=tail(pos, -1)),\n",
    "                    plus=as.numeric(cov[[\"+\"]][head(pos, -1)]),\n",
    "                    minus=-as.numeric(cov[[\"-\"]][head(pos, -1)]))\n",
    "        }\n",
    "    }\n",
    "    return(res)\n",
    "}\n",
    "                                         \n",
    "\n",
    "#args = commandArgs(trailingOnly=TRUE)\n",
    "\n",
    "#print(args)\n",
    "\n",
    "gene_name = \"Flt1\"\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "srrs_path = \"/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/final_outputs/20220103_merfish.csv\"\n",
    "bam_stem = \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/\"\n",
    "                                         \n",
    "#find which cell-types to plot from SRRS\n",
    "srrs_df = fread(srrs_path)\n",
    "srrs_df = srrs_df[gene == gene_name] #filter to results from the gene of interest\n",
    "srrs_df = srrs_df[bh_p < 0.05] #filter by bh_p\n",
    "srrs_df = srrs_df[order(med_score)] #order by effect size\n",
    "\n",
    "if(nrow(srrs_df) == 0){\n",
    "    print(\"No matching cell-types\")\n",
    "    stop()\n",
    "} else if (nrow(srrs_df) <= 4){\n",
    "    #plot all cell types\n",
    "    num_ontologies = nrow(srrs_df)\n",
    "    cell_types = srrs_df$ontology\n",
    "    effect_sizes = srrs_df$med_score\n",
    "} else {\n",
    "    #plot top 2 and bottom 2 cell-types by med score (effect size)\n",
    "    num_ontologies = 4\n",
    "    cell_types = c(head(srrs_df,2)$ontology, tail(srrs_df,2)$ontology)\n",
    "    effect_sizes = c(head(srrs_df,2)$med_score, tail(srrs_df,2)$med_score)\n",
    "}\n",
    "\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = gene_row$start\n",
    "window_end   = gene_row$end\n",
    "outfile = paste(gene_name,\".png\",sep=\"\")\n",
    "tracks = list()\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    feature(geneTrack) %in% c(\"CDS\", \"UTR\")\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "gTrack <- GenomeAxisTrack()                                         \n",
    "tracks = append(tracks,gTrack)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "for(i in 1:num_ontologies){\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    \n",
    "    effect_size = effect_sizes[[i]]\n",
    "    effect_size = format(round(effect_size, 2), nsmall = 2)\n",
    "    \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        name = paste(ct,effect_size,sep=\"\\n\"),\n",
    "        chromosome = chr,\n",
    "        importFunction=strandedBamImport,\n",
    "        stream=TRUE,\n",
    "    )\n",
    "    tracks = append(tracks,cov_track)\n",
    "}\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    type = \"h\",\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = \"UTR\",\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    col=c(\"red\",\"blue\"),\n",
    "    groups=c(\"+\", \"-\"),\n",
    "    fontcolor.exon = \"black\",\n",
    "    legend = F,\n",
    "    cex = 0.6\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/L6_CT.bam'"
      ],
      "text/latex": [
       "'/scratch/groups/horence/rob/data/MERFISH\\_scRNAseq/10X\\_mapping/merged\\_by\\_celltype/L6\\_CT.bam'"
      ],
      "text/markdown": [
       "'/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/L6_CT.bam'"
      ],
      "text/plain": [
       "[1] \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/L6_CT.bam\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bam_stem = \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/\"\n",
    "paste(bam_stem,ct,\".bam\",sep=\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<pre class=language-r><code>function (p, method = p.adjust.methods, n = length(p)) \n",
       "{\n",
       "<span style=white-space:pre-wrap>    method &lt;- match.arg(method)</span>\n",
       "<span style=white-space:pre-wrap>    if (method == \"fdr\") </span>\n",
       "<span style=white-space:pre-wrap>        method &lt;- \"BH\"</span>\n",
       "<span style=white-space:pre-wrap>    nm &lt;- names(p)</span>\n",
       "<span style=white-space:pre-wrap>    p &lt;- as.numeric(p)</span>\n",
       "<span style=white-space:pre-wrap>    p0 &lt;- setNames(p, nm)</span>\n",
       "<span style=white-space:pre-wrap>    if (all(nna &lt;- !is.na(p))) </span>\n",
       "<span style=white-space:pre-wrap>        nna &lt;- TRUE</span>\n",
       "<span style=white-space:pre-wrap>    p &lt;- p[nna]</span>\n",
       "<span style=white-space:pre-wrap>    lp &lt;- length(p)</span>\n",
       "<span style=white-space:pre-wrap>    stopifnot(n &gt;= lp)</span>\n",
       "<span style=white-space:pre-wrap>    if (n &lt;= 1) </span>\n",
       "<span style=white-space:pre-wrap>        return(p0)</span>\n",
       "<span style=white-space:pre-wrap>    if (n == 2 &amp;&amp; method == \"hommel\") </span>\n",
       "<span style=white-space:pre-wrap>        method &lt;- \"hochberg\"</span>\n",
       "<span style=white-space:pre-wrap>    p0[nna] &lt;- switch(method, bonferroni = pmin(1, n * p), holm = {</span>\n",
       "<span style=white-space:pre-wrap>        i &lt;- seq_len(lp)</span>\n",
       "<span style=white-space:pre-wrap>        o &lt;- order(p)</span>\n",
       "<span style=white-space:pre-wrap>        ro &lt;- order(o)</span>\n",
       "<span style=white-space:pre-wrap>        pmin(1, cummax((n + 1L - i) * p[o]))[ro]</span>\n",
       "<span style=white-space:pre-wrap>    }, hommel = {</span>\n",
       "<span style=white-space:pre-wrap>        if (n &gt; lp) p &lt;- c(p, rep.int(1, n - lp))</span>\n",
       "<span style=white-space:pre-wrap>        i &lt;- seq_len(n)</span>\n",
       "<span style=white-space:pre-wrap>        o &lt;- order(p)</span>\n",
       "<span style=white-space:pre-wrap>        p &lt;- p[o]</span>\n",
       "<span style=white-space:pre-wrap>        ro &lt;- order(o)</span>\n",
       "<span style=white-space:pre-wrap>        q &lt;- pa &lt;- rep.int(min(n * p/i), n)</span>\n",
       "<span style=white-space:pre-wrap>        for (j in (n - 1L):2L) {</span>\n",
       "<span style=white-space:pre-wrap>            ij &lt;- seq_len(n - j + 1L)</span>\n",
       "<span style=white-space:pre-wrap>            i2 &lt;- (n - j + 2L):n</span>\n",
       "<span style=white-space:pre-wrap>            q1 &lt;- min(j * p[i2]/(2L:j))</span>\n",
       "<span style=white-space:pre-wrap>            q[ij] &lt;- pmin(j * p[ij], q1)</span>\n",
       "<span style=white-space:pre-wrap>            q[i2] &lt;- q[n - j + 1L]</span>\n",
       "<span style=white-space:pre-wrap>            pa &lt;- pmax(pa, q)</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        pmax(pa, p)[if (lp &lt; n) ro[1L:lp] else ro]</span>\n",
       "<span style=white-space:pre-wrap>    }, hochberg = {</span>\n",
       "<span style=white-space:pre-wrap>        i &lt;- lp:1L</span>\n",
       "<span style=white-space:pre-wrap>        o &lt;- order(p, decreasing = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>        ro &lt;- order(o)</span>\n",
       "<span style=white-space:pre-wrap>        pmin(1, cummin((n + 1L - i) * p[o]))[ro]</span>\n",
       "<span style=white-space:pre-wrap>    }, BH = {</span>\n",
       "<span style=white-space:pre-wrap>        i &lt;- lp:1L</span>\n",
       "<span style=white-space:pre-wrap>        o &lt;- order(p, decreasing = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>        ro &lt;- order(o)</span>\n",
       "<span style=white-space:pre-wrap>        pmin(1, cummin(n/i * p[o]))[ro]</span>\n",
       "<span style=white-space:pre-wrap>    }, BY = {</span>\n",
       "<span style=white-space:pre-wrap>        i &lt;- lp:1L</span>\n",
       "<span style=white-space:pre-wrap>        o &lt;- order(p, decreasing = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>        ro &lt;- order(o)</span>\n",
       "<span style=white-space:pre-wrap>        q &lt;- sum(1/(1L:n))</span>\n",
       "<span style=white-space:pre-wrap>        pmin(1, cummin(q * n/i * p[o]))[ro]</span>\n",
       "<span style=white-space:pre-wrap>    }, none = p)</span>\n",
       "<span style=white-space:pre-wrap>    p0</span>\n",
       "}</code></pre>"
      ],
      "text/latex": [
       "\\begin{minted}{r}\n",
       "function (p, method = p.adjust.methods, n = length(p)) \n",
       "\\{\n",
       "    method <- match.arg(method)\n",
       "    if (method == \"fdr\") \n",
       "        method <- \"BH\"\n",
       "    nm <- names(p)\n",
       "    p <- as.numeric(p)\n",
       "    p0 <- setNames(p, nm)\n",
       "    if (all(nna <- !is.na(p))) \n",
       "        nna <- TRUE\n",
       "    p <- p{[}nna{]}\n",
       "    lp <- length(p)\n",
       "    stopifnot(n >= lp)\n",
       "    if (n <= 1) \n",
       "        return(p0)\n",
       "    if (n == 2 \\&\\& method == \"hommel\") \n",
       "        method <- \"hochberg\"\n",
       "    p0{[}nna{]} <- switch(method, bonferroni = pmin(1, n * p), holm = \\{\n",
       "        i <- seq\\_len(lp)\n",
       "        o <- order(p)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummax((n + 1L - i) * p{[}o{]})){[}ro{]}\n",
       "    \\}, hommel = \\{\n",
       "        if (n > lp) p <- c(p, rep.int(1, n - lp))\n",
       "        i <- seq\\_len(n)\n",
       "        o <- order(p)\n",
       "        p <- p{[}o{]}\n",
       "        ro <- order(o)\n",
       "        q <- pa <- rep.int(min(n * p/i), n)\n",
       "        for (j in (n - 1L):2L) \\{\n",
       "            ij <- seq\\_len(n - j + 1L)\n",
       "            i2 <- (n - j + 2L):n\n",
       "            q1 <- min(j * p{[}i2{]}/(2L:j))\n",
       "            q{[}ij{]} <- pmin(j * p{[}ij{]}, q1)\n",
       "            q{[}i2{]} <- q{[}n - j + 1L{]}\n",
       "            pa <- pmax(pa, q)\n",
       "        \\}\n",
       "        pmax(pa, p){[}if (lp < n) ro{[}1L:lp{]} else ro{]}\n",
       "    \\}, hochberg = \\{\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin((n + 1L - i) * p{[}o{]})){[}ro{]}\n",
       "    \\}, BH = \\{\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin(n/i * p{[}o{]})){[}ro{]}\n",
       "    \\}, BY = \\{\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        q <- sum(1/(1L:n))\n",
       "        pmin(1, cummin(q * n/i * p{[}o{]})){[}ro{]}\n",
       "    \\}, none = p)\n",
       "    p0\n",
       "\\}\n",
       "\\end{minted}"
      ],
      "text/markdown": [
       "```r\n",
       "function (p, method = p.adjust.methods, n = length(p)) \n",
       "{\n",
       "    method <- match.arg(method)\n",
       "    if (method == \"fdr\") \n",
       "        method <- \"BH\"\n",
       "    nm <- names(p)\n",
       "    p <- as.numeric(p)\n",
       "    p0 <- setNames(p, nm)\n",
       "    if (all(nna <- !is.na(p))) \n",
       "        nna <- TRUE\n",
       "    p <- p[nna]\n",
       "    lp <- length(p)\n",
       "    stopifnot(n >= lp)\n",
       "    if (n <= 1) \n",
       "        return(p0)\n",
       "    if (n == 2 && method == \"hommel\") \n",
       "        method <- \"hochberg\"\n",
       "    p0[nna] <- switch(method, bonferroni = pmin(1, n * p), holm = {\n",
       "        i <- seq_len(lp)\n",
       "        o <- order(p)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummax((n + 1L - i) * p[o]))[ro]\n",
       "    }, hommel = {\n",
       "        if (n > lp) p <- c(p, rep.int(1, n - lp))\n",
       "        i <- seq_len(n)\n",
       "        o <- order(p)\n",
       "        p <- p[o]\n",
       "        ro <- order(o)\n",
       "        q <- pa <- rep.int(min(n * p/i), n)\n",
       "        for (j in (n - 1L):2L) {\n",
       "            ij <- seq_len(n - j + 1L)\n",
       "            i2 <- (n - j + 2L):n\n",
       "            q1 <- min(j * p[i2]/(2L:j))\n",
       "            q[ij] <- pmin(j * p[ij], q1)\n",
       "            q[i2] <- q[n - j + 1L]\n",
       "            pa <- pmax(pa, q)\n",
       "        }\n",
       "        pmax(pa, p)[if (lp < n) ro[1L:lp] else ro]\n",
       "    }, hochberg = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin((n + 1L - i) * p[o]))[ro]\n",
       "    }, BH = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin(n/i * p[o]))[ro]\n",
       "    }, BY = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        q <- sum(1/(1L:n))\n",
       "        pmin(1, cummin(q * n/i * p[o]))[ro]\n",
       "    }, none = p)\n",
       "    p0\n",
       "}\n",
       "```"
      ],
      "text/plain": [
       "function (p, method = p.adjust.methods, n = length(p)) \n",
       "{\n",
       "    method <- match.arg(method)\n",
       "    if (method == \"fdr\") \n",
       "        method <- \"BH\"\n",
       "    nm <- names(p)\n",
       "    p <- as.numeric(p)\n",
       "    p0 <- setNames(p, nm)\n",
       "    if (all(nna <- !is.na(p))) \n",
       "        nna <- TRUE\n",
       "    p <- p[nna]\n",
       "    lp <- length(p)\n",
       "    stopifnot(n >= lp)\n",
       "    if (n <= 1) \n",
       "        return(p0)\n",
       "    if (n == 2 && method == \"hommel\") \n",
       "        method <- \"hochberg\"\n",
       "    p0[nna] <- switch(method, bonferroni = pmin(1, n * p), holm = {\n",
       "        i <- seq_len(lp)\n",
       "        o <- order(p)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummax((n + 1L - i) * p[o]))[ro]\n",
       "    }, hommel = {\n",
       "        if (n > lp) p <- c(p, rep.int(1, n - lp))\n",
       "        i <- seq_len(n)\n",
       "        o <- order(p)\n",
       "        p <- p[o]\n",
       "        ro <- order(o)\n",
       "        q <- pa <- rep.int(min(n * p/i), n)\n",
       "        for (j in (n - 1L):2L) {\n",
       "            ij <- seq_len(n - j + 1L)\n",
       "            i2 <- (n - j + 2L):n\n",
       "            q1 <- min(j * p[i2]/(2L:j))\n",
       "            q[ij] <- pmin(j * p[ij], q1)\n",
       "            q[i2] <- q[n - j + 1L]\n",
       "            pa <- pmax(pa, q)\n",
       "        }\n",
       "        pmax(pa, p)[if (lp < n) ro[1L:lp] else ro]\n",
       "    }, hochberg = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin((n + 1L - i) * p[o]))[ro]\n",
       "    }, BH = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin(n/i * p[o]))[ro]\n",
       "    }, BY = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        q <- sum(1/(1L:n))\n",
       "        pmin(1, cummin(q * n/i * p[o]))[ro]\n",
       "    }, none = p)\n",
       "    p0\n",
       "}\n",
       "<bytecode: 0x36f6f910>\n",
       "<environment: namespace:stats>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "p.adjust"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
