{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Automating gene plotting\n",
    "#Need a table of gene/ontology with mean RZS score\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Timp3\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/\"\n",
    "rzs_score_path = \"/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/analysis/gene_ontology_rzs.csv\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"correct_strand_only.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "df = fread(rzs_score_path)\n",
    "df = df[gene == gene_name]\n",
    "df = df[order(mean_rzs)]\n",
    "\n",
    "cell_types = df$ontology\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "        \n",
    "    cov_track = AlignmentsTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        #col = colors[[i]],\n",
    "        #fill = colors[[i]],\n",
    "        strand = gene_strand,\n",
    "        legend = TRUE,\n",
    "        type = \"coverage\",\n",
    "        window = -1,\n",
    "        windowSize = 1,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#TIMP3 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Timp3\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "cell_types = c(\"L6_IT\",\"L23_IT\",\"Astro\",\"SMC\")\n",
    "colors = c(\"#2777B1\",\"#FF7F2B\",\"#9667BA\",\"#3EBECD\")\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"correct_strand_only.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "        \n",
    "    cov_track = AlignmentsTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        strand = gene_strand,\n",
    "        legend = TRUE,\n",
    "        type = \"coverage\",\n",
    "        #aggregation = \"sum\",\n",
    "        window = -1,\n",
    "        windowSize = 1,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "49134"
      ],
      "text/latex": [
       "49134"
      ],
      "text/markdown": [
       "49134"
      ],
      "text/plain": [
       "[1] 49134"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_row$end-gene_row$start"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'chr10'"
      ],
      "text/latex": [
       "'chr10'"
      ],
      "text/markdown": [
       "'chr10'"
      ],
      "text/plain": [
       "[1] \"chr10\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "chr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 3"
      ],
      "text/latex": [
       "\\textbf{png:} 3"
      ],
      "text/markdown": [
       "**png:** 3"
      ],
      "text/plain": [
       "png \n",
       "  3 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAIAAAByhViMAAAACXBIWXMAABJ0AAASdAHeZh94\nAAAgAElEQVR4nOzdeXxc913v/8/sI2k0kiV5UxxncRYnUdIkTQp0gaaBlrZA2bpAudxy6eNS\noD/ovcCvUC5Nci+0BQq3LV0Che5pStNma9KkSezEseN9t2xL8iJr30fr7DPn3D9GmYxlWZrl\nbPOd1/MvzdGZ7/fjRDp6z/f7Pd/j0nVdAAAAUP3cdhcAAAAAYxDsAAAAFEGwAwAAUATBDgAA\nQBEEOwAAAEUQ7AAAABRBsAMAAFAEwQ4AAEARBDsAAABFrBLseC4FAABAtXCtGt16e3vT6bQ1\n1dilvb09FArZXQUAAEBFvKuekU6nU6mUBaXYSNM0u0sAAACoFGvsAAAAFEGwAwAAUATBDgAA\nQBEEOwAAAEUQ7AAAABRBsAMAAFAEwQ4AAEARBDsAAABFEOwAAAAUQbADAABQBMEOAABAEas/\nKxZlm5ycnJyctKYvl8t13XXXeTwea7oDAAAORLAz0ZkzZyZHesPBjNkd6bqMR4Nr165taWkx\nuy8AAOBYBDtztYfjr2+PmN1LVnP98NRmXdfN7ggAADgZa+wAAAAUQbADAABQBMEOAABAEQQ7\nAAAARRDsAAAAFEGwAwAAUATBDgAAQBEEOwAAAEUQ7AAAABRBsAMAAFAEwQ4AAEARBDsAAABF\nEOwAAAAUQbADAABQBMEOAABAEQQ7AAAARRDsAAAAFEGwAwAAUATBDgAAQBFeuwuAYc6ePTs0\nNGR3FQAAQETkqquuampqsrhTgp0KsrpLRBJT3ZpHs7sWAAAgYwvBQCBAsEP5OtZNt9an7K4C\nAADIC+c26Lpufb+ssQMAAFCE1SN2HR0d+a87OzuXvLS4GAAAAJVYOmJXGONWfQkAAICSWD1i\nVzgs19HRseSlxcUAAACohDV2AAAAiiDYAQAAKMKemydyM7D5eVgmYQEAACpn6Yjdyve9clcs\nAABAJey8eWLlgwAAACiJnU+eYBM7AAAAA9l28wSb2AEAABjLnhG7wlso8keWbGsHAACAktg2\nYrckwxHpAAAAKmRPsFvylFi55CkUAAAAKJWdU7ErLLMj5AEAAJSKJ08AAAAowp4Ru6oYkItG\no4FAwOu1c0cYAACA4pFaLuvw4cNTU1M33njj9ddfT7wDAADOZ1teWXbjOkeN5L3pTW+6cOHC\nyZMnu7u7r7/++htuuMHn89ldFAAAwGXZefOEY01PT587dy739YYNG+bn57u6uk6dOhUOh1tb\nW4tvJxKJbAiYUyIAAMAlbBuxc9TgHAAAgALsnIp1bLZbs2bNXXfdpWlabio2m81u3bq1jKnY\nPXv2SHTUpCIBAACWsO2u2NwzxC49bks9y3rllVe4eQIAAFQR1thd1p133sl2JwAAoIqwxu6y\nGhoa7C4BAACgBE55ViwAAAAq5KBnxUo1DOMBAAA4Fs+KBQAAUATPigUAAFAEI3YAAACKsHTE\nLreoboU7JxjJAwAAKBsjdgAAAIqwdMQuPyDHyBwAAIDhLB2xY+86AAAA8zAVCwAAoAiCHQAA\ngCKs3sdu5dlY1t4BAACUjRE7AAAARVg9YseYHAAAgEkYsQMAAFAEwQ4AAEAR9mxQDDN0jq/x\nezS7qwAAALKQ8rlcLuv7tXqNHczgcekiEmy9sa6uzu5aAACAXCuyYcMG6/sl2Knjuuuua21t\ntbsKAABgG9bYAQAAKIJgBwAAoAiCHQAAgCIIdgAAAIog2AEAACiCYAcAAKAIgh0AAIAiCHYA\nAACKINgBAAAogmAHAACgCIIdAACAIgh2AAAAiiDYAQAAKIJgBwAAoAiCHQAAgCIIdgAAAIog\n2AEAACiCYAcAAKAIgh0AAIAivHYXoLjhubqF1Hqze9F1ERGXy2V2RwAAwMkIdia6/vrrGxoa\nrOlrndfb1NRkTV8AAMCZCHYmamtra2trs7sKAABQK1hjBwAAoAiCHQAAgCIIdgAAAIog2AEA\nACiCYAcAAKAIgh0AAIAiCHYAAACKINgBAAAogmAHAACgCIIdAACAIgh2AAAAilj9WbE+n8+C\nOuzldhNwAQBA1XPpum53DQAAADAAI1UAAACKINgBAAAogmAHAACgCIIdAACAIgh2AAAAiiDY\nAQAAKIJgBwAAoAiCHQAAgCIIdgAAAIog2AEAACiCYAcAAKAIgh0AAIAiCHYAAACKINgBAAAo\ngmAHAACgCIIdAACAIgh2AAAAiiDYAQAAKIJgBwAAoAiCHQAAgCIIdgAAAIog2AEAACiCYAcA\nAKAIgh0AAIAiCHYAAACKINgBAAAogmAHAACgCIIdAACAIgh2AAAAiiDYAQAAKIJgBwAAoAiC\nHQAAgCIIdgAAAIrw2l0AVvfAAw/kv77vvvtWPV78Cblz8t8q5vxVSy3vjQAMxEUDqF06nO3+\n++9f9uXljq/6xiUHi2+wjGoBWI+LBlDLmIpVRBkfeVf+oMxnaEBthl80ADgBwU5ZK19/y7tA\n56ZdHnjggfz8S+HXhactexyAk3HRABTAGrvqlr8OlnTBXeECvWqD+ffmzsx/XbjmJn+Qz/eA\n0xh+0Sj+vVw0AAsQ7JzuvvvuW2Ed9KVXxiUuvYivnOpWPa3w4MonLKkcgDUsu2is0NGS01Y+\nh4sGYCCCndMtuaQWvizmc20xee7SkwFUL8suGit0BMAuBLtatOQzMZdjACvjogFUC4KdsoqZ\nS13hNAC1hosGoADuilWN7StUCu99s7cSAMWw/VeViwZgIEbsnO5yy5NXXbZc5Lrmss8vph0+\n1gPW46IB1DKXrut21wAAAAADMBULAACgCIIdAACAIgh2AAAAiuDmiRoVi8WeeuopEbn77ruv\nueYau8sBUAVGR0dffvllEbn33ntbW1vtLgfAMhixAwAAMF06nZ6ZmTG7F0bsAAAATPfiiy/O\nzMy0trbecsstGzZsMKkXgh0AAIBFpqamXn75ZfPiHVOxAAAAprvnnntuvvlmn88nr8a7bdu2\njY6OGtsLI3YAAACm8/l8HR0dW7duPXv27OnTp9PpdH707qabbmpvbzekF548UYLe3t6uri5N\n0+wuxAC6rsdiMREJBAJeL/kewOqy2WwikRCRYDDo8XjsLgeoYrquZzKZdDqdj2F+v/9d73qX\n3++vsGX+opegt7d3fn7e7ioMlkwmk8mk3VUAqCa5eAfAQKlUanh4+Oqrr66wHUbsShCJRHp7\ne9X4L6Zp2tTUlIg0NjYGg0G7ywFQBVKp1OzsrIisWbOGkX7AKJqmzc3NBQKBt7zlLZW3RrAD\nAABQBHfFAgAAKIKx9BJcuHChu7u7oaGBuUsoLJPJTE9Pi0g4HA4EAnaXA6AKJBKJ3Br0lpYW\nbqwpTygU2rp1a+XtEOxKcOTIkXQ6nVtiAihPvVuFAJiN60Yl1q1b19LSUmEjBLsSbNq0qbe3\nN//S4/H4/X63m+lsKEXTtHg8LmxpAaBomUwmt8FCXV0dfxbLEw6Hm5qaKm+HmydKk0wmu7u7\nz5w5k81mc0fa2to6OjrWrVtnb2GAUaanp59//nkRefOb32zUhpkA1NbX17dv3z4Rede73hUK\nhewup6YRq0sTCARuu+22d7/73TfccENuMGNycvKll17asWPH5OSk3dUBAICaRrArRzAYvP32\n2wvj3djY2Pbt28fGxuwuDQAA1C6CXfkujXcAAAA24uaJSuXi3U033ZRKpRobG+0uBwAA1C6C\nnTECgQA7fgEAAHtxVywAAIAiWGMHAACgCIIdAACAIgh2AAAAiuDmCaCKPfDAA/mv77vvvlWP\nF39C7pzctwpPXvktK5daxrsAGMuai0aR569aKheNcugAqtP999+/7MvLHV/1jUsOLnv8cuev\nqrx3ATCQZReNYs4vtVoUialYQHHGjq7xGRpQHkPyVY1gB9Scla+/5V2gc9MuDzzwQH7+pfDr\nwtOWPQ7AybhoVBHW2AFqyl8HS7rgrnyBLvK7ua7zXxeuuckf5PM94DRmXDSKfC8XDQMR7IBq\ndd99962wDvrSK+MSl17EK7xuFr532XbyB5dUDsAall00ivwd56JhBoIdUK2WXFILXxaTz0rN\nc3xcBqqdlReNy91yC7MR7AC8Zsn1lzAHYGVcNJyGYAfUnBVmTy53GhdroJaVetHgimEj7ooF\naoXtsyGF977ZWwmAYtj+q8pFowyM2AHV6nLroFdYH13kCSYp7JdP84D1LLtoGHWR4aJRBpeu\n63bXAAAAAAMwFQsAAKAIgh0AAIAiCHYAAACKINgBAAAogmAHAACgCIIdAACAIgh2AAAAiiDY\nAQAAKIJgBwAAoAiCHQAAgCIIdgAAAIog2AEAACiCYAcAAKAIgh0AAIAiCHYAAACKINgBAAAo\ngmAHAACgCIIdAACAIgh2AAAAiiDYAQAAKIJgBwAAoAiCHQAAgCIIdgAAAIog2AEAACiCYAcA\nAKAIgh0AAIAiCHYAAACKINgBAAAogmAHAACgCIIdAACAIgh2AAAAiiDYAQAAKIJgBwAAoAiC\nHQAAgCIIdgAAAIog2AEAACiCYAcAAKAIgh0AAIAiCHYAAACKINgBAAAogmAHAACgCIIdAACA\nIgh2AAAAiiDYAQAAKIJgBwAAoAiCHQAAgCIIdgAAAIog2AEAACiCYAcAAKAIgh0AAIAiCHYA\nAACKINgBAAAogmAHAACgCIIdAACAIgh2AAAAiiDYAQAAKIJgBwAAoAiCHQAAgCIIdgAAAIog\n2AEAACiCYAcAAKAIgh0AAIAiCHYAAACKINgBAAAogmAHAACgCIIdAACAIgh2AAAAiiDYAQAA\nKIJgBwAAoAiCHQAAgCIIdgAAAIog2AEAACiCYAcAAKAIgh0AAIAiCHYAAACKINgBAAAogmAH\nAACgCIIdAACAIgh2AAAAiiDYAQAAKIJgBwAAoAiCHQAAgCIIdgAAAIog2AEAACiCYAcAAKAI\ngh0AAIAiCHYAAACKINgBAAAogmAHAACgCK/dBThRMpmcmZmxuwoAAFDrfD5fS0tL8ecT7JYx\nMTGxe/duu6sAAAC1bu3atffcc0/x5zMVCwAAoAiXrut21+A4TMUCAAAnKHUqlmAHAACgCKZi\nAQAAFEGwAwAAUATBDgAAQBEEOwAAAEUQ7AAAABRBsAMAAFAEwQ4AAEARBDsAAABFEOwAAAAU\nQbADAABQBMEOAABAEQQ7AAAARRDsAAAAFEGwAwAAUATBDgAAQBEEOwAAAEUQ7AAAABRBsAMA\nAFAEwQ4AAEARBDsAAABFEOwAAAAUQbADAABQBMEOAABAEQQ7AAAARRDsAAAAFEGwAwAAUATB\nDgAAQBEEOwAAAEUQ7AAAABRBsAMAAFAEwQ4AAEARBDsAAABFEOwAAAAUQbADAABQBMEOAABA\nEQQ7AAAARRDsAAAAFEGwAwAAUATBDgAAQBEEOwAAAEUQ7AAAABRBsAMAAFAEwQ4AAEARBDsA\nAABFEOwAAAAUQbADAABQBMEOAABAEQQ7AAAARRDsAAAAFEGwAwAAUATBDgAAQBEEOwAAAEUQ\n7AAAABRBsAMAAFAEwQ4AAEARqwQ7XdetqQMAAAAVcq0a3Xp7e9PptDXV2KW9vT0UCtldBQAA\nQEW8q56RTqdTqZQFpdhI0zS7SwAAAKgUa+wAAAAUQbADAABQBMEOAABAEQQ7AAAARRDsAAAA\nFEGwAwAAUATBDgAAQBEEOwAAAEUQ7AAAABRBsAMAAFBE7Qa7UCjU3NxsdxUAAACGqelg19TU\nZHcVQO0am0vYXQIAqKZ2gx0AGz1+aOBtn3l+//lJuwsBAKUQ7ABYbWoh+emnT2Y1/c+/d3hq\nIWl3OQCgDoIdAKv93ZOds7GUiIzPJT7xyFG7ywEAdRDsAFhqfC7xzImh/Msd3WNPHxta4XwA\nQPEIdgAs9eLpUV2/6MinftQ5F0/bVA4AKIVgB8BSL3aNLTkytZD8p2dP21IMACiGYAfAOol0\ndu/ZZe6EfWR/34nBGevrAQDFEOwAWOeVMxOJdPbS45quf+pHnUumaAEApSLYAbDOS6eXzsPm\nHemLPH54wMpiAEA9BDsAFtF1ebn7ssFORD79VOc4j6MAgAoQ7ABY5MzY3MqPEZuLpz/52DHL\n6gEA9XjtLmBRR0dH/uvOzk5DDgJwlJ0946ue89LpsbNj89etb7SgHgBQjyNG7HKxrLOzM5fJ\nci8rPAjAaXZ2rx7sROSFkyNmVwIAqnJEsMtnMgCqiqeyh/sixZz5/MlRs4sBAFU5Itjl5cfe\n7C4EgMH2nptIZbRizjw5NDMYiZldDwAoySlr7MTkVBcOh1taWgqPBINBMzoCsKw9y+1LfDkv\nnh79L2+61rxiAEBVThmxM3usLpVKRS+m67rL5TKpOwBLHLwwVfzJL5xiNhYAyuGIETsLZmAT\niUQicdE+C+Fw2OfzmdcjgLyFZKZ7ZK748w/2Ts3F0+E6fkMBoDROGbEDoLBDF6ayWgnPC8tq\nejF7owAAlnDEiF3Okr3oOjs7Ozo68gdz43nFHwTgHId6i7ofttD2U6Pvft0VZhQDAApzRLC7\nXBRb9njxBwE4xIHeEhbY5bzcPZ7Jal4PswoAUAIumgDMlUhnOwdnSn3XfCJd5L53AIA8gh0A\ncx3snUpni9rBbokdXSyzA4DSEOwAmOuZ48PlvfEII3YAUCKCHQATZbJa2ZvSnRyaKW+oDwBq\nFsEOgIl2nZmYjaXKe28yo3WVsvsdAKDkYFe4KQkArOyZ40OVvP0os7EAUApG7ACYJavpL54e\nq6SFo/3TRhUDALWAYAfALGfG5ufi6UpaINgBQEnK2aB45dlY9goGkHO0v9KJ1KHp2OR8sq0x\nYEg9AKA8RuwAmOWYEeNtbHoCAMUrZ8TOpDG53EBgYeNLhgZz31rySNlLz2TIEHCII30GBLu9\n5yZ/oWNj5e0AQC1wxLNii5/bLQx/HR0dHR0dnZ2dyx40t2IAq5mNpfqmFipvZ++5ycobAYAa\n4Yip2M7Ozkuj2KUDeACqyNGBaV03oJ1z4/NjcwkDGgKAGlDyiJ3FSSs/mFdhvx6Pp66urvCI\n2+2IUAuoypAFdjn7zk3+yh2bjGoNABRWcrBbdYNio5LfkhnYCidYm5ubN25cukxH03haEWAW\nA3cq2XN2gmAHAMVwxBo7C0xNTU1NTRUe2bJli8/ns6seQG1ZTT9uXLBjmR0AFMm5U7GssQOq\n15mx+YVkxqjWRmbi/VPRza0NRjUIAKoya50Zj5QFatnhCwZvPrfvPIN2ALA6595AULh9Sf5l\n8QcB2KjyZ04ssf/81OonAUDNc9Aau0sD2bIRrfiDAOxi+OMi9p5lxA4AVufcETsAVWpqITkQ\niRnb5sR84sKkAdsdA4DaCHYADGb4ArscZmMBYFUEOwAGO2z0PGzOPjY9AYDVEOwAGOxgrylD\nazu6xhLprBktA4AyCHYAjBRNZk4Nz5rR8kIy8+zxYTNaBgBlmBXsuE0VqE2HLkSymm5S4z84\n2G9SywCghnK2OyncLm7JRsTkOaDGHTBzJ+GDvVPnJxauXRsyrwsAqGolj9gVJrlLHy/BAyeA\nGnfAnAV2eY8fHjC1fQCoauWM2BUOyy0ZoiPYAbUskc6eHDJlgV3eT04M/8933GRqFwBQvRz0\n5IlLHwhWGBPzx4s/CMBiR/qm01nN1C76JqM9o3M3bAib2gsAVKlybp4ofCqrIUN0+QaXHMx1\nUbiYr/iDAKx3fGDagl6e6xyxoBcAqEYlj9jlw1w+P1UepAhkgBpODFoR7J4/OfLRn7/Rgo4A\noOqUM2KXHxtb9luV1QOginUOmrvALqd7ZK5/KmpBRwBQdcpfY7dshuvo6HBmtmtqamprays8\nEggE7CoGUNL4XGJ0Nm5NX08eGWTQDgAuVStPnkgkEnMXy2azum7WNqpADTo+MGNZX48dGtD4\n/QWASzjorlhTJZPJiYmJwiPhcNjn89lVD6AeaxbY5QxNxw72Tr3h2rbVTwWAWlIrI3YAzHbC\nwhE7EXnsEDsVA8BSzg12+VtlC/e3K/4gACtlNb1zyNJg95MTI4l01soeAcD5Sp6KNW9TkksD\n2bIRrfiDACxzpC8yF09b2WMsldl/fupnb1xnZacA4HDOHbEDUEW2nRq1vtNdPePWdwoATlbO\nBsVm1AGgqm0/bUOw20mwA4CLMWIHoFI9o3N9kzbsGNw7sTAQiVnfLwA4FsEOQKVeOGnDcF0O\ns7EAUIhgB6BSz58csavrl7sJdgDwGoIdgIqcGZs/PWzFI2KXtatnfGohaVfvAOA0BDsAFXnc\n1o2C01nthwf6bSwAABylVh4pBsAMmq4/dXTQ3hq+f6Dvw2+9zu1y2VsGUIaspj97YvjQhanh\n6fgdV7W84drWrRub6vweu+tCFSPYASjf3nOTY3MJe2sYjMR29UywU7ECEuls0FdDmWZXz/hn\nnj55dmw+9/KlrjER8bhdb7t5w8ffdcumlnpbq0O1ItgBKN+Pjtg8XJfzpRe633zDWgbtqpSu\ny47usX978czhvki4zveWG9bd96u3het8dtdlorPj85995tRLp8cu/VZW05/vHNnZPf6Z993x\ni7e2W18bqp2jg92Sx5flnwm75MjlDgIwVTKjPd9p2/2whY4NTP/wYP97777K7kKwingqe99j\nxw70TqUy2mc/8PrXX93y+OGBb+06f3Z8cdRqLp5++tjQ0f7p9//UVde0hd528waPW6m8PjIb\n/9STndtOjWq6vsJpiXT2zx4+lM5ov3zHJstqgxocHexyCoNaLsDlE15HR0dnZ+eyB20qFqgh\nL50eXUhm7K5i0T8/e/rtt2xsqvfbXQhW8sATx598dZT3w1/bs6bBPzm/zE3NQ9Oxf372tIjc\nvnnNZz/wemUmJY/0RT767QNF3sed1fS/fOSI2+169+uuMLswqMS5d8UWxjUADvTUsSG7S3jN\ndDT11R1n7a4CK3n80EDhPdRZTV821RU62j/9a1/Y8ePjDvpJK9vOnvEPfXV3SbvzZDX9498/\nYuMG4KhGzg12OR2vqrAdr9fbdDGPx+NiRQ5Qrrl4+uWuZVYI2eihPb3saedYD++98Nc/PFrG\nG+cT6f/53UMf//7hkZm44VVZZlfP+Ee/tT+Z0Up9Yyarfey7B589MWxGVVCSc6dil8zAVjjB\nGg6H169fX3jE4/FoWsm/YwBynuscKeOvlKniqey/7zj78XffYnchuEginf27JzsfOdBXSSNP\nHB585vjw//n1299zZ/WtOTvcF/notw+U/fuSyWqst0PxnBvsjBWJRCKRSOGRLVu2+Hwq33UF\nmOqJw3buS3w5D++98JF7rmelnXNcmFz4o2/uPz+xUHlTqYz2yUePXrO24bYr11TemmWODUx/\n5Bv7EulsJY1kNf2vfnA0XO/7uRvXr342aptzp2INmYEFYIah6djBC1N2V7GMRDr71FEV1mOp\nYTAS+9BX9xiS6nKSGe1PvnNwYt7mrROLNDGf+OsfHP2tL++ai6crby2T1f70Owf3n5+svCmo\nzbnBDoBjPXF4cMW9Guz0qK2POEPeyEz89/599+iswQvjRmfjf/jN/RUOgJktnsp+ZXvPO/5x\n+w8P9q+8rUlJEunsf//6vh3dzlrbCqdx6av9zPX09KRSKWuqWaL4LevK2McuNxXb1dUlIps3\nbw6Hw5UXDNQCTdff+U/b+yajdhdyWU/86Vtv3MhvtJ0GIrEPfXX30HTMpPbvvXnD5z94l9fj\nuLEJTdf/c1/fF1/oNu8+Hp/H/Q/vv/Odt7F3MZbn6DV2y0a04g8CMMOPjw05OdWJyKOH+v/q\nl1jIYZtkRvv9/9hjXqoTkW2nRv/koYOf++27/F4HZbveiYW/efTYwV5zVymks9qff+/QfCL9\nvjewIzeW4aBfCQDOl8lqX3i+2+4qVvH9/X0XJg1b14VSfXPnuf4p06P/9lOjf/KdA1nNEWsC\n0lntiy90v+fzL5md6nKymv7JR4997rkux66IgI0IdgBK8MOD/Rb8za5QPJX9i+8ddsif/Foz\nMZ/415fOWNPXS11jn33mlDV9raB/KvpbX9n1xRe6U9ZuAPTg9p6Pf/9wJuusXYdgO4IdgGLN\nxdPOH67LOTE48zePHnPaTnu14AvPdUctfNDc13ee+5fnu+0K8Zquf2d37699YUfn4IwtBTx5\nZPCj3z4QSznlyX5wAoIdgGL935+crqJHOzx6sP8DX945GDFxpReWGIjEHrN8g8Mvbev+4IO7\nDL/9dlW5gbq/ffKElUH2Ui91jb3ncxZNAaMqEOwAFOXYwPT391f08ADrnR6e/cBXdp4csmc0\npQZ9eVu3LTODR/un3/elnZYNm8VSma/vPPern99xrH/amh5XNhCJ/e6/7f749w9bn27hQJ77\n779/5TOmpqayWUfvGFSelpYWj8czOTkpIk1NTYFAwO6KAOeajqZ+/z/2zhqxz6rFYqnsU0eH\nrl0b2rKu0e5aFHdqaPZvnzxh13L+aDLzxOEBv9f9uivXuE17Driuyw8O9n3kG/u3nRpNO2lx\nmy7SPTL33b0XhqZj128IN9XxXKXa5eh97EzFPnYqiURTXcOzJ4dn+yYXhqfjo7PxyYVkIp3N\nrWVuCHhbQ4GWBv+ahkDA69Z0fXI+mdV1v9c9GIlNzifbQoGWUCAU9G5sqtuyvvGN17Xd3N5s\n2p+G6pPV9A9/bc+es9W95f1H3nbDR++9wYE7nylA1+X7+y98+qmTTtg3+K5rWv/xA3dubKoz\nvOXukbm/f/rk7rMThrdsrKDP86dv3/q7b7rW4+YqVosIdgS7KhNLZY72TR/ui+w7N3m0fzro\n83g9rumowT+iG5rq7rlp/VtvWn/3Na31fkdv92i2eCr7sYcOqrHZ/c3tTf/rPbfeeVWL3YUs\nT9flH358cvfZicf/5K1V9Lni9PDs3z554tCFyOqnWqWp3v+p37z93ps3GNVgNJn59FOdjx4c\nMPAxEma7+5rWv3/fne1rjA+4cDiCHcGuOmSy2lNHhx49NHCkL2LlDIjH7brrmsxSuSUAACAA\nSURBVNa33bThnbe1rwsHLevXIXafnfjHH586PTxrdyFG2rox/DtvvPaX79gUcNLetiLydz/q\n/PYr50Xk4T988x1OTZ+FosnMvzzf/e3d5525s8wH33jN/3jHTaFARR/M0lltZ/f4Z54+6fxd\nfi7VEPD+5t2bf/dN116xpt7uWmAdRYIdjxRTg65L9+jcvnOT/ZHo2Gx8Lp5xu6Q1FJiNp7tG\nZifn7bwf0+N2/fSWtnfc2v4Lt2xc0+C3sRJrHO2f/oenTx7uc9AwjLHWNPh/8db2e27acM3a\nUHtzXTGTVoORWP9UdCaeWkhkdF13uVw+j3s2nkpndU3TZ2KphURmPpGeS6SHpmN1Ps/WjU23\nbmre2t4UTWZyG1I01fmvX9/Y1rjMit6H91544PHjua9/+2eu/uR7bjP232us4en4d/f2fn9/\nnyGPtzdPuM73wZ+55t23X3Fd6SssByKx7+w+/9ihAYf/G1fl87h/543X/N5bttTgR9PapEKw\ny6W6XJ4r/HplBDvnmJxPHh+Y3tkzvv3U6Nhcwu5yVuFxu37q2rZfuv2Kd9za3lDZYIADjc0l\nXjg58uyJ4QPna2j3hIDXvWV9Y3OdX0T8PndTnW99U926xuCaBn8slRmdSfSMzh0bmB436Icz\nXOe7ui1065XNt7Q3Xbuu8eq2hsmF5G/+y8v5BWprGvwvf+LtPuctB0xntR1d448e6t/RNebM\nUbrLubot9Lab17/uyjUbm+s8bpff67l2bWjZNB9NZnafmfjhwf6Xu8eraOJ1VS6XXL8+vHVj\neFNLfZ3fu6bef/36xk0t9a0hNW8cnI2lJheS09FUPJ2di6ejyUxDwLsuHKz3e1IZbWI+mdG0\nloZAc70/fxlvqvOF7b7pJKvpQ9Ox3smFM6Pzg5HYbDw1F09fsab+f//664pvhGC3TLCbWkge\nuHhPoIDX3RDwbmyuL/KTPQqls1o8lU1msjOx9EwsNT6XmJxPjs7GI9Hk2GzizNh8FW2NVijo\n8/zy7Zt+4+7NWzeGgz5PGS2kMtrZ8fnC38FURstdjETE63bVB7xb1jVev37pYEPuvpCMpuc2\n0JqNpYamY+fGF85PLESiSbn4dzrg8zTX+9pCgcY6n8/jbgh4c/9HWkOBlpC/we8VkdHZeOfg\n7P7eydPDswr9LasaPo87ndV8Xl86szg49On33rF1Y3gmlh6djefWHszG03OxlIh4Pe76gLcx\n6G1tCKxrCrY3161tNHIkZiGZuTCxMDaXmFpITkdTQ9Ox4Zn46Gw8spCciaWVyTp1fs81baGG\ngDeayszG0gGvu87vXUimh6fjjrrd1Ww+j7vO72mq87WvqV9T7/e4XQ0Br8/rrvd5PB53U52v\nMegLBfPRx7+xORiu83vdrmgq43W7At7FS19j0GflwlBdl76phZNDs32T0fG5RFbT5xLp6Whq\nLp6aS6Sn5pNlb05e7/dev6HxtivXbGgKNgZ9IuL3uhv83rbGwIbmuga/t+zwp+syHUtNLSQH\nI9HJhaSm6ZMLyamF5GwsPRdPR6KpyfnE5EJy2Y9MXZ/5leI7Um28wRDHB6Y/9tDBZb/lcbvW\nNgbbm+uuaKm/Yk39usZgS8gvIuGgT1zic7vrA8v8gZ9PZHJXw9wf1HRGi19y71gmq0dTGREJ\nB73ziUw0mXG5pO7VZft1Po/f685tNhH0ugM+TyjorfN5UpnXfgTq/V6vZ/EXq8Hv9XpcQZ/H\n7/WEg17Xq79wuV/aTFaLpRYLSGe1WCqbSGdTmex8IpNIZ90uV74dEfF53PV+j9vlCgV9Aa/b\n73XPJzIiktW0aDKT/6eJyHw8nXs5n0iPzSaGpmOD07HxuYThdzY4RCKdfeRA3yMH+twuV3O9\nryHgbWkItDYG1jYGW0P+Or83938qf34ynU28eq2ZiabOTyzsOzdZzJbxLQ3+hoA3msxkNH2h\n4D84lHFpkvirR44U//aA172+qW59ONgQ9Aa9HhGZS6RnY+n5RDqWytT5PCISS2ULewnX+Vwi\nHre7IeANeN1Bn8fjdmU1fWI+cW58oRZ+xuKp7Cm1Fo+WJ53V0nFtLp4eMGIr73Cdz+91hwK+\n1pC/rTG4PhxsCwUaAt6GgDcU9DUGvaGgtyHg9bjd0WQmq2nprJ67BjYU/P0SkVgym9Y0Eclf\n8eKp7PhcYngm3je5MBCJTcwnTHqAWyyVOdY/vfIOhfls11Tna6zz1fu9fo9bRFpCgSvW1Om6\nxFKZoen4xHwimc7mRuLnE5nZeMqaX6xaCXbNzc3r1q0rPOLz+S43Wjl1+SCS1fTR2fjobLxa\n1h4V89mi+BUk+daqfdGJsTRdj0RTkWiqjCvjH72n48tPrDLAnGu83Ooc4Z+e/syfvfsv7a7C\n6VadP7mcZEbrn4qWtLqfX2GYIfdzNTmfvFDdmyOtIv/r48zfo1qZivX7/aFQaMmRbDY7MTEh\nIps2bRocHLSoXAAAgLKsW7duyUDVErUyYpdKpSKR6hhjAwAAKI/j7roCAABAeVSYipWy9rEr\nxHYnAABAAYpMxZYR5gAAABTDVCwAAIAiCHYAAACKWH0qtq6uzuez+SEbZvN6FZmSBgAAtWyV\nmydyD7q2rBoAAACUbfW7Ynt7e9NpJ+6tbKD29vYl2xcDAABUndWnINPptPO3O6mQptXQI58B\nAICquHkCAABAEQQ7AAAARRDsAAAAFEGwAwAAUATBDgAAQBEEOwAAAEUQ7AAAABRBsAMAAFAE\nwQ4AAEARBDsAAABF1G6wC4VCzc3NdlcBAABgmJoOdk1NTXZXAdSqbEb6DtpdBACopnaDHQDb\naFn54f8vX/+QXDhgdykAoBSCHQDLPfYJOfWciMhP/lF0ze5qAEAdBDsA1or0y4mnF78eOSXH\nnrS1GgBQCsEOgLW6tl/08olPykN/KONnbKoGAJRCsANgrdMvXPRS1+TMTnn0r5iTBYDKEewA\nWGhhSgaPL3N8tEtO/NjyagBANQQ7ABbq2nbZkbntX5BMytpqAEA1BDsAFloyD1toZlgO/qeF\npQCAgrx2F7Coo6Mj/3VnZ6chBwE4S3RaevetdMLL/yZ3/JoEQlYVBACqccSIXS6WdXZ25jJZ\n7mWFBwE4zqmfiJZd6YTYtOz+plXVAICCHBHs8pkMgMo6n139nD3flFTM/FIAQE1OmYqVgpE2\nM0JeMBhc8mTYUCjkcrkM7wjA8ubHpf/w6qelYnJ+r2x9m/kFAYCCHDFil2PqXKrf76+7mN/v\n9/l8hncEYHknnxNdE72IM8+8bHoxAKAoB43YmWpubm5ubq7wyJYtWwh2gHXO7xYRKWaUvOdl\n0XVhQB0ASuegETsAytKy0lfEPGzO/DhPGAOA8jgi2HV0dHArK6Cy4U5JLJRwfs8O00oBAJW5\ndH2VNS89PT2plOnbwVu/j11uKrarq0tENm/eHA6HyywdwKp2flW2fb6E8zffKf/tW6ZVAwDK\ncsoau2XTWIUHAThF7/7Szh88KtFpaVhjTjUAoCxHTMUCUFk2LYNHS3uLpskZZmMBoGQEOwAm\nGzwmqXjJ7+p+0YRSAEBxBDsAJit1Hjbn3G7JJI0uBQAUR7ADYLLefeW8KxWX82W9EQBqWMnB\njn1JAJQgnZDB42W+t+clIysBgBrAiB0AM/Uflmy6zPeefcXQUgBAfQQ7AGYqb4FdzsyQTF4w\nrBIAqAHl7GO38mwsW8oBeE15C+zyzu6StquNqQQAagAjdgBMk4rLyKmKWji7y6BSAKAmlDNi\nx5gcgKKMnBItW1ELfQcknRRfwKCCAEBxjNgBME2Fw3Uikk5K30EjSgGAmkCwA2CaoRMGNHJ+\njwGNAEBtKDnYMQ8LoFjDJw1ohGAHAEUreY3dqhsUk/wAiIgkFmSq34B2RntkYVJCbQY0BQCq\nYyoWgDmGT4joRjSkV7QZHgDUkpJH7BiQA1CUIeOuFef3yK3vMqw1AFCXWSN2PFIWqHWV3xKb\nxzI7AChOOfvYmaEwCOYHBSs8CMBOg8cNa2p2VCZ7pe0awxoEAEU5Yo1dLpZ1dnbmMlnuZYUH\nAdhpflzmxoxs8NxuI1sDAEU5ItjlMxkARQwcNbjBs68Y3CAAqMgpU7E5+bE3w1v2er3BYLDw\niNvtdrlchncEQMTQedicC/slkxKv3+BmAUAtDgp25qU6EQmHw+vXry884vF4NE0zoy8Axjxz\nolA6IQNH5JqfMrhZAFCLU4KdqalORCKRSCQSKTyyZcsWn89nUndATdOyRt4Sm3duN8EOAFbm\niDV2Zqc6AJYa65FU3Phmu18yvk0AUItZwY6UBtQuwxfY5UycM36GFwDUUs5U7JJN44zaQ25J\nO7mW8wdzLRd/EIBt+g+b1fKRx+SKW81qHACqn0vXV3mYY09PTyqVyr9cdZe4aslVuTV2XV1d\nIrJ58+ZwOGx3RYAqPvd2mRk2peVgSP7sJfEFVz8TAGpSOSN2hdGto6NjyUsDigJQvebGzUp1\nIpJYkNMvyG2/ZFb7AFDlHHHzBAB1mDcPm3PqOXPbB4BqRrADYCizg935PZJJrX4aANSk8m+e\nyN+4UHgQQK0bOGJu+6m49B+Sa3/G3F4AoDqVPGK38r0R1XLnBABTJKMy1m16L2d2md4FAFSn\nSm+eWPkggNoyfFIseFLfmR3yjr8wvRcAqEIG7GO35FskPKB2jZ62opfJCxIZkJYrregLAKpK\nyVOxS9bSsbQOwGvGeizqqOclizoCgKpSzl2xna/KvSTbAVg0YsmInYic3mZRRwBQVSra7iQf\n78h2ACSblslei/rqPywLkxb1BQDVw4B97Mh2AERExs9KNm1RX7om3S9a1BcAVI9ygl1HR8eS\nGEe2AyCjXZZ2x2wsAFzCsH3suBkWqHWW3TmR07tPEvOW9ggAjlf+zRPLHq+4HgBVy+IRu2xa\nenZY2iMAOB7PigVgBF2z4pkTS3Rtt7pHAHA2gh0AI4z1SHzO6k7P7pJ00upOAcDBCHYAjHB+\nrw2dpmJyfo8N/QKAU5XzSDGT5G6qLVyot+yzy4o/CMA6tgQ7EenaJje+1Z6uAcB5Sg52Zuxp\nsmybhTkvt8FKZ2dn8QcNLxLAZWXT0n/Inq5PPy/v/Cvx19vTOwA4jCOmYi93my2A6jBwVFJx\ne7pOLMjxp+zpGgCcp+QRu+pNYB6Pp/Cly+WyqxJANXbNw+bse0he/17hNxoAHLXGzlRtbW0b\nNmxYclDTNFuKAVRzZqedvU+ck76DcvXddtYAQ8xPSCYp9Wsk0GB3KUC1qpVgF4lEotFo4ZFN\nmzYtGcMDUI7xszJyyuYadjxIsKti6aTsf0iOPi4T50VEvAG58a3ypt+X9pvtrgyoPrUS7DRN\ni8fjS44Q7AADHP+R3RWI9O6T09vkpnvtrgOlGzwmj/+1TF547UgmKSd/Iqefl7s+ID/3EWlo\nsa02oAo54uYJANVK15xy78Lzn5VMyu4iUARdk8OPyqN/KdOD0v2ifOP3Lkp1eZom+78rn3u7\n/OQfbLs1B6hCLl3XVz6jp6cnlbLicmnxPnZbtmzx+XxdXV0isnnz5nA4XH7pQM06v1e+9WG7\ni3jVu/+X3P0Bu4vAipIL8s3fl+GTIiK+oGgZyWZWf1frVfJrn5JNrzO7OkABDgp2FiPYAQZ4\n7BNy7Em7i3hV61Xy0R+Ji4kIB3vm07LvIRERXaSk+5jdbrn7t+Tej4m/zpzKAEXUyho7AMZL\nxeT083YXUWCqT7pflK2stHOqkdNy4OHFr0vdnUbTZN9DMj8u7/u/Rpdlq/lxOf60jJyUmRHx\n10vTBtn0OtnyM9J8hd2VoVoR7ACU69Rzjlv8tOdbBDuHyqTkR/dLhZtMnXpeDj0ir3+vQTXZ\nKhWTZz4tx3+0dDL6yGMiLrniFrn6DXL13XLdm9mjESVhzgJAuY4+YXcFl7hwSIaqdRN1xT31\nvxeX1lXo2b+X8bMGtGMjXZfe/fJv75cjj0k2I8ush9JlqFNe+Zo89Ify778tA8esrxHVi2AH\noCwzQ9Jn0/NhV+AS2fstu4vAJXZ+VY4+bkA7ukg6Id//H5KMrn6yM3U+I194p3zzv8lk7+KR\nlcfjhk7I139XdjxY6WAnagbBDkBZjv1IdEf+pTn1nMyO2l0ECmz7vGz7vDFN5TLQZK88+UmH\n/vitIBqRh/5IfvAXMj1Y2hu1rLz4Rfna79i/EziqAcEOQOl03UE3wxbSRbIZ2f+Q3XVARESy\nGXnib2TnV41v+eRP5MefktV2dXCQgaPyr78pZ14uv4XB4/LVD8jubxhWEhRFsANQuoGjEum3\nu4jl5EZ09n938eFUsFFsRr77h3LkMbPaP/A92f4Fsxo31r6H5BsfkrnxStvRNHnus/LE31Tx\nTDTMR7ADULpjzrttolA6KY/9ZVE738IkfYfkwd+Qc3vM7WXnVx06cpw3Nybf+1N55tNG/jQe\neUy+9MvS/aJhDUItBDsAJYrPysln7S5iNcOn5MUv2l1ErZobl+/+scyNWdHXj+6X/iNWdFS8\nVEySC5KKy8v/Jl/6FenaZnwXc+Py8P8nL36xmiajYRX2sQNQopf/VRILdhdRhN1fk6vulOt/\n1u46as+zn5GkVT8hmZQ8/MfyoW/I+hss6nEFyQXZ8y3Z801JRsXjNX3MeMeDMtYjv/opCYbM\n7QhVhRE7AKWIDMj+h1c/zQk0TR77hEQG7K6jxpzZKaees7TH+Jx85w9s/h8dnZZnPi3/fK+8\n9OXFBXDWrATo2i5ffT93y6IQwQ5A0TRNnvmUZNN211G02Ix8/XdltNvuOmpGMipP/x8b+p2f\nkG9/2IC7E8pzbo88+Ouy7yF77mmY6pN//6C88jU2ukMOU7FQTmxGzu2W/kMy1SeJOYnPiYi4\n3LLhRrn6brnqLll3Hc+JL9OP/1bO7LS7iBLNT8g3/qt88F/lytfZXUoNeOGfZWbYnq6nh+Tb\nH5bf/oqssfApq+d2y65/l/P7S370rbGyaXn+n6Vru7znb6XtaltLgf1c+mpLL3t6elKplDXV\nWGnLli0+n6+rq0tENm/eHA6H7a4IFZgdlYGjMtwpFw7IyOlVdi51u6WhVbwB8QZk7bWy/kbZ\n8kZp7xA3ae/ydE2e+6zsqdqHOgQa5INfkc132l2HupJR2ftteenLNu8bXL9G3v85uer15vai\n69J3UHY8KL37zO2oVB6f/PTvyFv+uwQb7S4FtiHYEeyqVjop53bJ2Vfk/B4DlteE18kdvyGv\nf6+E1xlRnFqSC/LYJ6Rru911lEsXcYn4gvLef5Ibfs7ualZ05DHpflE+UCXbs+XMjsiB78mh\nH0h81u5SRETEF5QPPihX32VK45O9cuLHcuIpR6/d9NfJrb8k93xUQq12lwIbEOwIdlVlZljO\n7JTxHpkdkQsHJBV/7Vv6ao9cLIbHJx3vlJ/6HWm/ueK2VDF4TH748ZIfguRMbo+88UNy2y/L\nuuvsLmU5vfvkO38g2Yz88ROydovd1awmk5ILB+TwD6Vrm2hZu6u5mL9ePvAFufanjWwzMS/b\nPicHH6mGR5npIi4Jr5P3fU423WZ3MbAawY5g52CZlMyPSzIqvfsW89zClIgYFOIukW9VF9l4\no2zYKpteJ7f8otTV6s/GyCl55Wty8rlq+EtWonXXyc1vl2t/Wto7xOsv7b3z45JJiYi43KJr\nMjMs/jpZu0Xmx2XwhJzZIZEBWbNJ2m+RLW+S9TeIq7if1dlRefDXF5eE/txH5J6PlvhPsoqm\nSe8+Of6kdG136PMPcr/Ibo+8/c/lDR80ZonF4DF55M+q7xnELrdcebvcdK9cfbes38pqkxqh\nSLDr6OjIf93Z2VnMWwh2DpWMSv8huXBQ+g/J8Cn7b8D0BuT6N8stvyg3/bx4fDYXY4Hcn+2z\nO6X7RUdPNhnF45W118nGm+SKW2XtFmm7Rhpalp6TXJAzu6R3n4z3yMS5ZfbwW+GDRn2zXHm7\nrL1OGlokuSCZpGhZCTZKy2a5/mcl0PDamf/5MTn9wuLXLZvlT35sxD/POMkFObdbzr4i3dsl\nOi0iZn2+Mlb9Grnp5+X17y1/DD4+K3u/Lbv+w/5rUYUa1sgNb5Wr7pJNt0nLVYS8i2SSEo2I\nv0GCIQVurVMh2OVSXS7PFX69MoKdU2hZmTgvA0dk+KSMnpbRbsdN6+Q0rpO73y9b3iQbb1bz\nmpiMyqFHZP/DMjNkdynWK8gowZC0bJbmTVLXJHpWxs/JiDkfMPz1css7pOOdsvn10ndAvvOR\nWzP/eML7F4vl/MF/Svstxndaqpkh6T8iZ3fJ6eclnbS7mnLpIps65I5flxvvkca1q5ycmJfp\nQYn0ycR5Ge6UCwclFbOkSgt5A7LuOtl4k7ReLY1rpX6N+OokkxBvQOpbpL5ZXG5FZipScUnM\nSXxWYtMyflZGu2T8jMRmFr+bjovHJ7ou8xOL8xIutzSskfo1Ur9GQm3S3C5Xv0E23WbdzSjJ\nqMRmXluumpgVj18aWku62Zlgt1ywmxuT8TPi9khi/rUHtgQbxRsQf50Ew+KrE49PXC7uPCpZ\nNCKRAYn0yVSfTPbKZK9MXXj1r2Y1DADoIv6gbNgqG26U9TdKc7s0XyFrrhRPuTsHJaOSXFj8\nL5BckIUpSS5IJiWZxOIJDS3SslmCTRKol8By+8tnUjIzJBPnZHZUolOia5JOyOzIRQsQPV7x\n14u/XtzexX9GYl4CDRIML86maRnpfvG16x3s8FqwE5Hwegmvl4YWaWgRl0dEJBpZTBhutwRC\nkpgXX1DarpWWzYujjJmkiIiWkUBI6pulaaOISCYl8TnRNfEGRNckMb84cJiXiklsZvGPX3RK\nkrHFpuIzMjNUxWHucpqvkJbN0twurVdLqFUySZkekrHuxf+8ubUfyPEGpLldGtdJsFECDeLx\niy8o3oCEWiXUJqG14q8Xt0fqm6W+WbwBGypMJyU+u5jb5sdlZkiiEUnMSWxWFiZlZnBxbcNK\nLv27s9xforqwiEtEF29QgiFpvUaa26WhRQIh8Xgl0CiBBmlctxgKi5zbScVlZkgi/TLaJSOn\nZaxHZoZFLhPJ7i9qKjKHfeyWM3JaHi56gYu/Tjw+8deLLyjBJvHXi8crwbAEQ6/+GXZJfbPU\nrxF/vQRD4m8QX5346xbfHgyLiGRTkk4snhxslMS8iC6ZpKSTko69NlqQzUgqJok5SSclk5RA\ng7g9i2/J8dWJ99UfKY9PfPXiC4ovIIHcCfpiVE3Mr/QvyqYk/Wom0BeLEhFx+8Rf/9ppyQXR\nlwytuSQYlnRcUjFJxRb/TkSnZX5cFiZlfkIWJi/6i7KU41OdiLhE0gkZOCoDR1876PFK08bF\nX/Lc39Tc/3p/g9Q1ib9+6Squ3KMkZ4bk/D4ZP1NaAYGQuN2LPzYiko6/uu6wUBkRuRpSdU2Z\nGyvuWavm36qs2I/GzFBNjkmXJZNc/Pi9KpdbmjdK8xWLH0JyvH5Zs0ka1y7+ALk9i9dDERFd\nYrOSTYsvICISaJRMUrKpxT9tOemEZFMiIqmYuNziC4qIZDMSnVrMQ8tc+spw6Q/3cj/urwXE\nOZkfl4nzKzXpr5e6sAQbpaFV1l0nobUiuU9Q07IwJbGILEzJwsRFn70NVSvBrrm5ed26i7ax\n8Pl8lx2tjEZKaDoVF4kX8bHgEisPdJfRYJEdGdgycrIZiQyUtyJt2y8+eO+zHynhDblHcK7y\nP7GMv8Mq/el2hPdnPvaf3s+V9JabxZGBgx8NrErXZHpIph35A2y93NDG7KjIGTm/1/r+a2Uq\n1u/3h0KhJUey2ezExISIbNq0aXBQid0cAACAutatW7dkoGqJWhmxS6VSkUgp43AAAADVRsWb\n+wAAAGqSClOxUtY+doXY7gQAAChAkanYMsIcAACAYpiKBQAAUATBDgAAQBGrT8XW1dX5fIo/\nItPrVWRKGgAA1LJVbp7Qdd3lYntKAACAKrD6XbG9vb3ptAkPwHaS9vb2JdsXAwAAVJ3VpyDT\n6bTztzupkKZpdpcAAABQKW6eAAAAUATBDgAAQBEEOwAAAEUQ7AAAABRBsAMAAFAEwQ4AAEAR\nBDsAAABFEOwAAAAUQbADAABQBMEOAABAEbUb7EKhUHNzs91VAAAAGKamg11TU5PdVQAAABjG\na3cBAAAAZknGMo9/9kBdo//6N2y46Y1XiMvugkxGsAMAAMra8Z1TkeEFERnqjtSH/Vfdutbu\nisxVu1OxAABAbadfGTp3eCz/8tCPe20sxhoEOwAAoKBsRjvw5LnCI2O9swOnpuyqxxoEOwAA\noKAz+0eis8klB/c/eU7L6rbUYw2CHQAAUI4uR5/ru/Tw+IXZVx7ptr4cyzj65omOjo7Cl52d\nnUsO5o5c7iAAAKhNfZ0T06PRZb/V+dJA09r62+7dbHFJ1qiCEbvOV8mrAW7Jy2UPAgCAmnVm\n/+gK333lke6f/OuxdDJrWT2WcW6wy8c1uwsBAADVJJvWLpyYXPmc80fGT78yZE09VnJusMvp\neJXdhQAAgOrQf2oqncisepqSd8g6d41d4VhdLttVMnrX1ta2YcOGJQc1TSu7QQAA4EznC/au\nW8FwdyST1rw+pw9ylcS5wc5YkUgkGr1oEeWmTZs8Ho9d9QAAADNoGa3vxEQxZ2bS2siZ6Stv\nbjW7JCs5N9gZu8ZO07R4PL7kCMEOAADFdO0dScZWn4fNGTg5pViwU2r4EQAA1LJsWjv09Pni\nz+8/tco9FlXHucEuv3dJ4dBd8QcBAECtOf5i/8J0ovjzp0eiM2Mx8+qxnnOnYuUyEa34gwAA\noIbocuz5ZZ42sbJTLw++8b03mFGOLZw7YgcAAFC8yMhCfD5V6ru69gxlUursVEywAwAAKhg5\nO1PGu5KxzNmDKz2moroQ7AAAgApGz5UT7ETk1C51HkFBsAMAACoYOTNd3hsnLsxl0oo8s4Bg\nBwAAql50JjkfKeF+2EKapk8NzBtbj10IdgAAoOoNlztclzN+YdaoSuxFBSz2igAAIABJREFU\nsAMAAFWv7AV2OeN9c0ZVYi9H72OXV7jzcO7rnPz2dcseBAAANWKismQ2fkGRYFcFI3aFoS2f\n8ApD3rIHAQBAjdB1PTK8UEkLM+PR4p8w62ROD3YdHR2MwAEAgBXMjsXSyco2GdZlckCFQbtK\np2KrZQ7U7/eHQqHCI16v1+Vy2VUPAAAwyuSgAfe0jl+Yu+LGlsrbsVdFI3ZLJj0NnwM1cLiu\noaGh7WIEOwAA1DA1VNE8bE6F99U6RPkjdoU3NOSPGBjFjJ2EnZ6enp6+6H/Yli1bfD6fUe0D\nAAC7TBqxC91wdyST1rw+p69SW1lFU7FLgldnZ6exg3ZmjwgCAAAFGLK9cCatjZyZvvLm1sqb\nspFzp2I7C+SPGNg+AABQQHw+FZ1NGtJU/8lJQ9qxUfkjdrnxuSVhzuzstaTTXHfLHgQAALVg\nyog7J3L6O6fe9F6jGrNHpVOx1twVW9jysr0Q5gAAqE2j5wx7GtjMWHRuMh5uqzOqQetVut0J\niQoAANjo/JExw9rSZbgnEm67wrAGLVf+GrtL52EBAACsNDseM2Svk0UuiQxFDWvNDtV9Ty8A\nAKhl5w+PG9vg1LBhK/ZsUX6wYxIWAADY65yB87AiIhIxcPzPDpVuUHzpbCyBDwAAWGBhOjHR\nb/ADXmNzqfh8qq7Rb2yzlmEqFgAAVKWx3lnRjW82MlzFg3YV7WNnYB3LWnYvleIPAgAAhRny\nJLFLRYajV9zYYkbLFqjorthVj1Qi/yzaXFArnPkt5iAAAFCbWcFuqIrvnzByKtbYZ8UWPkwM\nAABgiUmjF9jlTNXaVGw+vVkwNsaDwgAAwKWiM8nYXMqMliPDUdFFXGa0bbpKnzyxhOHxKz+7\n2tHRUUnj4XC4peWi+fJAIFBpcQAAwCYmzcOKSDqRmZuq1geLlRPs8mGrikbRUqlUPB4vPBIM\nBl2u6kzjAADUPJPmYRcbH5ivoWCXU0WpTkQSiUQikSg8EgqFfD6fXfUAAOAcwz3Tx7f1xeZS\nazY03Hbv5tZNjXZXtLrJQRNvcZgcmL/2jnXmtW+eiqZi84N2ZqyEy9/ralSDAADgUjNj0Wcf\nPJqMZURkrHe2a+/wpq0tt//C1Vfe3Gp3aSuZ6Dcx2Bm+77FlDNjupPAWCmPvipVXV9flXxZ/\nEAAArCqdyDzz5WO5VLdIl8HTkae+cPjC8Qn76lpFbC41PxVf/bxymbeAz2wVjdgVRigz9pBb\nNqIVfxAAAKyse+/IzFh02W9t/+bJ9/31T4daghaXVIyx8zOmth+bTcbmUvXh6nuwWKX72BUO\nkrEzMAAA1eX0K0OX+1Yymt72jU4zntlVubHzs2Z3YerNGeapKNiR5AAAqF4TfXMrzzkO90yf\nOzxmWT3FG+s1PdhNVOdsbPnBLj/7eemELAAAcL5Tuy47XJe397Ez2bRmQTHF07L6eJ/pw2m1\nOGK35KlfpDoAAKpFNqOdPTC66mlzk/H/196dx8d1VXkCv7VXaSmtlmVZlrfEjh2bJGQDmoTO\nB5q1gRkaAvTQzEzTTIcGeuYDdDfdM/PpHnr7wACTJiSdAGEJIYQACSSxszh2FjuObVm2tVjW\nLpVqU6n2vd56549KilKpJNXy6m36ff/Ix3queu/GsqWjc+85Z/QltwzrqVzYk+RZodFPCTY+\ndmwEKWfFFupSAQAAQOV8U1E2x2/8OkJGjrtEXkVJu6XGH7AjhCQjuWQkt/HrVEaCkWII5gAA\nADTHNVppN5N0jJkZCuy7dVtD11O5pdnGlsQW+CYj+9/aJ8+zpFJvg+LiD7EVCwAAoBWusVDl\nL770/MK+W7YRFUziFAXqHg/L8yzfdGyzBHbFId3Y2FiD5saWPKXaiwAAAFBW1J9OBKto8Bv2\npnwz0b6rOxq3pAr5pqNMhpPpWVMReR4koarP2BUPeCgpnpBW8VMKH1Z+EQAAAEoUjsotVLwP\nW1BJpYUMFoaXZXtWIpRNae2YnQRn7BoEiTcAAAAJXX7Zc+qXk3uu72lqs16poNFJibmLy2//\nxDVGo8LbsQvDsg46801H1XO4sBJVB3Ylg1lJ4yMwSSbAGgwGi8VScqWuZQEAAGjH9ODSyV9M\nUJHOnK8x8ZZNsr7JaP+BTmkXVpWQOylzpap3MqLzwC5v9U5oQ4/Z1X/nrq6u3t7ekouiqKLi\nbQAAgAZJRnIv/uQyFeudDjZ7IaBsYLd4uYqCD0m4x8OEEjVUjVSorq3YkjNtUgVhBRLeMBwO\nJxIrOg0ODAyYzerdiQYAAJDK0NE5QYpGdHMXA7d9UsndWP90VOYnpmNM2Jfq2t4i83NrJkFk\nU7Y0tX7ShomUUpZlS65IcmcAAAA1iwczk6d9ktwql+KCrsTW3W2S3K1aokj9s3K0Ji6xeDmk\nocBOyskTDS2SBQAAgGoJnHjq0Qmx7k3YAs+EYh1AQotJrrJRGdJyX5apbZ4k1L4XWdKgLt8z\nr6Ruo+xFAACATS6X5p6575K0cxq8E5Eb37dbwhtWTv592NefOxvjcrzFrvaQKU+9q1wrPit7\nHcEcAABAMVGkz90/vDQbk/bs/9JsjGcFs9Uk2R0r5lMosBN50TMR2X19jyJPr5aUW7EAAACg\nEmeemH49EpK01EHgRf+MTKNai1FKZRsRu9rcRfm6ItcJgR0AAIDejJ5YHD7matDNFTlmNzsU\nyKVlmiS22sJIUJKyYhkgsAMAANCV80/PnXpssnH3XxyTu5lcyJ188aFxmR9ajM3ynivamBuL\nwA4AAEA/wt7U4JHZhj4i4kuF3MmGPqKYKNLnHhjmWUG2J5Y1dyGg7AIqhMAOAABAP4aPLZDG\n92mdfE2axniVcI0EE6GsbI9by/xIUNTCbiwCOwAAAJ1Ix5jp83IklqYHl0RBpj7/4yc98jxo\nfUyaGzmxqPQqNobADgAAQCeGj7vkySplk6w8Y1sToax7XC2H284fnU/HGKVXsQH19rHLWz1Y\nrKRl8ToXAQBA/Sil/unY7FAg6E44ux3b93fuf0uf0aSdoeuq4Z+JjR6XL6U0fW5p15u2NPop\nV0551TMClMvxr/5y8t2ffZPSC1mPegO7spNni+O8/LSJ/NiJ1RflXSwAAFRN4MWLzy5cedWb\niubyVwJz8elzSxeemW/ptGcS7MDBrr79nWF30mI3HXz7dq20/ldENsm+8OCohKPDNuQaDfKc\naLY0cOtPFOjEaW/j7l+D2aHA8G7Xde/aqfRC1qTefySFQE3phQAAgPQCc/GXfzYe9qZW/1Yi\nlM0flo8tpQunmoaOzm/Z2drktF33rp3dO1qlXYx3MjJx2nf4joGeXU5p7yyPbIJ9+p4LhfhY\nHhwjLI6F9tzQwHkMCyPBTIJt3P1r89rj0209TTJkK2uj3sBOWna7vbV1xRcCs9lsMCDVDwAg\nN99UdOiZuWq7gjEZLv+WmfNL1//Brhs/sKe2XFEqmpsdCoyf8mYTbN++jq7+FoEVh19wiSKd\nOue/+ubed/zxAW2lBjMJ9jffHIwvZ+R/9NyFQEMDO5WUTZSgIn3+eyPvveu6gUPdSq+lDC39\n3a2H1Wp1Olf8HGYyKTDnDgBgM4supV/52ZU6J36KAr3w7PzM+aXbPnFNhd9ZuRx/8tGJiC/N\nsUJsKV24Pn9pef5S0agoSqbPLYXcyXf96WHJk4INQik9/qMxRaI6QsjCaEjgRFNjdmMToaxq\newILvPjs/cMf/tJNW/e0Kb2WUpslsEskEolEovjK3r17LRaLUusBANhcKBl9cfHME9M8J03N\nZiKUPfLdi1ff0vv2O6+xt6z3xTzkTh7/0VjEV2bPt6yoP/2rfzm779be3/vYfluz2r9NDB9z\nea6ElXo6l+PnLgauvmVbI+584sdj6imbWE3gxcGnZ//wL9+s9EJKbZbADgAAlJKOMSd+crkR\n8cf0uaXAXPz9X7iho7eZEEIoYbIck+Gb221Gk2HyjP/S8wtRf3qj25SilE6e8Qfm4+///A1t\nPU2SL1sq8WDm3JONHTKxofNH5q+6qddglPJoUy7FHb3vYmAuTighKj4z5b4SDrmTakvuIrAD\nAIAG4hjh6e9cqDxhVq1EKPv4189d/+5dWwacZ38znR91ZTQZHK3WOluOxQKZx79x7o/+5lbn\nFodEi5XY6V9OKT6ZPhZIT51b2v8WyZJ2gbn4898feb0QRMVRHSGEUHLxuYU/+LPDSq9jBY01\nKC7uaVL4sOxFAABQg1ceudK4qC6PzfLnfjtz5J4LhQGmokDTMab+yVr51BGb5eu9UQO4x8ML\nI0GlV0EIIUNH5qRqsxILZJ68e0jm8t56zF4IqGHcWTG1Z+xWR2ll4zYEcwAAKjR5xj911q/Y\n46XI90T96WfvH37v566zqqlUlmeFk49OKL2K18WDmakz/mve1lfnfUSRnvjxGM8KkqxKHlSk\nwy+4bvvENUov5Hc0lrEDAACtYNLca7+eUnoVEvBORn7zfwdTERWlkc7+ZkapStiyho7O1T86\ndujIXGA+Xn+eVWYTp325FKf0Kn5HRT9/AAA0AscInith33Q07EmxWZ4YyPZ9HdfevkO1B6d0\n47UnprNJ1XWXrU3Ym/rF115760f3Hfy97Yof/HKPh0dfciu8iJUSoezka74Db99e8x3GT3nP\nH5kjRPXn6lbhWWHsZfdNH9ij9EJeh8AOAHSFyXCZONve22QwGHhOvPDM/KVjC8LKFhtBV2Lk\n+OLbPrb/8B07lFqnvqWiudO/mpodCii9ECmxOf7lh8cXx0J3fPpaW5Ni3z3nLy0fe3CUyjg6\nrEJnn5zZcbCrpdNew3snTvtOPnJF8iXJ5tIx1/b9Hduu6lB6IYQQYtiwSczU1BTL6uRHrmL5\nPnYTExOEkIGBgZL2xQCgOWFv6uJz83MXlwVOtDdbHK3WTJJlUtw6CYA9N/QcvK1/29UdDZ13\nublQcvkVz2tPTHM5NRYcSKJtS9N//KubHU6rzM/NpbnBp2bHX/HIORC2Kt07Wv/DV2622Kro\n/y8KdPCp2QvPzjduVfKw2Ezvveu6/gNdSi8Egd3mCOyySXbxctg7Ec6lOYEXbU2Wrv7Wfbdu\na+20q7xLEEAlQu7kxecWZocCNbczbWqzmS1Gk9m479Zt176j39ak9ra06hSYi5/+9dTSbEzp\nhTTc1t1tH/7STXVOXBAF6puOeq6EmQxvazIfvK3f2e0ghGQSbGA+3tpp79zeYjQaCCHRpfTl\nVzyTp32s6sPl/gNd773rug1jO54TQ4uJwHx89EV3MqyuqtKaGU2Gt3/8mmtv71d2GQjs9BzY\nRXypc0/Oeicj69XqG4iz29G1vbWtp6lnl3Pn4S1IXYCGLM3Gzh+Zc49L2fnW3mL5/T+5dvd1\nKp3wrUIRX2rwqdmgK5FUU3lBo+042HXHp69tbrfV9nbfdPTEjy8XxzRGo6GrvzWX5pKRbL6A\nwGw1dW1vEXix0MZFE7Zd1f6eP7/O0bpmRjO4mHjugRHdxHMlbvnwVTe+b7eCC9BJYJfvYJdX\nYesT/QV2okA5RrDYTJ4r4YWR4LIrEXInqz2HYbGZzFYTFemWnc7ePe0tnXZnt2PLTmdVqXUA\naQm8uDgWCizECSFbdjh3Hu5ediVmzi95J6KxQNVDBSq054aea9+xY/v+DoMBOe015dLcyPHF\ni88viEq3yVWExW6+7l0D196+o6mabdlskj3/9Nzlkx4VnpOTACXEQGzNlls+uHffrdusDjMh\nRODFheEgk+F4Vgx7k9ODS4JEk+XU6R2fOniwjjqSOukhsCvuS1x5j2I9BXaxpfTlk56ps/4G\nVVwbjAZHq9XRarU1me3Nli07nT07nVsGnI2YokgpTUcZUaSOViuiyU2LijQVY9LRXCqac42G\nXKNBJqPMDpSj1brzcPeu63p2HOg0Wzf1X0gmzcUCmVya41nBZDFyOX7xcnju4rK2uo41gsls\nvPYd/Te+f499oy+JsUB6/KT3yqtedXY8lpzJbOzZ1dbSafdNReqcAqItRqPh8DsHbvngXkW+\naKAqVjNEkcaW0gIvmszG5nYbxwhhTyq0mPBORb1TkYY2/qEizcSZTPz1f5ZzF5fzv2jpsDu3\nOPIRWEuH3dFqtbdYtux0tm2paLQim+WjS+mwJxnyJKO+dL7VeCbOvD4j3EDaups6tjU3t9us\nDrPRaLDYzbYm89Y97Z3bmqWdS6gSsUAm6Erkv3GKgphLc6lILh1jUpEcxwoCJ7Z2OxwtFp4V\nOVbgWcHebHF2O7Zf09ne08xkOS7HiwI1W00mi9HmsBBC2BxPKbU5LBabydZkdnY7jGbV7bNz\njLC8EE+Gc7k0F3InI95kLs1lU5xKMkDZJDtx2jdx2kcIaXJaWzrtrZ2Oru0tW3a19e5py2cj\ndCziSwUXk4G5mHs8rLb2+uoh8OLI8cXxVzw9u9q6d7Tmv2QRQpgsz7OC2WLicnwinAvMxxLB\nzfVnKPCifyaq9CoUIIp0+JhrcSz0wf9+Y82b9TVDxq5RGbtskh05seifjlodZrPVZLaarHZT\nPnjnGIHJcEajwd5qtVhf3/pkczzPCjwrsjne5jDbW6yF7xmZBBNyJ4OuBMdo4yfj/Dl0s9Vk\nb7HYmy1Wu9loNua7A3CMkE2yqWguEczW3ODKaDK0dNitDjPPic1tttYue2dfS0vHihp7q8Pc\nsa255GIxJsMxGZ5nBY4RMnEmHWeYFMe9kXswmY1mq6m53da3r2Odm1Qik2DDnmQimE3HcqJI\n2RzPpHmO4TlGSEVyqUhOtgI3g9HQ5LQ6Wq0d25q37mprarNZbKZ8HGl1mMtuODa12br7Wyx2\ns8iL2RSXS7EcI3CMwGZ5i81ka7IYzQYqkvz1kvfamiwWm8lkMTIZLhHKRv2vx+4Gg8HqMJss\nRlEQA3PxsDel0Q0pg8HgcFp5VjCZjY5Wa/eO1t497W09jnwm22w15X8GM6kvmN5QxJdyjYam\nzvobPQoMQM3+/D0/f+C5T9Zzh/atTe/7i+vbtzZLtaRK6PzHzYL29vaenp7iKxaLZa2gduwl\n97knZ0suGk0Gg9FgrCxRRCkVeFFVrajlVMjtNYgo0ELyILa05hGrksJGg9FgMBIq/O7TzmQ2\n/gStVR1pMBpMZsOGp68EXhQFWsmDZEBFmo4x6RgTcienzy1V+K7iPwGV/I+oBKW08Fc9m2Qj\nvtTq2Vm2ZovZajQUVZ6LgiiKpPJY1mAgRrPRaDTk/y5t8GKjwWgkRtN6oSQlVBRo8T+E1dT2\nib7plf9z/va/X/3rGvyyXfhYrK7dsY5PXR19eDr/33ruA5LLf3I7PnU1IST68PRH3vrc4tfP\nVPi35SNvfe7x195TcjGY6KxzSbFA5vGvD5osBqOx9h/wnN2OD3/5pspfv1kydlartaWlpeSK\nIAjBYJAQ0t/f7/F4ZFouAAAAQE16enpKElUlNkvGjmXZSCSi9CoAAAAAGkh7hz8AAAAAoCw9\nbMWSmvrYFdN6uxMAAAAAoput2BqCOQAAAACdwVYsAAAAgE4gsAMAAADQiY23Yh0Oh8Ui/eQo\nVTGbdbIlDQAAAJvZBsUTlFIMwAYAAADQhI2rYufn5zlOXV3IJdfX11fSvhgAAABAczbeguQ4\nTv3tTuokiqoYNw4AAABQDxRPAAAAAOgEAjsAAAAAnUBgBwAAAKATCOwAAAAAdAKBHQAAAIBO\nILADAAAA0AkEdgAAAAA6gcAOAAAAQCcQ2AEAAADoBAI7AAAAAJ3YvIFdS0tLe3u70qsAAAAA\nkMymDuza2tqUXgUAAACAZDZvYAcAAACgMwjsAAAAAHQCgR0AAABsImleOBtOPOZejrKc0muR\nnlnpBQAAAADI55uTiyeWo4QQu8n4ob5upZcjMWTsAAAAQIcygvA/Lk1/7fI8XXl9IpnJ/8Kb\nZeRfVaMhYwcAAAB6kxGEvx6eHY2nCCG7Wxx/srM3fz3NC/4sQwihhHgzCOxU4NChQ8Ufjo2N\nlVzMXwEAAIBN6zfeUD6qI4T8cN7/5vbWa9uaCSFz6Ww+gWcgxJPNKbfARtFeYJdXHL3lo7pC\nhHfo0CHEdgAAAJvZaCxV+LVI6elwPB/YTSezheu+LCtSYjQosLzG0dgZu+IYDgAAAGA1SshY\nIl18ZeSN7N1MKlO4yIhiiGVlXVnjaTJjV9h4RYQHAAAAJebT2QTHF1+5kkhzIrUYDTOpbPF1\nb4bpsVnlXV1jaSywK9mBrXzXtbu7u7e3t+SiKIpSLg4AAABUYKRoHzaPE+lkMnPA2TSfXnGu\nzpNlbuholXFpDaexwK5mkUgknV6Rle3v7zeZTEqtBwAAABqksPFabDSeajIb2ZU5Hf11PNFY\nYFfzGTtRFLPZbMkVBHYAAAD6MxZPr744Gk8xq3bqPLrreKKxwA4AAABgHYEcG8iVKYl4NRR/\nNRQvuejRXcZOY1WxxT1NCh+WvQgAAACb0HC5fdi1+LIM3fhVWqK9jF3ZuA3BHAAAAJBylRPr\nYEQxkGN77fopjNVYxg4AAABgHVUFdoSQ6WRm4xdpBwI7AAAA0Ik4xy9mqhsUNr2ys53WIbAD\nAAAAnRiJpao9MzeFjB0AAACACpXtYLe+CQR2AAAAACo0XOUBO0JIlOXCLNeIxSgCgR0AAADo\ngUDpXLqWA3N62o1FYAcAAAB64M0ynFhLWzo9BXba62NXUNyOOP/rPPS0AwAA2IRcVdbDFuip\nMFarGbviSK4Q4a0O8gAAAGCTWEzXGNhdSaR1M39Ck4HdoUOHkJYDAACAYjVn7EIMt1BrUKg2\nGt6KrYrVam1paSm+YjabDQaDUusBAAAAadUc2BFCBiOJ3c12CRejFO0FdrWl65qbm7u7u4uv\nmM1mSnWTeQUAANjUKCHuDFPze89FEnfu6JF2SYqoJbArrlpY/6Lkat6EjUaj0Wi0+MrevXst\nFotE6wIAAAAlhRguzQu1vddAyHAslRNEu0mTR9SKVf0/sFYAJ1vhwqE3FK8HAAAANjNXTR3s\nClhRrGFqhQppbCu2OKAshJiI7QAAADY5V637sAXnI8lbOp2SLEZBtaQc19oMVapStZAslGc7\nGAAAANSmnsqJvOmUHtoU13jGrmzwJHPmrHgNCOYAAAA2LUrIYCRR503mdNGmWPOHBAEAAGCT\nG4unfNl6t2JjHB9lOUnWo6CqA7u1iiSwDQoAAACKOBaISHKfOe23Ka69eAIlCwAAAKA4TqQv\nLsckudVsKntjR6skt1JKLYFd2aQdcnUAAAAgvzOReILjJbnVXH09U9Sg9owdIjkAAABQ3MsS\npeuILuonUDwBAAAAWsVT+lo4LtXdFtI5UePjRqvO2G14tE6GqWKrn1X2IgAAAOjbUCSZqnWS\n2GqMKPpybL/DJtUN5aexjF2h9rb4nF/ZiwAAAKB7J0OS7cPmTSa03aa46oxdSfAkc3oM2TgA\nAADIEyl5NSTZPmzes0vhd27tkPaecqqxeKJkiheRN+RSKqwEAAAA9TgTiUekbil8PprwZZk+\nze7G1l4VS1YecZOzQXFxWFnhE51OZ2dnZ/EVm02rnzMAAAAghDziCkh+T5GSJ32hu/Zul/zO\n8pDmjJ36z7exLJtdSRRFqvHKFwAAgE3rUiw5Gk814s5H/eG0dAUZMqsrY1eg/r3RXC6Xy62Y\nE9LS0mKxWJRaDwAAANTj4Qak6/LiHP/3l+e//qa9JoOhQY9onLoCO/mbjGAiLQAAAFBCLsWS\njbv/YCRx34z3i1f3N+4RDVJjYKdgVezqio2yFwEAAECv4hzPiY09T/XsUnhTBHYlp+hWH6pr\ndFxV9v4I5gAAADaP5Rzb6EekeCHIcFtsGju1pbEGxQAAAABBRuIuJ2XNp7U3OrbGBsUAAAAA\nSgkyDc/YEULm09lbOp0yPEhCyNgBAMBmwYii0ktQnSjLPzjvE7TW/yskU8Yut/GLVKaW4omS\n0lRULQAAgPpFWe4TZy732W0f6Ov6aH+P0stRi29Muk6H4k6z+WM7tPRngq3YtVSdsVuneEK1\n3YkBAABOLMdygjiXzt4z7XnGH1Z6OarwtC90OhQnhPxw3i9PDkwq8mzFutK5BpfeSq/GrdiS\n5Jz6J08AAMAmd3w5Uvj13dPuxYz2dtmkFWK4e2e8+bglIwj/PutVeEHVkCdjlxXEgCwRpIRq\nCexKNmGxAwsAACrnyzLj8XThw5wg/r8pt4LrUYP7Z70ZQSiMVng5GE1pZ46WPBk7Qsh8SmO7\nsdKMFMvLNwqW8IZllR13If8MDAAA0JDjy9GSLbWL0eRsKru3xaHMgpQ2Fk8fC0SKr3AiPRmK\nva+3S6klVS7JC1lBpjqY+XT2bd1t8jxLErVk7PJR1Op0nWxRXcnOb9mLAAAABS8tR0uuUEJ+\n6wsqshjFiZT823SZhOWLq/6U1Ckk4/bogtYKY2s8Y7c6eJInnCpEbwAAABUK5NiZchtqzy9F\n0trZfJTQ0/7QVDKz+vpQJJngePnXUy15DtjlubR2FrPqwK44rlodY8kWdVV7vM9gMFhXMhgM\nG78NAAC073Q4XvZ6VhCPbL7y2CQvPDjnK/tbPKXHtZC0kzOwW8zktFUXW8sZO8WntdZQtNHV\n1dXb21tyUUSnSgCATSDf0aOsRxYDH+rrtps2S7t+Ssi3Jhdja6fl7p/17m9tOuhslnNV1ZKt\ncoIQkhXE5Ry71W6V7Yl1krJ4Qh61leKGw+FEIlF8ZWBgwGzW3v8+AABUJSMIF2PJtX43ynKP\ne4N/PLBVziUp6AdzvvUP0uUE8W9HZr9/8zU9NvWGMnJm7AghrkxOQ4Gdxn5GqbnBCqWUXYlq\nbXwKAADUYDCS5NZtMvvzxcAmOWk3Gk897Fra8GUxjv/5YkCG9dRsVt4WJC5N1U9oLLADAACo\nytGNTtElOP5Rt6rjGKmcqPj83FF/WLVVFAmOn0ymN36ddLRVP6HJwO5QEfJGAq/kQwAAgMlk\n5swalRPFfrG4LPPunvxESl4Jxip8cU4Qn/CqtBfMYDQp85gvbQ3q2bZFAAAMHElEQVQp0dgh\ns7WCNgRzAACw2g/WKP8swYjiQwv+L+8faPR6FDQWT1U1DfYJb/A/7ew1q6+DxGAksfGLJIWt\nWAAAAIV5s8y9M55zFQcBT/pCj3tUmqOSxIvB6vqYRFn+QnTNohOlUCUCuxjHx9W6Mb0aAjsA\nANAbb5b5L+euPOZerupd/zbt1mtsJ1Ba+T5swcnq39JoC+lsVXlHqWgoaYfADgAA1OW33lDZ\nQRGVe2hhia2pU+kRf6ie56rW8UC0hnjoVCgu82m2DQ1GlEkizqVlrcOtBwI7AABQEU6k9816\nHpj11nwHT5YpGW9fOVcmJ+iuGZZIyc8WN+5yslqE5S4nUpKvpx4jMWXWM1FuAps6IbADAAAV\nGY2ncoI4GEnMr7H5FWS478/5kmt0nruSSH97crHm4IwTqTvD1PZe1ToVitU8yb6GDdzGoYSM\nxJUJ7CYTCOwAAACqly93oIQ8tkZvuSP+0MOupf98dnxo1dH+B+d9dw1Nrr5elXntbLpVIiuI\n36+sNHg1SsiJ5ah6UpiudE6pIoaFTC4naGMMKQI7AABQkULN47FApGxvuVPBOCEkzHL/cHm+\nuIkuJ9InvRKckFsrU6hR35pcrLkNm4GQEMOdXHvSrsxGFUrXEUJESqdT2kjaaTKwK/QiLrmy\n+joAAGhIhOUK06I4kX5jwlWSLArk2Jk3vr8mOP4H8/7Cb70ajq8z275yc/KOq2qo5wORmo8b\nFjzhqa64uHGUOmCXN6GR3ViNBXZlQ7fCwInCCAoFVgYAAHUbjCSLI7lzkcSvPcu+LOPLMv4c\nK1J6KhQvfsFTvtDUG6faj0pU0Dqrl61YkZKfLtRSM1FiOJZSyfb0sHIZO6Kd+glNTp5A6AYA\noEtnIyt2/Sgh90x77pn25D+8xtkkUkIJKQxDECn9p/GF7910TYTlpGqEsZRjsoLoMGks8bHa\nq+GYJLOwKCE/Xwz83YFd9d+qHq5MLpBjlXo6JUTmAbU101hgVzO73d7a2lp8xWw2G9Q3KQUA\nQK/OhON9DttAk32tF/CUng2vGCpQ8jU6vxdWctGVyf3vsbnpVFaU6Iy/SMl8OnvQ2SzJ3RT0\n6GL56pMaHAtEPznQu7t5zc9do4mUfHtyUamnE0IMhHgyTJTlOqwWBZdRCc3/RFIhq9XqXMlk\nMiGwAwCQh0Dp18YXPjM48bBrKV/YOJvK+lcmYEZjqdQaTUzWdy6SiLJSTiPQUDfatUwmM2Nx\nyTJMIqUPztdYWiuJJ7zBS4oesCOEUEIeckmwtd1omyVjl0gkEokVPwju3bvXYlF73A0AoA9j\n8XSaFwgh35/zPTjvazWb4xy/1W794c0HWsym/GtOh1VTfRlL/+G2bqVXUZeS3Gf9TgVjLwQi\n79raKe1tK/GEN/jvsx75n7vak97QR7b37GiyKb2Q9WyWjB0AACjoXNHgdpGSfNIukGPvnnIX\nrr+qmrYag5GEWlq31ep8VOLAjhLyj+ML35mWNcCaSmb+bnT27ik3p47RZjyl96kjxFwHAjsA\nAJCASGli7W4jZ9bIxh0LRB5ZDFBCxhNpb1YtIx/CLKeSOtDaZAVxPNGQk/6/9izL1kzuaX/o\ns+cn1BPu550OxX/rDSq9ivXoYSt2bGysuA1KvnIWAABk8HIw5s0ycY4/HohkBPHeN+9ffcQ+\nXNSdbrUHZr2vBGMzKuv+OhhJ7Gl2KL2KGl2KJRuX4nrSFzrc1tKgmxfwlD4kRa+WRvjujPeA\ns3lfa5PSCylPk4Hd6tANwRwAgPx+OO//yYK/+Mr/HJ29/8b9TsvvvrlQQo4HoutHGVcak16q\nx2Ak+fEdW5VeRY3OS9T5payXlmNfvIov/hQ3wguBiILNTdbHiuKXh2f+8dCe69sbHuDWQJOB\nHQAAKIsSct+M5zF36UwCb5b56OmxLTZL8xslEUGGi0hasiqP4ViKEUWbUZMHliQ/YFeMFcVn\nlyJ37uiR/M5ZQTQQYjcZeUofcUnWq6UREhz/leHpPc0Oq9H46V29t3Q6lV7R7yCwAwCA6oiU\nfHtq8Slf+UkPjCh6VHNarmasKJ4NJ27f0q70Qqo2n84uNHjc7a88yx/q67ZL2sM5J4ifG5pc\nZti3dDlHYqmyY4JVhRPpZDJDCPmr4Zk7d/R8/qp+pVf0Ok3+LAIAAEoZiaXuGppYK6rTk6cl\nmlEmsyP+cKMfEcixJVvw9bt72j2fzqZ54Xggqv6orsQv3cvqqbZBxg5UJH8Ke50XGAjptFoO\ntzX3OVTdRghArx5zL987o/Z2D1IZjCQDOXar3ar0QqrAifS5pYgMD3rMvfzu3s7dddSX8JQ+\n6w8neCHNC1PJTHFDHM2hhDy6uPy3B3YqvRBCENiBeozF0/9web6SoUDtFvPdN+xTcLgNwCYU\nYriHXEsqb/QgLZHSo/7wf929TemFVOFkKLZO0xkJ8ZR++dLMv75p7/6aikPn0tl/Hl+YWbtW\nWnOOL0f+bE/fFpvygw8MdKPvo1NTUyyr0sqUeuQnT0xMTBBCBgYGnE4VnXzchFhR/MzgROXz\nqjutlnvfvA95O7WJcfzzS5F2i/kQsqr68ph7+YFZLy/RMFYNcVrMH+rrdpiMQ9FkmhdMBsOf\n7t52s5pOyhNCJhKZu6fd+T6CYZZnRVG2R9tNxjt6Om7vbn9bd1vl71rM5D5/YUqeAFRO2+zW\nNqv5zv6t79zaoeAydBLYFZrYkYpbnyCwU5UfzPl+WuUMvg/2dX9l/0CD1gNVSfOCJ8uEGe5b\nU4shhiOEbLNbH3nLISOmMetCnOM//tpYVpAvXFAzk8Fw197tt21p3/bGFq1ISYzjYhzPibTV\nbOqyWeSspQ2z3H87PxFS+lDaX17d/0f9FdXJRljuc0OTS2ptZVI/o8HwpX07Ptin2Eg6PWzF\n5qO6fDyX71SMtnbaEsixq5smbOjEcvSLV/drtBmBnrgyuS9dmi75vuLPseejCVW1AICaPboY\nQFRXIFB674zn3hnPvtamfzm8dyGd/dr4Qkny6d29nX+9f6el8T/ZhBjuf43NKR7VEULunfF2\nWS2/37NBpooS8k/jCzqO6gghIqXfnFzMCmIjOsJUQg+BHawlxvGnQ3F3BfubHVbLTZ2tSrVZ\n/8Gcj6l+7yDNC6eCcWUz3ionUjKZTOe3kNZ/pcFguKrFcWuns9B7bENRlv+tL5gTxGf84Vi5\nLZWnfSEEdjoQ4/gnNtO5uspNJTOfPX8lwQnCqo2v55cingyzunut1Wh8T2+nJKcUOJEe9Ye/\nN+dNbfSvWx4CpX9/ef6OYOwLV/V3r33O7Nee5aFoA5snq8e9M540LyhyQBOBnT4leeH+We9R\nf7iSWoSCg87mrx7YubNJvqKEfFXUC8s11nA9sxRGYLeW0Xjq6xMud6aKdmJWo/HjO3o+vavX\num4eVKT0QjT5rxOu9fMEr4biUZbrsCp/lBhqFsixXx2Z1WK67ovb++/x1li9W/l7o+yap8TG\nE+my01ofdi29b1vXnTt6Bmr9SpsVxKd8ocfcARX2BHlxOTocS/3z4T0Hnc0lvyVS+rQ//MCs\nr3Clns9RLWuLffeO9i9Ejgc637k1//SPXP7qHe1faNwTf7zgzwjCX1zVL/OZlM0S2LW3t/f0\nrEiKWiyWtc4XHvGHfqTWEXUV4kQaq77V+3gi/YULUzZJe06uTxBpPS3pJ5KZj76GbffycoKY\nrPJsMiuKP3UtHfGHTetuIbGiGF/7+1kBT+lnzk8YDThnp2G1fSVRg2Wu9s2+et67IZ7Sp3yh\nl4Ixu9FIavrHofJPSoTl/mZkdvX3EYHSyMpItKF/zqtNmnoIIaZWS+Hp+SsN9Zh7+cRyrM6v\ngn1223duuLry1+uheKLkjB0pVz9htVpbWlpKrgiCEAwGCSH9/f0ez2bpzAQAAAAa1dPTU5Ko\nKrFZMnYsy0YicvRsBAAAAFAKKgoBAAAAdEIPW7Gkpj52xdDHDgAAAHRAJ1uxaFwHAAAAgK1Y\nAAAAAJ1AYAcAAACgExtvxVos+u8vasRYKgAAANC+jYsnAAAAAEATkKkCAAAA0AkEdgAAAAA6\ngcAOAAAAQCcQ2AEAAADoBAI7AAAAAJ1AYAcAAACgEwjsAAAAAHQCgR0AAACATiCwAwAAANAJ\n4/T0tNJrAAAAAAAJ/H/noScLTmrbhQAAAABJRU5ErkJggg==",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "#TIMP3 3' UTR ZOOM\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Timp3\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "cell_types = c(\"L6_IT\",\"L23_IT\",\"Astro\",\"SMC\")\n",
    "colors = c(\"#2777B1\",\"#FF7F2B\",\"#9667BA\",\"#3EBECD\")\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/\"        \n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = 86345570\n",
    "window_end   = 86349639\n",
    "\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_3utr.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "gTrack <- GenomeAxisTrack()\n",
    "tracks = append(tracks,gTrack)\n",
    "\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#        Sequence Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "#seqTrack <- SequenceTrack(\"mm10.fa\", chromosome = chr)\n",
    "#tracks = append(tracks,seqTrack)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "        \n",
    "    cov_track = AlignmentsTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        strand = gene_strand,\n",
    "        legend = TRUE,\n",
    "        type = \"coverage\",\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "#png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#TIMP3 3' UTR First peak zoom\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Timp3\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "cell_types = c(\"L6_IT\",\"L23_IT\",\"Astro\",\"SMC\")\n",
    "colors = c(\"#2777B1\",\"#FF7F2B\",\"#9667BA\",\"#3EBECD\")\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/\"        \n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = 86347150\n",
    "window_end   = 86347200\n",
    "\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_3utr.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "gTrack <- GenomeAxisTrack()\n",
    "tracks = append(tracks,gTrack)\n",
    "\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#        Sequence Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "seqTrack <- SequenceTrack(\"mm10.fa\", chromosome = chr)\n",
    "tracks = append(tracks,seqTrack)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "        \n",
    "    cov_track = AlignmentsTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        strand = gene_strand,\n",
    "        legend = TRUE,\n",
    "        type = \"coverage\",\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: grid\n",
      "\n",
      "Loading required package: AnnotationDbi\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'data.table'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:GenomicRanges':\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from 'package:IRanges':\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:S4Vectors':\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Cbln2\"\n",
    "cell_types = c(\"Vip\",\"L5 IT\",\"L6 IT\")\n",
    "colors = c(\"#2777B1\",\"#9467BD\",\"#7F7F7F\")\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/\"        \n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = gene_row$start\n",
    "window_end   = gene_row$end\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"correct_strand_only.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "        \n",
    "    cov_track = AlignmentsTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        strand = gene_strand,\n",
    "        legend = TRUE,\n",
    "        type = \"coverage\",\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = \"UTR\",\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Sst\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L23_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6_IT\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#C1ql3 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"C1ql3\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "\n",
    "cell_types = c(\"L5_IT\",\"Sst\",\"L23_IT\",\"L6_IT\")\n",
    "colors = c(blue_color,orange_color,green_color,red_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Vip\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6b\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Col15a1 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Col15a1\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "\n",
    "cell_types = c(\"L5_IT\",\"Vip\",\"L6b\")\n",
    "colors = c(blue_color,orange_color,green_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L23_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6_CT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Lamp5\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Ptpru ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Ptpru\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "\n",
    "cell_types = c(\"L23_IT\",\"L5_IT\",\"L6_CT\",\"Lamp5\")\n",
    "colors = c(blue_color,orange_color,purple_color,grey_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L56_NP\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Rxfp2 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Rxfp2\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "\n",
    "cell_types = c(\"L56_NP\",\"L5_IT\")\n",
    "colors = c(blue_color,orange_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_ET\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Vip\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L56_NP\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Adra1b ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Adra1b\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "\n",
    "cell_types = c(\"L5_ET\",\"L5_IT\",\"Vip\",\"L56_NP\")\n",
    "colors = c(blue_color,orange_color,grey_color,light_green_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Vip\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L56_NP\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6_CT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6_IT\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Cbln2 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Cbln2\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "\n",
    "cell_types = c(\"Vip\",\"L56_NP\",\"L6_CT\",\"L6_IT\")\n",
    "colors = c(blue_color,orange_color,red_color,grey_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Pvalb\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L23_IT\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Barx2 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Barx2\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "\n",
    "cell_types = c(\"Pvalb\",\"L23_IT\")\n",
    "colors = c(blue_color,orange_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "31842673"
      ],
      "text/latex": [
       "31842673"
      ],
      "text/markdown": [
       "31842673"
      ],
      "text/plain": [
       "[1] 31842673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "window_start"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "31916833"
      ],
      "text/latex": [
       "31916833"
      ],
      "text/markdown": [
       "31916833"
      ],
      "text/plain": [
       "[1] 31916833"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "window_end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Pvalb\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L23_IT\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Barx2 Zoom\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Barx2\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "\n",
    "cell_types = c(\"Pvalb\",\"L23_IT\")\n",
    "colors = c(blue_color,orange_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "#window_start = round(gene_row$start-gene_buffer)\n",
    "#window_end   = round(gene_row$end+gene_buffer)\n",
    "window_start = 31845673\n",
    "window_end =   31847673\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual_zoom.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"l\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "        window = -1,\n",
    "        windowSize = 1,\n",
    "        aggregation = \"sum\",\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_ET\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6b\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Fezf2 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Fezf2\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "\n",
    "cell_types = c(\"L5_ET\",\"L5_IT\",\"L6b\")\n",
    "colors = c(blue_color,orange_color,purple_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_ET\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Hs3st5 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Hs3st5\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "\n",
    "cell_types = c(\"L5_IT\",\"L6_IT\",\"L5_ET\")\n",
    "colors = c(blue_color,orange_color,green_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Oligo\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L23_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Astro\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6_CT\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Itgb8 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Itgb8\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "pink_color = \"#E377C2\"\n",
    "\n",
    "cell_types = c(\"Oligo\",\"L23_IT\",\"Astro\",\"L6_CT\")\n",
    "colors = c(blue_color,orange_color,purple_color,pink_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Endo\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Astro\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Phactr2 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Phactr2\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "pink_color = \"#E377C2\"\n",
    "\n",
    "cell_types = c(\"Endo\",\"Astro\",\"L5_IT\")\n",
    "colors = c(blue_color,orange_color,purple_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L23_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6_CT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Prdm8 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Prdm8\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "pink_color = \"#E377C2\"\n",
    "\n",
    "cell_types = c(\"L23_IT\",\"L6_CT\",\"L5_IT\")\n",
    "colors = c(blue_color,orange_color,green_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6b\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6_IT\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Prss23 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Prss23\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "pink_color = \"#E377C2\"\n",
    "\n",
    "cell_types = c(\"L6b\",\"L5_IT\",\"L6_IT\")\n",
    "colors = c(blue_color,green_color,purple_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Endo\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L56_NP\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L6_CT\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Rgs5 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Rgs5\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "pink_color = \"#E377C2\"\n",
    "\n",
    "cell_types = c(\"Endo\",\"L56_NP\",\"L6_CT\")\n",
    "colors = c(orange_color,green_color,purple_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in asMethod(object, strict = FALSE): The character vector to convert to a GRanges object must contain\n  strings of the form \"chr:start-end\" or \"chr:start-end:strand\", with end\n  >= start - 1, or \"chr:pos\" or \"chr:pos:strand\". For example:\n  \"chr1:2501-2900\", \"chr1:2501-2900:+\", or \"chr1:740\". Note that \"..\" is\n  a valid alternate start/end separator. Strand can be \"+\", \"-\", \"*\", or\n  missing.\n",
     "output_type": "error",
     "traceback": [
      "Error in asMethod(object, strict = FALSE): The character vector to convert to a GRanges object must contain\n  strings of the form \"chr:start-end\" or \"chr:start-end:strand\", with end\n  >= start - 1, or \"chr:pos\" or \"chr:pos:strand\". For example:\n  \"chr1:2501-2900\", \"chr1:2501-2900:+\", or \"chr1:740\". Note that \"..\" is\n  a valid alternate start/end separator. Strand can be \"+\", \"-\", \"*\", or\n  missing.\nTraceback:\n",
      "1. genes(txdb) == \"Slc17a6\"",
      "2. genes(txdb) == \"Slc17a6\"",
      "3. pcompare(e1, e2)",
      "4. pcompare(e1, e2)",
      "5. bindROWS(x, list(y))",
      "6. bindROWS(x, list(y))",
      "7. S4Vectors:::prepare_objects_to_bind(x, objects)",
      "8. lapply(unname(delete_NULLs(objects)), coerce2, x)",
      "9. lapply(unname(delete_NULLs(objects)), coerce2, x)",
      "10. FUN(X[[i]], ...)",
      "11. FUN(X[[i]], ...)",
      "12. as(from, to_class, strict = FALSE)",
      "13. asMethod(object, strict = FALSE)",
      "14. stop(wmsg(error_msg))"
     ]
    }
   ],
   "source": [
    "txdb = loadDb(txdb_path)\n",
    "any(genes(txdb) == \"Slc17a6\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L56_NP\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L23_IT\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Slc17a6 ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Slc17a6\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "pink_color = \"#E377C2\"\n",
    "\n",
    "cell_types = c(\"L5_IT\",\"L56_NP\",\"L23_IT\")\n",
    "colors = c(orange_color,green_color,red_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Sst\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"L5_IT\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Vip\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Ubash3b ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Ubash3b\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "pink_color = \"#E377C2\"\n",
    "\n",
    "cell_types = c(\"Sst\",\"L5_IT\",\"Vip\")\n",
    "colors = c(blue_color,green_color,pink_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ABOUT TO MAKE GENE TRACK\"\n",
      "[1] \"ABOUT TO MAKE DATA TRACKS\"\n",
      "[1] \"WORKING ON CT\"\n",
      "[1] \"Oligo\"\n",
      "[1] \"RIGHT BEFORE PLOTTING\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Unc5b ENTIRE GENE\n",
    "\n",
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "options(ucscChromosomeNames=FALSE)\n",
    "\n",
    "#Manually setting gene and cell-types to plot\n",
    "gene_name = \"Unc5b\"\n",
    "gene_padding = 0.05 #how much to pad upstream and downstream of the gene\n",
    "\n",
    "blue_color = \"#2777B4\"\n",
    "orange_color = \"#F67F11\"\n",
    "green_color = \"#2EA02D\"\n",
    "red_color = \"#D62727\"\n",
    "purple_color = \"#9467BD\"\n",
    "grey_color = \"#7F7F7F\"\n",
    "light_green_color = \"#BCBD22\"\n",
    "pink_color = \"#E377C2\"\n",
    "\n",
    "cell_types = c(\"Oligo\")\n",
    "colors = c(orange_color)\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "bam_stem = \"/oak/stanford/groups/horence/rob/MERFISH_scRNAseq/merged_by_celltype/\"\n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "gene_buffer = (gene_row$end-gene_row$start)*gene_padding\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = round(gene_row$start-gene_buffer)\n",
    "window_end   = round(gene_row$end+gene_buffer)\n",
    "gene_strand  = gene_row$strand\n",
    "outfile = paste(gene_name,\"_manual.png\",sep=\"\")\n",
    "tracks = list()\n",
    "\n",
    "print(\"ABOUT TO MAKE GENE TRACK\")\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    #strand = gene_strand,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    gene(geneTrack) == gene_name\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "print(\"ABOUT TO MAKE DATA TRACKS\")\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Tracks           #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "data_tracks = list()\n",
    "\n",
    "for(i in 1:length(cell_types)){\n",
    "    print(\"WORKING ON CT\")\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    print(ct)\n",
    "        \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        type = \"a\",\n",
    "        name = ct,\n",
    "        chromosome = chr,\n",
    "        start = window_start,\n",
    "        end = window_end,\n",
    "        col = colors[[i]],\n",
    "        fill = colors[[i]],\n",
    "        stream = TRUE,\n",
    "    )\n",
    "    data_tracks = append(data_tracks,cov_track)\n",
    "}\n",
    "tracks = append(tracks, data_tracks)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 400)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "print(\"RIGHT BEFORE PLOTTING\")\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    sizes = rep(1/length(tracks),length(tracks)), #give equal vertical space to each plot\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = c(\"utr3\",\"utr5\"),\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    fontcolor.exon = \"black\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
