{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "#support for plus/minus strand bam plotting from:\n",
    "#https://support.bioconductor.org/p/56047/\n",
    "library(Rsamtools)\n",
    "strandedBamImport <- function (file, selection) {\n",
    "    if (!file.exists(paste(file, \"bai\", sep = \".\")))\n",
    "        stop(\"Unable to find index for BAM file '\", file, \"'. You can\n",
    "build an index using the following command:\\n\\t\",\n",
    "             \"library(Rsamtools)\\n\\tindexBam(\\\"\", file, \"\\\")\")\n",
    "    sinfo <- scanBamHeader(file)[[1]]\n",
    "    res <- if (!as.character(seqnames(selection)[1]) %in%\n",
    "               names(sinfo$targets)) {\n",
    "        mcols(selection) <- DataFrame(score = 0)\n",
    "        selection\n",
    "    }else {\n",
    "        param <- ScanBamParam(what = c(\"pos\", \"qwidth\", \"strand\"),\n",
    "                              which = selection, flag =\n",
    "scanBamFlag(isUnmappedQuery = FALSE))\n",
    "        x <- scanBam(file, param = param)[[1]]\n",
    "        gr <- GRanges(strand=x[[\"strand\"]], ranges=IRanges(x[[\"pos\"]],\n",
    "width = x[[\"qwidth\"]]), seqnames=seqnames(selection)[1])\n",
    "        grs <- split(gr, strand(gr))\n",
    "        cov <- lapply(grs[c(\"+\", \"-\")], function(y)\n",
    "coverage(ranges(y),\n",
    "width=end(selection)))\n",
    "        pos <- sort(unique(unlist(lapply(cov, function(y) c(start(y),\n",
    "end(y))))))\n",
    "        if(length(pos)==0){\n",
    "            mcols(selection) <- DataFrame(plus=0, minus=0)\n",
    "            selection\n",
    "        }else{\n",
    "            GRanges(seqnames = seqnames(selection)[1],\n",
    "ranges=IRanges(start=head(pos, -1), end=tail(pos, -1)),\n",
    "                    plus=as.numeric(cov[[\"+\"]][head(pos, -1)]),\n",
    "                    minus=-as.numeric(cov[[\"-\"]][head(pos, -1)]))\n",
    "        }\n",
    "    }\n",
    "    return(res)\n",
    "}\n",
    "                                         \n",
    "\n",
    "#args = commandArgs(trailingOnly=TRUE)\n",
    "\n",
    "#print(args)\n",
    "\n",
    "gene_name = \"Teddm3\"\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "srrs_path = \"/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/outputs/gene_ontology/MOp_periphal_ReadZs_10X_gene_ontology.csv\"\n",
    "bam_stem = \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/\"\n",
    "                                         \n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "bam_ontologies = lapply(lapply(bam_paths,basename),tools::file_path_sans_ext)\n",
    "                                         \n",
    "#find which cell-types to plot from SRRS\n",
    "srrs_df = fread(srrs_path)\n",
    "srrs_df = srrs_df[gene == gene_name] #filter to results from the gene of interest\n",
    "srrs_df = srrs_df[bh_p < 0.05] #filter by bh_p\n",
    "srrs_df = srrs_df[order(med_score)] #order by effect size\n",
    "srrs_df = srrs_df[ontology %in% bam_ontologies] #keep only ontologies which have corresponding bams\n",
    "srrs_df = unique(srrs_df,by='ontology') #deduplicate by ontology\n",
    "\n",
    "n_onts_to_plot = 6\n",
    "                                         \n",
    "if(nrow(srrs_df) == 0){\n",
    "    print(\"No matching cell-types\")\n",
    "    stop()\n",
    "} else if (nrow(srrs_df) <= n_onts_to_plot){\n",
    "    #plot all cell types\n",
    "    n_onts_to_plot = nrow(srrs_df)\n",
    "    cell_types = srrs_df$ontology\n",
    "    effect_sizes = srrs_df$med_score\n",
    "} else {\n",
    "    #plot top n/2 and bottom n/2 cell-types by med score (effect size)\n",
    "    n = n_onts_to_plot/2\n",
    "    cell_types = c(head(srrs_df,n)$ontology, tail(srrs_df,n)$ontology)\n",
    "    effect_sizes = c(head(srrs_df,n)$med_score, tail(srrs_df,n)$med_score)\n",
    "}\n",
    "\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = gene_row$start\n",
    "window_end   = gene_row$end\n",
    "outfile = paste(gene_name,\".png\",sep=\"\")\n",
    "tracks = list()\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    feature(geneTrack) %in% c(\"CDS\", \"UTR\")\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "gTrack <- GenomeAxisTrack()                                         \n",
    "tracks = append(tracks,gTrack)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "for(i in 1:n_onts_to_plot){\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    \n",
    "    effect_size = effect_sizes[[i]]\n",
    "    effect_size = format(round(effect_size, 2), nsmall = 2)\n",
    "    \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        name = paste(ct,effect_size,sep=\"\\n\"),\n",
    "        chromosome = chr,\n",
    "        importFunction=strandedBamImport,\n",
    "        stream=TRUE,\n",
    "    )\n",
    "    tracks = append(tracks,cov_track)\n",
    "}\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 800)\n",
    "par(mar=c(5,6,4,1)+.1)\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    type = \"h\",\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = \"UTR\",\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    col=c(\"red\",\"blue\"),\n",
    "    groups=c(\"+\", \"-\"),\n",
    "    fontcolor.exon = \"black\",\n",
    "    legend = F,\n",
    "    cex = 0.6\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
