{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol>\n",
       "\t<li>'Astro'</li>\n",
       "\t<li>'doublet'</li>\n",
       "\t<li>'Endo'</li>\n",
       "\t<li>'L23_IT'</li>\n",
       "\t<li>'L56_NP'</li>\n",
       "\t<li>'L5_ET'</li>\n",
       "\t<li>'L5_IT'</li>\n",
       "\t<li>'L6b'</li>\n",
       "\t<li>'L6_CT'</li>\n",
       "\t<li>'L6_IT'</li>\n",
       "\t<li>'L6_IT_Car3'</li>\n",
       "\t<li>'Lamp5'</li>\n",
       "\t<li>'Low_Quality'</li>\n",
       "\t<li>'Macrophage'</li>\n",
       "\t<li>'Oligo'</li>\n",
       "\t<li>'OPC'</li>\n",
       "\t<li>'SMC'</li>\n",
       "\t<li>'Sncg'</li>\n",
       "\t<li>'Sst'</li>\n",
       "\t<li>'Vip'</li>\n",
       "\t<li>'VLMC'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate}\n",
       "\\item 'Astro'\n",
       "\\item 'doublet'\n",
       "\\item 'Endo'\n",
       "\\item 'L23\\_IT'\n",
       "\\item 'L56\\_NP'\n",
       "\\item 'L5\\_ET'\n",
       "\\item 'L5\\_IT'\n",
       "\\item 'L6b'\n",
       "\\item 'L6\\_CT'\n",
       "\\item 'L6\\_IT'\n",
       "\\item 'L6\\_IT\\_Car3'\n",
       "\\item 'Lamp5'\n",
       "\\item 'Low\\_Quality'\n",
       "\\item 'Macrophage'\n",
       "\\item 'Oligo'\n",
       "\\item 'OPC'\n",
       "\\item 'SMC'\n",
       "\\item 'Sncg'\n",
       "\\item 'Sst'\n",
       "\\item 'Vip'\n",
       "\\item 'VLMC'\n",
       "\\end{enumerate}\n"
      ],
      "text/markdown": [
       "1. 'Astro'\n",
       "2. 'doublet'\n",
       "3. 'Endo'\n",
       "4. 'L23_IT'\n",
       "5. 'L56_NP'\n",
       "6. 'L5_ET'\n",
       "7. 'L5_IT'\n",
       "8. 'L6b'\n",
       "9. 'L6_CT'\n",
       "10. 'L6_IT'\n",
       "11. 'L6_IT_Car3'\n",
       "12. 'Lamp5'\n",
       "13. 'Low_Quality'\n",
       "14. 'Macrophage'\n",
       "15. 'Oligo'\n",
       "16. 'OPC'\n",
       "17. 'SMC'\n",
       "18. 'Sncg'\n",
       "19. 'Sst'\n",
       "20. 'Vip'\n",
       "21. 'VLMC'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[[1]]\n",
       "[1] \"Astro\"\n",
       "\n",
       "[[2]]\n",
       "[1] \"doublet\"\n",
       "\n",
       "[[3]]\n",
       "[1] \"Endo\"\n",
       "\n",
       "[[4]]\n",
       "[1] \"L23_IT\"\n",
       "\n",
       "[[5]]\n",
       "[1] \"L56_NP\"\n",
       "\n",
       "[[6]]\n",
       "[1] \"L5_ET\"\n",
       "\n",
       "[[7]]\n",
       "[1] \"L5_IT\"\n",
       "\n",
       "[[8]]\n",
       "[1] \"L6b\"\n",
       "\n",
       "[[9]]\n",
       "[1] \"L6_CT\"\n",
       "\n",
       "[[10]]\n",
       "[1] \"L6_IT\"\n",
       "\n",
       "[[11]]\n",
       "[1] \"L6_IT_Car3\"\n",
       "\n",
       "[[12]]\n",
       "[1] \"Lamp5\"\n",
       "\n",
       "[[13]]\n",
       "[1] \"Low_Quality\"\n",
       "\n",
       "[[14]]\n",
       "[1] \"Macrophage\"\n",
       "\n",
       "[[15]]\n",
       "[1] \"Oligo\"\n",
       "\n",
       "[[16]]\n",
       "[1] \"OPC\"\n",
       "\n",
       "[[17]]\n",
       "[1] \"SMC\"\n",
       "\n",
       "[[18]]\n",
       "[1] \"Sncg\"\n",
       "\n",
       "[[19]]\n",
       "[1] \"Sst\"\n",
       "\n",
       "[[20]]\n",
       "[1] \"Vip\"\n",
       "\n",
       "[[21]]\n",
       "[1] \"VLMC\"\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bam_paths = Sys.glob(\"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/*.bam\")\n",
    "bam_ontologies = lapply(lapply(bam_paths,basename),tools::file_path_sans_ext)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(Gviz)\n",
    "library(GenomicFeatures)\n",
    "library(data.table)\n",
    "\n",
    "#support for plus/minus strand bam plotting from:\n",
    "#https://support.bioconductor.org/p/56047/\n",
    "library(Rsamtools)\n",
    "strandedBamImport <- function (file, selection) {\n",
    "    if (!file.exists(paste(file, \"bai\", sep = \".\")))\n",
    "        stop(\"Unable to find index for BAM file '\", file, \"'. You can\n",
    "build an index using the following command:\\n\\t\",\n",
    "             \"library(Rsamtools)\\n\\tindexBam(\\\"\", file, \"\\\")\")\n",
    "    sinfo <- scanBamHeader(file)[[1]]\n",
    "    res <- if (!as.character(seqnames(selection)[1]) %in%\n",
    "               names(sinfo$targets)) {\n",
    "        mcols(selection) <- DataFrame(score = 0)\n",
    "        selection\n",
    "    }else {\n",
    "        param <- ScanBamParam(what = c(\"pos\", \"qwidth\", \"strand\"),\n",
    "                              which = selection, flag =\n",
    "scanBamFlag(isUnmappedQuery = FALSE))\n",
    "        x <- scanBam(file, param = param)[[1]]\n",
    "        gr <- GRanges(strand=x[[\"strand\"]], ranges=IRanges(x[[\"pos\"]],\n",
    "width = x[[\"qwidth\"]]), seqnames=seqnames(selection)[1])\n",
    "        grs <- split(gr, strand(gr))\n",
    "        cov <- lapply(grs[c(\"+\", \"-\")], function(y)\n",
    "coverage(ranges(y),\n",
    "width=end(selection)))\n",
    "        pos <- sort(unique(unlist(lapply(cov, function(y) c(start(y),\n",
    "end(y))))))\n",
    "        if(length(pos)==0){\n",
    "            mcols(selection) <- DataFrame(plus=0, minus=0)\n",
    "            selection\n",
    "        }else{\n",
    "            GRanges(seqnames = seqnames(selection)[1],\n",
    "ranges=IRanges(start=head(pos, -1), end=tail(pos, -1)),\n",
    "                    plus=as.numeric(cov[[\"+\"]][head(pos, -1)]),\n",
    "                    minus=-as.numeric(cov[[\"-\"]][head(pos, -1)]))\n",
    "        }\n",
    "    }\n",
    "    return(res)\n",
    "}\n",
    "                                         \n",
    "\n",
    "#args = commandArgs(trailingOnly=TRUE)\n",
    "\n",
    "#print(args)\n",
    "\n",
    "gene_name = \"Timp3\"\n",
    "\n",
    "txdb_path = \"txdb.mm10.sqlite\"\n",
    "bed_path = \"MERFISH_genes.bed\"\n",
    "srrs_path = \"/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/final_outputs/20220103_merfish.csv\"\n",
    "bam_stem = \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/\"\n",
    "                                         \n",
    "bam_paths = Sys.glob(paste(bam_stem,\"*.bam\",sep=\"\"))\n",
    "bam_ontologies = lapply(lapply(bam_paths,basename),tools::file_path_sans_ext)\n",
    "                                         \n",
    "#find which cell-types to plot from SRRS\n",
    "srrs_df = fread(srrs_path)\n",
    "srrs_df = srrs_df[gene == gene_name] #filter to results from the gene of interest\n",
    "srrs_df = srrs_df[bh_p < 0.05] #filter by bh_p\n",
    "srrs_df = srrs_df[order(med_score)] #order by effect size\n",
    "srrs_df = srrs_df[ontology %in% bam_ontologies] #keep only ontologies which have corresponding bams\n",
    "\n",
    "n_onts_to_plot = 40\n",
    "                                         \n",
    "if(nrow(srrs_df) == 0){\n",
    "    print(\"No matching cell-types\")\n",
    "    stop()\n",
    "} else if (nrow(srrs_df) <= n_onts_to_plot){\n",
    "    #plot all cell types\n",
    "    num_ontologies = nrow(srrs_df)\n",
    "    cell_types = srrs_df$ontology\n",
    "    effect_sizes = srrs_df$med_score\n",
    "} else {\n",
    "    #plot top n/2 and bottom n/2 cell-types by med score (effect size)\n",
    "    n = n_onts_to_plot/2\n",
    "    cell_types = c(head(srrs_df,n)$ontology, tail(srrs_df,n)$ontology)\n",
    "    effect_sizes = c(head(srrs_df,n)$med_score, tail(srrs_df,n)$med_score)\n",
    "}\n",
    "\n",
    "                                         \n",
    "gene_locs = fread(bed_path)\n",
    "gene_row = gene_locs[gene == gene_name]\n",
    "\n",
    "chr = gene_row[, get(\"#chr\")]\n",
    "window_start = gene_row$start\n",
    "window_end   = gene_row$end\n",
    "outfile = paste(gene_name,\".png\",sep=\"\")\n",
    "tracks = list()\n",
    "                                         \n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Gene Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "txdb = loadDb(txdb_path)\n",
    "\n",
    "geneTrack = GeneRegionTrack(\n",
    "    range = txdb,\n",
    "    chromosome = chr,\n",
    "    start = window_start,\n",
    "    end = window_end,\n",
    "    name = gene_name,\n",
    ")\n",
    "\n",
    "geneTrack <- geneTrack[\n",
    "    feature(geneTrack) %in% c(\"CDS\", \"UTR\")\n",
    "]\n",
    "tracks = append(tracks,geneTrack)\n",
    "\n",
    "gTrack <- GenomeAxisTrack()                                         \n",
    "tracks = append(tracks,gTrack)\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Data Track            #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "for(i in 1:num_ontologies){\n",
    "    ct = cell_types[[i]]\n",
    "    ct = sub(\" \",\"_\",ct)\n",
    "    \n",
    "    effect_size = effect_sizes[[i]]\n",
    "    effect_size = format(round(effect_size, 2), nsmall = 2)\n",
    "    \n",
    "    cov_track = DataTrack(\n",
    "        range = paste(bam_stem,ct,\".bam\",sep=\"\"),\n",
    "        name = paste(ct,effect_size,sep=\"\\n\"),\n",
    "        chromosome = chr,\n",
    "        importFunction=strandedBamImport,\n",
    "        stream=TRUE,\n",
    "    )\n",
    "    tracks = append(tracks,cov_track)\n",
    "}\n",
    "\n",
    "####################################\n",
    "#                                  #\n",
    "#                                  #\n",
    "#            Plotting              #\n",
    "#                                  #\n",
    "#                                  #\n",
    "####################################\n",
    "png(outfile, width = 800, height = 800)\n",
    "\n",
    "plotTracks(\n",
    "    tracks,\n",
    "    type = \"h\",\n",
    "    from = window_start, \n",
    "    to = window_end,\n",
    "    thinBoxFeature = \"UTR\",\n",
    "    collapse = F,\n",
    "    transcriptAnnotation = \"gene\",\n",
    "    col=c(\"red\",\"blue\"),\n",
    "    groups=c(\"+\", \"-\"),\n",
    "    fontcolor.exon = \"black\",\n",
    "    legend = F,\n",
    "    cex = 0.6\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/L6_CT.bam'"
      ],
      "text/latex": [
       "'/scratch/groups/horence/rob/data/MERFISH\\_scRNAseq/10X\\_mapping/merged\\_by\\_celltype/L6\\_CT.bam'"
      ],
      "text/markdown": [
       "'/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/L6_CT.bam'"
      ],
      "text/plain": [
       "[1] \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/L6_CT.bam\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bam_stem = \"/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_celltype/\"\n",
    "paste(bam_stem,ct,\".bam\",sep=\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<pre class=language-r><code>function (p, method = p.adjust.methods, n = length(p)) \n",
       "{\n",
       "<span style=white-space:pre-wrap>    method &lt;- match.arg(method)</span>\n",
       "<span style=white-space:pre-wrap>    if (method == \"fdr\") </span>\n",
       "<span style=white-space:pre-wrap>        method &lt;- \"BH\"</span>\n",
       "<span style=white-space:pre-wrap>    nm &lt;- names(p)</span>\n",
       "<span style=white-space:pre-wrap>    p &lt;- as.numeric(p)</span>\n",
       "<span style=white-space:pre-wrap>    p0 &lt;- setNames(p, nm)</span>\n",
       "<span style=white-space:pre-wrap>    if (all(nna &lt;- !is.na(p))) </span>\n",
       "<span style=white-space:pre-wrap>        nna &lt;- TRUE</span>\n",
       "<span style=white-space:pre-wrap>    p &lt;- p[nna]</span>\n",
       "<span style=white-space:pre-wrap>    lp &lt;- length(p)</span>\n",
       "<span style=white-space:pre-wrap>    stopifnot(n &gt;= lp)</span>\n",
       "<span style=white-space:pre-wrap>    if (n &lt;= 1) </span>\n",
       "<span style=white-space:pre-wrap>        return(p0)</span>\n",
       "<span style=white-space:pre-wrap>    if (n == 2 &amp;&amp; method == \"hommel\") </span>\n",
       "<span style=white-space:pre-wrap>        method &lt;- \"hochberg\"</span>\n",
       "<span style=white-space:pre-wrap>    p0[nna] &lt;- switch(method, bonferroni = pmin(1, n * p), holm = {</span>\n",
       "<span style=white-space:pre-wrap>        i &lt;- seq_len(lp)</span>\n",
       "<span style=white-space:pre-wrap>        o &lt;- order(p)</span>\n",
       "<span style=white-space:pre-wrap>        ro &lt;- order(o)</span>\n",
       "<span style=white-space:pre-wrap>        pmin(1, cummax((n + 1L - i) * p[o]))[ro]</span>\n",
       "<span style=white-space:pre-wrap>    }, hommel = {</span>\n",
       "<span style=white-space:pre-wrap>        if (n &gt; lp) p &lt;- c(p, rep.int(1, n - lp))</span>\n",
       "<span style=white-space:pre-wrap>        i &lt;- seq_len(n)</span>\n",
       "<span style=white-space:pre-wrap>        o &lt;- order(p)</span>\n",
       "<span style=white-space:pre-wrap>        p &lt;- p[o]</span>\n",
       "<span style=white-space:pre-wrap>        ro &lt;- order(o)</span>\n",
       "<span style=white-space:pre-wrap>        q &lt;- pa &lt;- rep.int(min(n * p/i), n)</span>\n",
       "<span style=white-space:pre-wrap>        for (j in (n - 1L):2L) {</span>\n",
       "<span style=white-space:pre-wrap>            ij &lt;- seq_len(n - j + 1L)</span>\n",
       "<span style=white-space:pre-wrap>            i2 &lt;- (n - j + 2L):n</span>\n",
       "<span style=white-space:pre-wrap>            q1 &lt;- min(j * p[i2]/(2L:j))</span>\n",
       "<span style=white-space:pre-wrap>            q[ij] &lt;- pmin(j * p[ij], q1)</span>\n",
       "<span style=white-space:pre-wrap>            q[i2] &lt;- q[n - j + 1L]</span>\n",
       "<span style=white-space:pre-wrap>            pa &lt;- pmax(pa, q)</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        pmax(pa, p)[if (lp &lt; n) ro[1L:lp] else ro]</span>\n",
       "<span style=white-space:pre-wrap>    }, hochberg = {</span>\n",
       "<span style=white-space:pre-wrap>        i &lt;- lp:1L</span>\n",
       "<span style=white-space:pre-wrap>        o &lt;- order(p, decreasing = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>        ro &lt;- order(o)</span>\n",
       "<span style=white-space:pre-wrap>        pmin(1, cummin((n + 1L - i) * p[o]))[ro]</span>\n",
       "<span style=white-space:pre-wrap>    }, BH = {</span>\n",
       "<span style=white-space:pre-wrap>        i &lt;- lp:1L</span>\n",
       "<span style=white-space:pre-wrap>        o &lt;- order(p, decreasing = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>        ro &lt;- order(o)</span>\n",
       "<span style=white-space:pre-wrap>        pmin(1, cummin(n/i * p[o]))[ro]</span>\n",
       "<span style=white-space:pre-wrap>    }, BY = {</span>\n",
       "<span style=white-space:pre-wrap>        i &lt;- lp:1L</span>\n",
       "<span style=white-space:pre-wrap>        o &lt;- order(p, decreasing = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>        ro &lt;- order(o)</span>\n",
       "<span style=white-space:pre-wrap>        q &lt;- sum(1/(1L:n))</span>\n",
       "<span style=white-space:pre-wrap>        pmin(1, cummin(q * n/i * p[o]))[ro]</span>\n",
       "<span style=white-space:pre-wrap>    }, none = p)</span>\n",
       "<span style=white-space:pre-wrap>    p0</span>\n",
       "}</code></pre>"
      ],
      "text/latex": [
       "\\begin{minted}{r}\n",
       "function (p, method = p.adjust.methods, n = length(p)) \n",
       "\\{\n",
       "    method <- match.arg(method)\n",
       "    if (method == \"fdr\") \n",
       "        method <- \"BH\"\n",
       "    nm <- names(p)\n",
       "    p <- as.numeric(p)\n",
       "    p0 <- setNames(p, nm)\n",
       "    if (all(nna <- !is.na(p))) \n",
       "        nna <- TRUE\n",
       "    p <- p{[}nna{]}\n",
       "    lp <- length(p)\n",
       "    stopifnot(n >= lp)\n",
       "    if (n <= 1) \n",
       "        return(p0)\n",
       "    if (n == 2 \\&\\& method == \"hommel\") \n",
       "        method <- \"hochberg\"\n",
       "    p0{[}nna{]} <- switch(method, bonferroni = pmin(1, n * p), holm = \\{\n",
       "        i <- seq\\_len(lp)\n",
       "        o <- order(p)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummax((n + 1L - i) * p{[}o{]})){[}ro{]}\n",
       "    \\}, hommel = \\{\n",
       "        if (n > lp) p <- c(p, rep.int(1, n - lp))\n",
       "        i <- seq\\_len(n)\n",
       "        o <- order(p)\n",
       "        p <- p{[}o{]}\n",
       "        ro <- order(o)\n",
       "        q <- pa <- rep.int(min(n * p/i), n)\n",
       "        for (j in (n - 1L):2L) \\{\n",
       "            ij <- seq\\_len(n - j + 1L)\n",
       "            i2 <- (n - j + 2L):n\n",
       "            q1 <- min(j * p{[}i2{]}/(2L:j))\n",
       "            q{[}ij{]} <- pmin(j * p{[}ij{]}, q1)\n",
       "            q{[}i2{]} <- q{[}n - j + 1L{]}\n",
       "            pa <- pmax(pa, q)\n",
       "        \\}\n",
       "        pmax(pa, p){[}if (lp < n) ro{[}1L:lp{]} else ro{]}\n",
       "    \\}, hochberg = \\{\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin((n + 1L - i) * p{[}o{]})){[}ro{]}\n",
       "    \\}, BH = \\{\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin(n/i * p{[}o{]})){[}ro{]}\n",
       "    \\}, BY = \\{\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        q <- sum(1/(1L:n))\n",
       "        pmin(1, cummin(q * n/i * p{[}o{]})){[}ro{]}\n",
       "    \\}, none = p)\n",
       "    p0\n",
       "\\}\n",
       "\\end{minted}"
      ],
      "text/markdown": [
       "```r\n",
       "function (p, method = p.adjust.methods, n = length(p)) \n",
       "{\n",
       "    method <- match.arg(method)\n",
       "    if (method == \"fdr\") \n",
       "        method <- \"BH\"\n",
       "    nm <- names(p)\n",
       "    p <- as.numeric(p)\n",
       "    p0 <- setNames(p, nm)\n",
       "    if (all(nna <- !is.na(p))) \n",
       "        nna <- TRUE\n",
       "    p <- p[nna]\n",
       "    lp <- length(p)\n",
       "    stopifnot(n >= lp)\n",
       "    if (n <= 1) \n",
       "        return(p0)\n",
       "    if (n == 2 && method == \"hommel\") \n",
       "        method <- \"hochberg\"\n",
       "    p0[nna] <- switch(method, bonferroni = pmin(1, n * p), holm = {\n",
       "        i <- seq_len(lp)\n",
       "        o <- order(p)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummax((n + 1L - i) * p[o]))[ro]\n",
       "    }, hommel = {\n",
       "        if (n > lp) p <- c(p, rep.int(1, n - lp))\n",
       "        i <- seq_len(n)\n",
       "        o <- order(p)\n",
       "        p <- p[o]\n",
       "        ro <- order(o)\n",
       "        q <- pa <- rep.int(min(n * p/i), n)\n",
       "        for (j in (n - 1L):2L) {\n",
       "            ij <- seq_len(n - j + 1L)\n",
       "            i2 <- (n - j + 2L):n\n",
       "            q1 <- min(j * p[i2]/(2L:j))\n",
       "            q[ij] <- pmin(j * p[ij], q1)\n",
       "            q[i2] <- q[n - j + 1L]\n",
       "            pa <- pmax(pa, q)\n",
       "        }\n",
       "        pmax(pa, p)[if (lp < n) ro[1L:lp] else ro]\n",
       "    }, hochberg = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin((n + 1L - i) * p[o]))[ro]\n",
       "    }, BH = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin(n/i * p[o]))[ro]\n",
       "    }, BY = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        q <- sum(1/(1L:n))\n",
       "        pmin(1, cummin(q * n/i * p[o]))[ro]\n",
       "    }, none = p)\n",
       "    p0\n",
       "}\n",
       "```"
      ],
      "text/plain": [
       "function (p, method = p.adjust.methods, n = length(p)) \n",
       "{\n",
       "    method <- match.arg(method)\n",
       "    if (method == \"fdr\") \n",
       "        method <- \"BH\"\n",
       "    nm <- names(p)\n",
       "    p <- as.numeric(p)\n",
       "    p0 <- setNames(p, nm)\n",
       "    if (all(nna <- !is.na(p))) \n",
       "        nna <- TRUE\n",
       "    p <- p[nna]\n",
       "    lp <- length(p)\n",
       "    stopifnot(n >= lp)\n",
       "    if (n <= 1) \n",
       "        return(p0)\n",
       "    if (n == 2 && method == \"hommel\") \n",
       "        method <- \"hochberg\"\n",
       "    p0[nna] <- switch(method, bonferroni = pmin(1, n * p), holm = {\n",
       "        i <- seq_len(lp)\n",
       "        o <- order(p)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummax((n + 1L - i) * p[o]))[ro]\n",
       "    }, hommel = {\n",
       "        if (n > lp) p <- c(p, rep.int(1, n - lp))\n",
       "        i <- seq_len(n)\n",
       "        o <- order(p)\n",
       "        p <- p[o]\n",
       "        ro <- order(o)\n",
       "        q <- pa <- rep.int(min(n * p/i), n)\n",
       "        for (j in (n - 1L):2L) {\n",
       "            ij <- seq_len(n - j + 1L)\n",
       "            i2 <- (n - j + 2L):n\n",
       "            q1 <- min(j * p[i2]/(2L:j))\n",
       "            q[ij] <- pmin(j * p[ij], q1)\n",
       "            q[i2] <- q[n - j + 1L]\n",
       "            pa <- pmax(pa, q)\n",
       "        }\n",
       "        pmax(pa, p)[if (lp < n) ro[1L:lp] else ro]\n",
       "    }, hochberg = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin((n + 1L - i) * p[o]))[ro]\n",
       "    }, BH = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        pmin(1, cummin(n/i * p[o]))[ro]\n",
       "    }, BY = {\n",
       "        i <- lp:1L\n",
       "        o <- order(p, decreasing = TRUE)\n",
       "        ro <- order(o)\n",
       "        q <- sum(1/(1L:n))\n",
       "        pmin(1, cummin(q * n/i * p[o]))[ro]\n",
       "    }, none = p)\n",
       "    p0\n",
       "}\n",
       "<bytecode: 0x36f6f910>\n",
       "<environment: namespace:stats>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "p.adjust"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
