{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## First issue is that 10X and MOp cluster_label celltypes are not explicitly matched\n",
    "\n",
    "I have ontologies like L5 IT Tcap_2 for 10X, but L5_IT_1 from MERFISH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cell_id</th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>library</th>\n",
       "      <th>cell_bc</th>\n",
       "      <th>cluster_label</th>\n",
       "      <th>subclass_label</th>\n",
       "      <th>class_label</th>\n",
       "      <th>cluster_color</th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>L8TX_181211_01_G12_AAACCCAAGCTTCATG</td>\n",
       "      <td>42</td>\n",
       "      <td>L8TX_181211_01_G12</td>\n",
       "      <td>AAACCCAAGCTTCATG</td>\n",
       "      <td>L5 IT Tcap_2</td>\n",
       "      <td>L5 IT</td>\n",
       "      <td>Glutamatergic</td>\n",
       "      <td>#52CA74</td>\n",
       "      <td>17334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>L8TX_181211_01_G12_AAACCCACACCAGCCA</td>\n",
       "      <td>42</td>\n",
       "      <td>L8TX_181211_01_G12</td>\n",
       "      <td>AAACCCACACCAGCCA</td>\n",
       "      <td>L5 IT Tcap_2</td>\n",
       "      <td>L5 IT</td>\n",
       "      <td>Glutamatergic</td>\n",
       "      <td>#52CA74</td>\n",
       "      <td>17334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>L8TX_181211_01_G12_AAACGAACAACGATTC</td>\n",
       "      <td>42</td>\n",
       "      <td>L8TX_181211_01_G12</td>\n",
       "      <td>AAACGAACAACGATTC</td>\n",
       "      <td>L5 IT Tcap_2</td>\n",
       "      <td>L5 IT</td>\n",
       "      <td>Glutamatergic</td>\n",
       "      <td>#52CA74</td>\n",
       "      <td>17334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L8TX_181211_01_G12_AAACGAATCTCGTGAA</td>\n",
       "      <td>42</td>\n",
       "      <td>L8TX_181211_01_G12</td>\n",
       "      <td>AAACGAATCTCGTGAA</td>\n",
       "      <td>L5 IT Tcap_2</td>\n",
       "      <td>L5 IT</td>\n",
       "      <td>Glutamatergic</td>\n",
       "      <td>#52CA74</td>\n",
       "      <td>17334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>L8TX_181211_01_G12_AAACGCTGTAGTCACT</td>\n",
       "      <td>42</td>\n",
       "      <td>L8TX_181211_01_G12</td>\n",
       "      <td>AAACGCTGTAGTCACT</td>\n",
       "      <td>L5 IT Tcap_2</td>\n",
       "      <td>L5 IT</td>\n",
       "      <td>Glutamatergic</td>\n",
       "      <td>#52CA74</td>\n",
       "      <td>17334</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               cell_id  cluster_id             library  \\\n",
       "0  L8TX_181211_01_G12_AAACCCAAGCTTCATG          42  L8TX_181211_01_G12   \n",
       "1  L8TX_181211_01_G12_AAACCCACACCAGCCA          42  L8TX_181211_01_G12   \n",
       "2  L8TX_181211_01_G12_AAACGAACAACGATTC          42  L8TX_181211_01_G12   \n",
       "3  L8TX_181211_01_G12_AAACGAATCTCGTGAA          42  L8TX_181211_01_G12   \n",
       "4  L8TX_181211_01_G12_AAACGCTGTAGTCACT          42  L8TX_181211_01_G12   \n",
       "\n",
       "            cell_bc cluster_label subclass_label    class_label cluster_color  \\\n",
       "0  AAACCCAAGCTTCATG  L5 IT Tcap_2          L5 IT  Glutamatergic       #52CA74   \n",
       "1  AAACCCACACCAGCCA  L5 IT Tcap_2          L5 IT  Glutamatergic       #52CA74   \n",
       "2  AAACGAACAACGATTC  L5 IT Tcap_2          L5 IT  Glutamatergic       #52CA74   \n",
       "3  AAACGAATCTCGTGAA  L5 IT Tcap_2          L5 IT  Glutamatergic       #52CA74   \n",
       "4  AAACGCTGTAGTCACT  L5 IT Tcap_2          L5 IT  Glutamatergic       #52CA74   \n",
       "\n",
       "    size  \n",
       "0  17334  \n",
       "1  17334  \n",
       "2  17334  \n",
       "3  17334  \n",
       "4  17334  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rzs_meta = pd.read_csv('../../../readzs_fork/MOp_10Xv3_metadata.tsv',sep='\\t')\n",
    "rzs_meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "78"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rzs_meta['cluster_label'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['L5 IT Tcap_2', 'L5 IT Tcap_1', 'Vip Gpc3', 'Vip Htr1f_2',\n",
       "       'L6 IT Sulf1_4', 'Vip C1ql1', 'L6 CT Nxph2 Kit', 'L2/3 IT',\n",
       "       'Oligo Opalin_1', 'L6 CT Cpa6_2', 'L5 IT S100b', 'L5 IT Pld5',\n",
       "       'L6 CT Cpa6_1', 'Sst Crhr2_1', 'Lamp5 Slc35d3_1', 'Vip Cbln4_2',\n",
       "       'L6 IT Sulf1_3', 'L6 NP Trh_1', 'Vip Chat', 'L5 NP Slc17a8_2',\n",
       "       'Vip Mybpc1_1', 'Sst Hpse_2', 'L6 IT Sulf1_1', 'L5 NP Slc17a8_1',\n",
       "       'Macrophage', 'Vip Serpinf1', 'Sst Myh8_2', 'L6 NP Trh_3',\n",
       "       'Lamp5 Egln3_2_1', 'Sst Myh8_1', 'Vip Crispld2', 'Astro Aqp4',\n",
       "       'OPC Pdgfra', 'L6 IT Sulf1_2', 'L6b Rprm', 'L6 CT Nxph2 Pou3f2_2',\n",
       "       'Lamp5 Pax6', 'Vip Igfbp6', 'Lamp5 Pdlim5_2', 'VLMC Kcnj8',\n",
       "       'Lamp5 Pdlim5_1', 'L6 CT Gpr139', 'Vip Lypd1', 'Sst Calb2',\n",
       "       'Sst Pvalb Etv1', 'Endo Slc38a5_1', 'Sst C1ql3_1',\n",
       "       'L5 NP Slc17a8_3', 'Vip Col14a1', 'L5 ET_2', 'Sst Hpse_3',\n",
       "       'L6 IT Car3', 'SMC', 'Sncg_2', 'Vip Cbln4_1', 'L6b Shisa6',\n",
       "       'Vip Mybpc1_2', 'Lamp5 Lhx6', 'Sst Crhr2_2',\n",
       "       'L6 CT Nxph2 Pou3f2_1', 'L5 ET_3', 'Oligo Enpp6_2', 'Sst Hspe_1',\n",
       "       'Sst C1ql3_2', 'Vip Htr1f_1', 'L5 ET_1', 'Oligo Opalin_2',\n",
       "       'Sncg_1', 'L6b Ror1', 'Sst Th', 'L6 NP Trh_2', 'Oligo Enpp6_1',\n",
       "       'Sst Chodl', 'Pvalb Reln', 'Pvalb Gabrg1', 'L6b Kcnip1',\n",
       "       'Pvalb Calb1', 'Pvalb Vipr2'], dtype=object)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rzs_meta['cluster_label'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>slice_id</th>\n",
       "      <th>class_label</th>\n",
       "      <th>subclass</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10000143038275111136124942858811168393</td>\n",
       "      <td>mouse2_sample4</td>\n",
       "      <td>mouse2_slice31</td>\n",
       "      <td>Other</td>\n",
       "      <td>Astro</td>\n",
       "      <td>Astro_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>100001798412490480358118871918100400402</td>\n",
       "      <td>mouse2_sample5</td>\n",
       "      <td>mouse2_slice160</td>\n",
       "      <td>Other</td>\n",
       "      <td>Endo</td>\n",
       "      <td>Endo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100006878605830627922364612565348097824</td>\n",
       "      <td>mouse2_sample6</td>\n",
       "      <td>mouse2_slice109</td>\n",
       "      <td>Other</td>\n",
       "      <td>SMC</td>\n",
       "      <td>SMC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>100007228202835962319771548915451072492</td>\n",
       "      <td>mouse1_sample2</td>\n",
       "      <td>mouse1_slice71</td>\n",
       "      <td>Other</td>\n",
       "      <td>Endo</td>\n",
       "      <td>Endo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>100009332472089331948140672873134747603</td>\n",
       "      <td>mouse2_sample5</td>\n",
       "      <td>mouse2_slice219</td>\n",
       "      <td>Glutamatergic</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>L23_IT_3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Unnamed: 0       sample_id         slice_id  \\\n",
       "0   10000143038275111136124942858811168393  mouse2_sample4   mouse2_slice31   \n",
       "1  100001798412490480358118871918100400402  mouse2_sample5  mouse2_slice160   \n",
       "2  100006878605830627922364612565348097824  mouse2_sample6  mouse2_slice109   \n",
       "3  100007228202835962319771548915451072492  mouse1_sample2   mouse1_slice71   \n",
       "4  100009332472089331948140672873134747603  mouse2_sample5  mouse2_slice219   \n",
       "\n",
       "     class_label subclass     label  \n",
       "0          Other    Astro   Astro_1  \n",
       "1          Other     Endo      Endo  \n",
       "2          Other      SMC       SMC  \n",
       "3          Other     Endo      Endo  \n",
       "4  Glutamatergic  L2/3 IT  L23_IT_3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "99"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "array(['Astro_1', 'Endo', 'SMC', 'L23_IT_3', 'VLMC', 'L6_CT_5',\n",
       "       'L45_IT_2', 'L45_IT_SSp_2', 'Peri', 'L5_ET_1', 'L45_IT_1',\n",
       "       'L45_IT_5', 'L45_IT_3', 'L23_IT_4', 'L5_IT_1', 'Micro_1',\n",
       "       'L23_IT_2', 'L5_ET_5', 'Sst_3', 'L6_IT_3', 'Oligo_1', 'L5_ET_4',\n",
       "       'Sncg_1', 'L6_IT_Car3', 'Vip_7', 'PVM', 'L5_ET_3', 'L56_NP_1',\n",
       "       'Pvalb_6', 'L6_CT_3', 'L5_IT_2', 'L6_CT_1', 'L45_IT_SSp_1',\n",
       "       'Astro_2', 'Pvalb_5', 'L56_NP_2', 'Oligo_2', 'L6_IT_1', 'L6_CT_2',\n",
       "       'Pvalb_3', 'OPC', 'striatum_1', 'L5_IT_3', 'L23_IT_1', 'Oligo_3',\n",
       "       'Pvalb_4', 'Vip_1', 'L23_IT_5', 'L6_CT_6', 'Pvalb_1', 'Pvalb_7',\n",
       "       'L5_IT_4', 'L6b_2', 'L6_CT_4', 'Vip_3', 'Vip_10', 'striatum_2',\n",
       "       'Sst_2', 'L6_IT_2', 'L6b_1', 'L6_CT_8', 'Pvalb_10', 'L5_ET_2',\n",
       "       'L6_CT_7', 'Micro_2', 'L45_IT_4', 'Sst_1', 'Pvalb_2', 'Vip_6',\n",
       "       'Lamp5_1', 'L6b_3', 'ventricle_1', 'Lamp5_5', 'Sncg_2', 'Sst_6',\n",
       "       'Vip_8', 'Sst_5', 'Sst_4', 'Astro_3', 'Vip_2', 'Pvalb_8',\n",
       "       'Lamp5_9', 'Lamp5_3', 'Vip_4', 'Lamp5_8', 'Sst_7', 'Sst_8',\n",
       "       'L6_CT_9', 'Lamp5_4', 'Pvalb_9', 'Pvalb_11', 'Lamp5_6', 'Lamp5_2',\n",
       "       'Sst_Chodl', 'Vip_9', 'Lamp5_7', 'Vip_5', 'ventricle_2',\n",
       "       'Pvalb_12'], dtype=object)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mop_meta = pd.read_csv(\n",
    "    '/oak/stanford/groups/horence/rob/isoform_localizations/unprocessed_downloads/dl_scripts/revised_collection/cell_labels.csv',\n",
    ")\n",
    "display(mop_meta.head())\n",
    "display(mop_meta['label'].nunique())\n",
    "mop_meta['label'].unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Attempting to get comparable labels from Pachter's paper\n",
    "\n",
    "They did clustering etc to get their own labels to be able to compare between MERFISH and 10X and SS2\n",
    "\n",
    "I cloned their repo here: https://github.com/pachterlab/BYVSTZP_2020/tree/master/reference but I didn't find any mapping from MERFISH cell-type to 10X or SS2 cell-type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I also checked the supplemental tables, and the info is not there either\n",
    "\n",
    "Re-reading the paper and looking through the analysis jupyter notebooks in the github repo, they didn't do any analysis on the \"cluster-label\" level for the MERFISH data, everything is at the \"subclass_label\" which is what I'm currently doing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>label</th>\n",
       "      <th>subclass</th>\n",
       "      <th>sampleID</th>\n",
       "      <th>sliceID</th>\n",
       "      <th>label_v1</th>\n",
       "      <th>AP_distance (um)</th>\n",
       "      <th>pos_x (um)</th>\n",
       "      <th>pos_y (um)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00002e16-356f-419f-b96e-ec7a9327033d</td>\n",
       "      <td>Sst_1</td>\n",
       "      <td>Sst</td>\n",
       "      <td>dataset2_sample6</td>\n",
       "      <td>dataset2_slice109</td>\n",
       "      <td>I30</td>\n",
       "      <td>1180</td>\n",
       "      <td>-79.474594</td>\n",
       "      <td>2211.467867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>000068a9-c788-4aa6-becc-175159da9f44</td>\n",
       "      <td>Endo</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>dataset1_sample1</td>\n",
       "      <td>dataset1_slice50</td>\n",
       "      <td>Endo</td>\n",
       "      <td>490</td>\n",
       "      <td>3837.284502</td>\n",
       "      <td>-3847.753461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0000691f-e1ef-43a9-90c4-82f1ea0c3fde</td>\n",
       "      <td>L6_IT_1</td>\n",
       "      <td>L6_IT</td>\n",
       "      <td>dataset2_sample3</td>\n",
       "      <td>dataset2_slice139</td>\n",
       "      <td>E8</td>\n",
       "      <td>1480</td>\n",
       "      <td>3502.243851</td>\n",
       "      <td>-3511.940334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>00008097-f502-40d1-a486-c61cdd999e7d</td>\n",
       "      <td>L2/3_IT_3</td>\n",
       "      <td>L2/3_IT</td>\n",
       "      <td>dataset2_sample1</td>\n",
       "      <td>dataset2_slice61</td>\n",
       "      <td>E2</td>\n",
       "      <td>700</td>\n",
       "      <td>-3975.718063</td>\n",
       "      <td>-2609.660891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0000892d-7e0b-4ff4-b445-7ceb69ac8cfb</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>dataset1_sample5</td>\n",
       "      <td>dataset1_slice260</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>2590</td>\n",
       "      <td>-6370.114257</td>\n",
       "      <td>623.408623</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Unnamed: 0      label     subclass  \\\n",
       "0  00002e16-356f-419f-b96e-ec7a9327033d      Sst_1          Sst   \n",
       "1  000068a9-c788-4aa6-becc-175159da9f44       Endo  Endothelial   \n",
       "2  0000691f-e1ef-43a9-90c4-82f1ea0c3fde    L6_IT_1        L6_IT   \n",
       "3  00008097-f502-40d1-a486-c61cdd999e7d  L2/3_IT_3      L2/3_IT   \n",
       "4  0000892d-7e0b-4ff4-b445-7ceb69ac8cfb       VLMC         VLMC   \n",
       "\n",
       "           sampleID            sliceID label_v1  AP_distance (um)  \\\n",
       "0  dataset2_sample6  dataset2_slice109      I30              1180   \n",
       "1  dataset1_sample1   dataset1_slice50     Endo               490   \n",
       "2  dataset2_sample3  dataset2_slice139       E8              1480   \n",
       "3  dataset2_sample1   dataset2_slice61       E2               700   \n",
       "4  dataset1_sample5  dataset1_slice260     VLMC              2590   \n",
       "\n",
       "    pos_x (um)   pos_y (um)  \n",
       "0   -79.474594  2211.467867  \n",
       "1  3837.284502 -3847.753461  \n",
       "2  3502.243851 -3511.940334  \n",
       "3 -3975.718063 -2609.660891  \n",
       "4 -6370.114257   623.408623  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#MERFISH cell-labels table has a row for each cell\n",
    "#with label column (nearly directly from MERFISH metadata)\n",
    "#and label_v1 which are new labels from the Pacther analysis\n",
    "merf_cell_labels = pd.read_csv('/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/pachter_annotations/BYVSTZP_2020/reference/merfish/cell_labels.csv')\n",
    "merf_cell_labels.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "99\n",
      "93\n",
      "62\n"
     ]
    }
   ],
   "source": [
    "merf_labels = set(mop_meta['label'])\n",
    "pachter_merf_labels = set(merf_cell_labels['label'])\n",
    "\n",
    "print(len(merf_labels))\n",
    "print(len(pachter_merf_labels))\n",
    "print(len(merf_labels.intersection(pachter_merf_labels)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['I30', 'Endo', 'E8', 'E2', 'VLMC', 'I32', 'E1', 'E25', 'E9', 'E17',\n",
       "       'E7', 'I20', 'I39', 'E16', 'E22', 'Oligo', 'I29', 'E26', 'Micro',\n",
       "       'E27', 'E3', 'E11', 'E5', 'E13', 'Peri', 'Astro', 'E0', 'I6',\n",
       "       'E29', 'E4', 'E10', 'E24', 'E23', 'E18', 'OPC', 'I3', 'E36', 'E41',\n",
       "       '-1', 'E6', 'I27', 'SMC', 'E37', 'I8', 'E12', 'I26', 'I35', 'I2',\n",
       "       'other', 'E31', 'E14', 'E34', 'E35', 'I33', 'E30', 'I44', 'Macro',\n",
       "       'E39', 'I38', 'I15', 'E15', 'I31', 'I14', 'E28', 'I18', 'E33',\n",
       "       'I17', 'I1', 'I21', 'I28', 'I34', 'I16', 'I0', 'I47', 'I4', 'I12',\n",
       "       'E32', 'I13', 'I37', 'I41', 'E42', 'I49', 'I50', 'I23', 'Epen'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merf_cell_labels['label_v1'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>fovID</th>\n",
       "      <th>fov_x</th>\n",
       "      <th>fov_y</th>\n",
       "      <th>volume</th>\n",
       "      <th>center_x</th>\n",
       "      <th>center_y</th>\n",
       "      <th>slice_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>label</th>\n",
       "      <th>subclass</th>\n",
       "      <th>class_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.000010e+37</td>\n",
       "      <td>13</td>\n",
       "      <td>4539.310</td>\n",
       "      <td>2986.899</td>\n",
       "      <td>992.830060</td>\n",
       "      <td>4738.397298</td>\n",
       "      <td>3075.327399</td>\n",
       "      <td>mouse2_slice31</td>\n",
       "      <td>dataset2_sample4</td>\n",
       "      <td>Astro_1</td>\n",
       "      <td>Astrocytes</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.000020e+38</td>\n",
       "      <td>290</td>\n",
       "      <td>-4016.860</td>\n",
       "      <td>1404.780</td>\n",
       "      <td>610.091871</td>\n",
       "      <td>-3964.711404</td>\n",
       "      <td>1451.889047</td>\n",
       "      <td>mouse2_slice160</td>\n",
       "      <td>dataset2_sample5</td>\n",
       "      <td>Endo</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.000070e+38</td>\n",
       "      <td>113</td>\n",
       "      <td>750.406</td>\n",
       "      <td>1143.102</td>\n",
       "      <td>148.277718</td>\n",
       "      <td>805.843523</td>\n",
       "      <td>1215.035473</td>\n",
       "      <td>mouse2_slice109</td>\n",
       "      <td>dataset2_sample6</td>\n",
       "      <td>SMC</td>\n",
       "      <td>SMC</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.000070e+38</td>\n",
       "      <td>432</td>\n",
       "      <td>1262.680</td>\n",
       "      <td>-3752.030</td>\n",
       "      <td>249.515471</td>\n",
       "      <td>1348.686198</td>\n",
       "      <td>-3589.955255</td>\n",
       "      <td>mouse1_slice71</td>\n",
       "      <td>dataset1_sample2</td>\n",
       "      <td>Endo</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1.000090e+38</td>\n",
       "      <td>354</td>\n",
       "      <td>-3627.450</td>\n",
       "      <td>-1987.730</td>\n",
       "      <td>1779.831996</td>\n",
       "      <td>-3584.650904</td>\n",
       "      <td>-1883.105955</td>\n",
       "      <td>mouse2_slice219</td>\n",
       "      <td>dataset2_sample5</td>\n",
       "      <td>L23_IT_3</td>\n",
       "      <td>L23_IT</td>\n",
       "      <td>Glutamatergic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0  fovID     fov_x     fov_y       volume     center_x  \\\n",
       "0  1.000010e+37     13  4539.310  2986.899   992.830060  4738.397298   \n",
       "1  1.000020e+38    290 -4016.860  1404.780   610.091871 -3964.711404   \n",
       "2  1.000070e+38    113   750.406  1143.102   148.277718   805.843523   \n",
       "3  1.000070e+38    432  1262.680 -3752.030   249.515471  1348.686198   \n",
       "4  1.000090e+38    354 -3627.450 -1987.730  1779.831996 -3584.650904   \n",
       "\n",
       "      center_y         slice_id         sample_id     label     subclass  \\\n",
       "0  3075.327399   mouse2_slice31  dataset2_sample4   Astro_1   Astrocytes   \n",
       "1  1451.889047  mouse2_slice160  dataset2_sample5      Endo  Endothelial   \n",
       "2  1215.035473  mouse2_slice109  dataset2_sample6       SMC          SMC   \n",
       "3 -3589.955255   mouse1_slice71  dataset1_sample2      Endo  Endothelial   \n",
       "4 -1883.105955  mouse2_slice219  dataset2_sample5  L23_IT_3       L23_IT   \n",
       "\n",
       "     class_label  \n",
       "0          Other  \n",
       "1          Other  \n",
       "2          Other  \n",
       "3          Other  \n",
       "4  Glutamatergic  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# MERFISH cell_metadata.csv I think is straight from the Zhuang paper without changes\n",
    "merf_cell_meta = pd.read_csv('/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/pachter_annotations/BYVSTZP_2020/reference/merfish/cell_metadata.csv')\n",
    "merf_cell_meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>cluster_label</th>\n",
       "      <th>subclass_label</th>\n",
       "      <th>class_label</th>\n",
       "      <th>cluster_color</th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Lamp5 Pax6</td>\n",
       "      <td>Lamp5</td>\n",
       "      <td>GABAergic</td>\n",
       "      <td>#DDACC9</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Lamp5 Egln3_2_1</td>\n",
       "      <td>Lamp5</td>\n",
       "      <td>GABAergic</td>\n",
       "      <td>#DD8091</td>\n",
       "      <td>179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>Lamp5 Pdlim5_1</td>\n",
       "      <td>Lamp5</td>\n",
       "      <td>GABAergic</td>\n",
       "      <td>#FF829E</td>\n",
       "      <td>247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>Lamp5 Pdlim5_2</td>\n",
       "      <td>Lamp5</td>\n",
       "      <td>GABAergic</td>\n",
       "      <td>#FF7290</td>\n",
       "      <td>537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>Lamp5 Slc35d3_1</td>\n",
       "      <td>Lamp5</td>\n",
       "      <td>GABAergic</td>\n",
       "      <td>#FFA388</td>\n",
       "      <td>1275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>143</td>\n",
       "      <td>L6b Shisa6_low_1</td>\n",
       "      <td>Low Quality</td>\n",
       "      <td>Low Quality</td>\n",
       "      <td>#5A7A65</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>144</td>\n",
       "      <td>L6b Shisa6_low_2</td>\n",
       "      <td>Low Quality</td>\n",
       "      <td>Low Quality</td>\n",
       "      <td>#1F7C70</td>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>145</td>\n",
       "      <td>Oligo Opalin_neuon</td>\n",
       "      <td>doublet</td>\n",
       "      <td>Low Quality</td>\n",
       "      <td>#41775C</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>146</td>\n",
       "      <td>Endo Slc38a5_5</td>\n",
       "      <td>doublet</td>\n",
       "      <td>Low Quality</td>\n",
       "      <td>#4C6863</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>147</td>\n",
       "      <td>Endo_VLMC</td>\n",
       "      <td>doublet</td>\n",
       "      <td>Low Quality</td>\n",
       "      <td>#5A6E5E</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>147 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     cluster_id       cluster_label subclass_label  class_label cluster_color  \\\n",
       "0             1          Lamp5 Pax6          Lamp5    GABAergic       #DDACC9   \n",
       "1             2     Lamp5 Egln3_2_1          Lamp5    GABAergic       #DD8091   \n",
       "2             3      Lamp5 Pdlim5_1          Lamp5    GABAergic       #FF829E   \n",
       "3             4      Lamp5 Pdlim5_2          Lamp5    GABAergic       #FF7290   \n",
       "4             5     Lamp5 Slc35d3_1          Lamp5    GABAergic       #FFA388   \n",
       "..          ...                 ...            ...          ...           ...   \n",
       "142         143    L6b Shisa6_low_1    Low Quality  Low Quality       #5A7A65   \n",
       "143         144    L6b Shisa6_low_2    Low Quality  Low Quality       #1F7C70   \n",
       "144         145  Oligo Opalin_neuon        doublet  Low Quality       #41775C   \n",
       "145         146      Endo Slc38a5_5        doublet  Low Quality       #4C6863   \n",
       "146         147           Endo_VLMC        doublet  Low Quality       #5A6E5E   \n",
       "\n",
       "     size  \n",
       "0      81  \n",
       "1     179  \n",
       "2     247  \n",
       "3     537  \n",
       "4    1275  \n",
       "..    ...  \n",
       "142    84  \n",
       "143    95  \n",
       "144    16  \n",
       "145    36  \n",
       "146    18  \n",
       "\n",
       "[147 rows x 6 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10Xv3 cluster.annotation.csv does not have the \"label_v1\" column I'm looking for\n",
    "tenX_meta = pd.read_csv('/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/pachter_annotations/BYVSTZP_2020/reference/10xv3/cluster.annotation.csv')\n",
    "tenX_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>aggr_num</th>\n",
       "      <th>umi.counts</th>\n",
       "      <th>gene.counts</th>\n",
       "      <th>library_id</th>\n",
       "      <th>tube_barcode</th>\n",
       "      <th>Seq_batch</th>\n",
       "      <th>Region</th>\n",
       "      <th>Lib_type</th>\n",
       "      <th>Gender</th>\n",
       "      <th>...</th>\n",
       "      <th>Live_percent</th>\n",
       "      <th>Total_Cells</th>\n",
       "      <th>Live_Cells</th>\n",
       "      <th>method</th>\n",
       "      <th>exp_component_name</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>unmapped_reads</th>\n",
       "      <th>nonconf_mapped_reads</th>\n",
       "      <th>total.reads</th>\n",
       "      <th>doublet.score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAACCCAAGCTTCATG-1L8TX_181211_01_G12</td>\n",
       "      <td>1</td>\n",
       "      <td>55812</td>\n",
       "      <td>7122</td>\n",
       "      <td>L8TX_181211_01_G12</td>\n",
       "      <td>SM-HH8FP-1</td>\n",
       "      <td>RTX-841</td>\n",
       "      <td>MOp</td>\n",
       "      <td>Cells_FACS-tdT</td>\n",
       "      <td>Male</td>\n",
       "      <td>...</td>\n",
       "      <td>75%</td>\n",
       "      <td>7882</td>\n",
       "      <td>5898</td>\n",
       "      <td>10xv3</td>\n",
       "      <td>AAACCCAAGCTTCATG-1L8TX_181211_01_G12</td>\n",
       "      <td>141238</td>\n",
       "      <td>2460</td>\n",
       "      <td>11355</td>\n",
       "      <td>155053.0</td>\n",
       "      <td>0.154639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAACCCAAGGCGCTTC-1L8TX_181211_01_G12</td>\n",
       "      <td>1</td>\n",
       "      <td>1670</td>\n",
       "      <td>668</td>\n",
       "      <td>L8TX_181211_01_G12</td>\n",
       "      <td>SM-HH8FP-1</td>\n",
       "      <td>RTX-841</td>\n",
       "      <td>MOp</td>\n",
       "      <td>Cells_FACS-tdT</td>\n",
       "      <td>Male</td>\n",
       "      <td>...</td>\n",
       "      <td>75%</td>\n",
       "      <td>7882</td>\n",
       "      <td>5898</td>\n",
       "      <td>10xv3</td>\n",
       "      <td>AAACCCAAGGCGCTTC-1L8TX_181211_01_G12</td>\n",
       "      <td>4160</td>\n",
       "      <td>64</td>\n",
       "      <td>1049</td>\n",
       "      <td>5273.0</td>\n",
       "      <td>0.030928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAACCCAAGGCTTAAA-1L8TX_181211_01_G12</td>\n",
       "      <td>1</td>\n",
       "      <td>1873</td>\n",
       "      <td>406</td>\n",
       "      <td>L8TX_181211_01_G12</td>\n",
       "      <td>SM-HH8FP-1</td>\n",
       "      <td>RTX-841</td>\n",
       "      <td>MOp</td>\n",
       "      <td>Cells_FACS-tdT</td>\n",
       "      <td>Male</td>\n",
       "      <td>...</td>\n",
       "      <td>75%</td>\n",
       "      <td>7882</td>\n",
       "      <td>5898</td>\n",
       "      <td>10xv3</td>\n",
       "      <td>AAACCCAAGGCTTAAA-1L8TX_181211_01_G12</td>\n",
       "      <td>6404</td>\n",
       "      <td>47</td>\n",
       "      <td>2539</td>\n",
       "      <td>8990.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAACCCAAGTGAGGTC-1L8TX_181211_01_G12</td>\n",
       "      <td>1</td>\n",
       "      <td>36622</td>\n",
       "      <td>6125</td>\n",
       "      <td>L8TX_181211_01_G12</td>\n",
       "      <td>SM-HH8FP-1</td>\n",
       "      <td>RTX-841</td>\n",
       "      <td>MOp</td>\n",
       "      <td>Cells_FACS-tdT</td>\n",
       "      <td>Male</td>\n",
       "      <td>...</td>\n",
       "      <td>75%</td>\n",
       "      <td>7882</td>\n",
       "      <td>5898</td>\n",
       "      <td>10xv3</td>\n",
       "      <td>AAACCCAAGTGAGGTC-1L8TX_181211_01_G12</td>\n",
       "      <td>90968</td>\n",
       "      <td>1351</td>\n",
       "      <td>7997</td>\n",
       "      <td>100316.0</td>\n",
       "      <td>0.163265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAACCCACACCAGCCA-1L8TX_181211_01_G12</td>\n",
       "      <td>1</td>\n",
       "      <td>90475</td>\n",
       "      <td>8322</td>\n",
       "      <td>L8TX_181211_01_G12</td>\n",
       "      <td>SM-HH8FP-1</td>\n",
       "      <td>RTX-841</td>\n",
       "      <td>MOp</td>\n",
       "      <td>Cells_FACS-tdT</td>\n",
       "      <td>Male</td>\n",
       "      <td>...</td>\n",
       "      <td>75%</td>\n",
       "      <td>7882</td>\n",
       "      <td>5898</td>\n",
       "      <td>10xv3</td>\n",
       "      <td>AAACCCACACCAGCCA-1L8TX_181211_01_G12</td>\n",
       "      <td>240181</td>\n",
       "      <td>4107</td>\n",
       "      <td>20590</td>\n",
       "      <td>264878.0</td>\n",
       "      <td>0.164948</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Unnamed: 0  aggr_num  umi.counts  gene.counts  \\\n",
       "0  AAACCCAAGCTTCATG-1L8TX_181211_01_G12         1       55812         7122   \n",
       "1  AAACCCAAGGCGCTTC-1L8TX_181211_01_G12         1        1670          668   \n",
       "2  AAACCCAAGGCTTAAA-1L8TX_181211_01_G12         1        1873          406   \n",
       "3  AAACCCAAGTGAGGTC-1L8TX_181211_01_G12         1       36622         6125   \n",
       "4  AAACCCACACCAGCCA-1L8TX_181211_01_G12         1       90475         8322   \n",
       "\n",
       "           library_id tube_barcode Seq_batch Region        Lib_type Gender  \\\n",
       "0  L8TX_181211_01_G12   SM-HH8FP-1   RTX-841    MOp  Cells_FACS-tdT   Male   \n",
       "1  L8TX_181211_01_G12   SM-HH8FP-1   RTX-841    MOp  Cells_FACS-tdT   Male   \n",
       "2  L8TX_181211_01_G12   SM-HH8FP-1   RTX-841    MOp  Cells_FACS-tdT   Male   \n",
       "3  L8TX_181211_01_G12   SM-HH8FP-1   RTX-841    MOp  Cells_FACS-tdT   Male   \n",
       "4  L8TX_181211_01_G12   SM-HH8FP-1   RTX-841    MOp  Cells_FACS-tdT   Male   \n",
       "\n",
       "   ...  Live_percent Total_Cells Live_Cells  method  \\\n",
       "0  ...           75%        7882       5898   10xv3   \n",
       "1  ...           75%        7882       5898   10xv3   \n",
       "2  ...           75%        7882       5898   10xv3   \n",
       "3  ...           75%        7882       5898   10xv3   \n",
       "4  ...           75%        7882       5898   10xv3   \n",
       "\n",
       "                     exp_component_name mapped_reads unmapped_reads  \\\n",
       "0  AAACCCAAGCTTCATG-1L8TX_181211_01_G12       141238           2460   \n",
       "1  AAACCCAAGGCGCTTC-1L8TX_181211_01_G12         4160             64   \n",
       "2  AAACCCAAGGCTTAAA-1L8TX_181211_01_G12         6404             47   \n",
       "3  AAACCCAAGTGAGGTC-1L8TX_181211_01_G12        90968           1351   \n",
       "4  AAACCCACACCAGCCA-1L8TX_181211_01_G12       240181           4107   \n",
       "\n",
       "   nonconf_mapped_reads total.reads  doublet.score  \n",
       "0                 11355    155053.0       0.154639  \n",
       "1                  1049      5273.0       0.030928  \n",
       "2                  2539      8990.0       0.000000  \n",
       "3                  7997    100316.0       0.163265  \n",
       "4                 20590    264878.0       0.164948  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10Xv3 sample_metadata does not have the \"label_v1\" column I'm looking for either\n",
    "tenX_sample_meta = pd.read_csv('/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/pachter_annotations/BYVSTZP_2020/reference/10xv3/sample_metadata.csv')\n",
    "tenX_sample_meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'aggr_num', 'umi.counts', 'gene.counts', 'library_id',\n",
       "       'tube_barcode', 'Seq_batch', 'Region', 'Lib_type', 'Gender', 'Donor',\n",
       "       'Amp_Name', 'Amp_Date', 'Amp_PCR_cyles', 'Lib_Name', 'Lib_Date',\n",
       "       'Replicate_Lib', 'Lib_PCR_cycles', 'Lib_PassFail', 'Cell_Capture',\n",
       "       'Lib_Cells', 'Mean_Reads_perCell', 'Median_Genes_perCell',\n",
       "       'Median_UMI_perCell', 'Saturation', 'Live_percent', 'Total_Cells',\n",
       "       'Live_Cells', 'method', 'exp_component_name', 'mapped_reads',\n",
       "       'unmapped_reads', 'nonconf_mapped_reads', 'total.reads',\n",
       "       'doublet.score'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tenX_sample_meta.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>http://data.nemoarchive.org/biccn/lab/zeng/transcriptome/scell/10X/raw/MOp_v3/L8TX_181211_01_A01_S01_L003.fastq.tar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://data.nemoarchive.org/biccn/lab/zeng/tra...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>http://data.nemoarchive.org/biccn/lab/zeng/tra...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>http://data.nemoarchive.org/biccn/lab/zeng/tra...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>http://data.nemoarchive.org/biccn/lab/zeng/tra...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>http://data.nemoarchive.org/biccn/lab/zeng/tra...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  http://data.nemoarchive.org/biccn/lab/zeng/transcriptome/scell/10X/raw/MOp_v3/L8TX_181211_01_A01_S01_L003.fastq.tar\n",
       "0  http://data.nemoarchive.org/biccn/lab/zeng/tra...                                                                 \n",
       "1  http://data.nemoarchive.org/biccn/lab/zeng/tra...                                                                 \n",
       "2  http://data.nemoarchive.org/biccn/lab/zeng/tra...                                                                 \n",
       "3  http://data.nemoarchive.org/biccn/lab/zeng/tra...                                                                 \n",
       "4  http://data.nemoarchive.org/biccn/lab/zeng/tra...                                                                 "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#file links is just the url links of where to download the difference 10Xv3 fastqs\n",
    "tenX_flinks = pd.read_csv('/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/pachter_annotations/BYVSTZP_2020/reference/10xv3/10xv3_file_links.txt')\n",
    "tenX_flinks.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>x</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAACCCAAGCTTCATG-1L8TX_181211_01_G12</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAACCCAAGTGAGGTC-1L8TX_181211_01_G12</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAACCCACACCAGCCA-1L8TX_181211_01_G12</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAACCCAGTGAACGGT-1L8TX_181211_01_G12</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAACCCAGTGGCATCC-1L8TX_181211_01_G12</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Unnamed: 0   x\n",
       "0  AAACCCAAGCTTCATG-1L8TX_181211_01_G12  42\n",
       "1  AAACCCAAGTGAGGTC-1L8TX_181211_01_G12  41\n",
       "2  AAACCCACACCAGCCA-1L8TX_181211_01_G12  42\n",
       "3  AAACCCAGTGAACGGT-1L8TX_181211_01_G12  41\n",
       "4  AAACCCAGTGGCATCC-1L8TX_181211_01_G12  20"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10X membership just has a mapping from cell-id to annotation\n",
    "tenX_mem = pd.read_csv('/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/pachter_annotations/BYVSTZP_2020/reference/10xv3/cluster.membership.csv')\n",
    "tenX_mem.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Is it possible to manually align some of the cluster_labels?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Astro_1',\n",
       " 'Astro_2',\n",
       " 'Astro_3',\n",
       " 'Endo',\n",
       " 'L23_IT_1',\n",
       " 'L23_IT_2',\n",
       " 'L23_IT_3',\n",
       " 'L23_IT_4',\n",
       " 'L23_IT_5',\n",
       " 'L45_IT_1',\n",
       " 'L45_IT_2',\n",
       " 'L45_IT_3',\n",
       " 'L45_IT_4',\n",
       " 'L45_IT_5',\n",
       " 'L45_IT_SSp_1',\n",
       " 'L45_IT_SSp_2',\n",
       " 'L56_NP_1',\n",
       " 'L56_NP_2',\n",
       " 'L5_ET_1',\n",
       " 'L5_ET_2',\n",
       " 'L5_ET_3',\n",
       " 'L5_ET_4',\n",
       " 'L5_ET_5',\n",
       " 'L5_IT_1',\n",
       " 'L5_IT_2',\n",
       " 'L5_IT_3',\n",
       " 'L5_IT_4',\n",
       " 'L6_CT_1',\n",
       " 'L6_CT_2',\n",
       " 'L6_CT_3',\n",
       " 'L6_CT_4',\n",
       " 'L6_CT_5',\n",
       " 'L6_CT_6',\n",
       " 'L6_CT_7',\n",
       " 'L6_CT_8',\n",
       " 'L6_CT_9',\n",
       " 'L6_IT_1',\n",
       " 'L6_IT_2',\n",
       " 'L6_IT_3',\n",
       " 'L6_IT_Car3',\n",
       " 'L6b_1',\n",
       " 'L6b_2',\n",
       " 'L6b_3',\n",
       " 'Lamp5_1',\n",
       " 'Lamp5_2',\n",
       " 'Lamp5_3',\n",
       " 'Lamp5_4',\n",
       " 'Lamp5_5',\n",
       " 'Lamp5_6',\n",
       " 'Lamp5_7',\n",
       " 'Lamp5_8',\n",
       " 'Lamp5_9',\n",
       " 'Micro_1',\n",
       " 'Micro_2',\n",
       " 'OPC',\n",
       " 'Oligo_1',\n",
       " 'Oligo_2',\n",
       " 'Oligo_3',\n",
       " 'PVM',\n",
       " 'Peri',\n",
       " 'Pvalb_1',\n",
       " 'Pvalb_10',\n",
       " 'Pvalb_11',\n",
       " 'Pvalb_12',\n",
       " 'Pvalb_2',\n",
       " 'Pvalb_3',\n",
       " 'Pvalb_4',\n",
       " 'Pvalb_5',\n",
       " 'Pvalb_6',\n",
       " 'Pvalb_7',\n",
       " 'Pvalb_8',\n",
       " 'Pvalb_9',\n",
       " 'SMC',\n",
       " 'Sncg_1',\n",
       " 'Sncg_2',\n",
       " 'Sst_1',\n",
       " 'Sst_2',\n",
       " 'Sst_3',\n",
       " 'Sst_4',\n",
       " 'Sst_5',\n",
       " 'Sst_6',\n",
       " 'Sst_7',\n",
       " 'Sst_8',\n",
       " 'Sst_Chodl',\n",
       " 'VLMC',\n",
       " 'Vip_1',\n",
       " 'Vip_10',\n",
       " 'Vip_2',\n",
       " 'Vip_3',\n",
       " 'Vip_4',\n",
       " 'Vip_5',\n",
       " 'Vip_6',\n",
       " 'Vip_7',\n",
       " 'Vip_8',\n",
       " 'Vip_9',\n",
       " 'striatum_1',\n",
       " 'striatum_2',\n",
       " 'ventricle_1',\n",
       " 'ventricle_2']"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(mop_meta['label'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Astro Aqp4',\n",
       " 'Endo Slc38a5_1',\n",
       " 'L2/3 IT',\n",
       " 'L5 ET_1',\n",
       " 'L5 ET_2',\n",
       " 'L5 ET_3',\n",
       " 'L5 IT Pld5',\n",
       " 'L5 IT S100b',\n",
       " 'L5 IT Tcap_1',\n",
       " 'L5 IT Tcap_2',\n",
       " 'L5 NP Slc17a8_1',\n",
       " 'L5 NP Slc17a8_2',\n",
       " 'L5 NP Slc17a8_3',\n",
       " 'L6 CT Cpa6_1',\n",
       " 'L6 CT Cpa6_2',\n",
       " 'L6 CT Gpr139',\n",
       " 'L6 CT Nxph2 Kit',\n",
       " 'L6 CT Nxph2 Pou3f2_1',\n",
       " 'L6 CT Nxph2 Pou3f2_2',\n",
       " 'L6 IT Car3',\n",
       " 'L6 IT Sulf1_1',\n",
       " 'L6 IT Sulf1_2',\n",
       " 'L6 IT Sulf1_3',\n",
       " 'L6 IT Sulf1_4',\n",
       " 'L6 NP Trh_1',\n",
       " 'L6 NP Trh_2',\n",
       " 'L6 NP Trh_3',\n",
       " 'L6b Kcnip1',\n",
       " 'L6b Ror1',\n",
       " 'L6b Rprm',\n",
       " 'L6b Shisa6',\n",
       " 'Lamp5 Egln3_2_1',\n",
       " 'Lamp5 Lhx6',\n",
       " 'Lamp5 Pax6',\n",
       " 'Lamp5 Pdlim5_1',\n",
       " 'Lamp5 Pdlim5_2',\n",
       " 'Lamp5 Slc35d3_1',\n",
       " 'Macrophage',\n",
       " 'OPC Pdgfra',\n",
       " 'Oligo Enpp6_1',\n",
       " 'Oligo Enpp6_2',\n",
       " 'Oligo Opalin_1',\n",
       " 'Oligo Opalin_2',\n",
       " 'Pvalb Calb1',\n",
       " 'Pvalb Gabrg1',\n",
       " 'Pvalb Reln',\n",
       " 'Pvalb Vipr2',\n",
       " 'SMC',\n",
       " 'Sncg_1',\n",
       " 'Sncg_2',\n",
       " 'Sst C1ql3_1',\n",
       " 'Sst C1ql3_2',\n",
       " 'Sst Calb2',\n",
       " 'Sst Chodl',\n",
       " 'Sst Crhr2_1',\n",
       " 'Sst Crhr2_2',\n",
       " 'Sst Hpse_2',\n",
       " 'Sst Hpse_3',\n",
       " 'Sst Hspe_1',\n",
       " 'Sst Myh8_1',\n",
       " 'Sst Myh8_2',\n",
       " 'Sst Pvalb Etv1',\n",
       " 'Sst Th',\n",
       " 'VLMC Kcnj8',\n",
       " 'Vip C1ql1',\n",
       " 'Vip Cbln4_1',\n",
       " 'Vip Cbln4_2',\n",
       " 'Vip Chat',\n",
       " 'Vip Col14a1',\n",
       " 'Vip Crispld2',\n",
       " 'Vip Gpc3',\n",
       " 'Vip Htr1f_1',\n",
       " 'Vip Htr1f_2',\n",
       " 'Vip Igfbp6',\n",
       " 'Vip Lypd1',\n",
       " 'Vip Mybpc1_1',\n",
       " 'Vip Mybpc1_2',\n",
       " 'Vip Serpinf1']"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(rzs_meta['cluster_label'].unique())"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
