{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>sample</th>\n",
       "      <th>mouse</th>\n",
       "      <th>chr</th>\n",
       "      <th>ontology</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>/scratch/groups/horence/rob/data/MERFISH_scRNA...</td>\n",
       "      <td>L8TX_181211_01_A01</td>\n",
       "      <td>Mouse1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>L23_IT.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>/scratch/groups/horence/rob/data/MERFISH_scRNA...</td>\n",
       "      <td>L8TX_181211_01_A01</td>\n",
       "      <td>Mouse1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>L56_NP.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>/scratch/groups/horence/rob/data/MERFISH_scRNA...</td>\n",
       "      <td>L8TX_181211_01_A01</td>\n",
       "      <td>Mouse1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>L5_ET.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>/scratch/groups/horence/rob/data/MERFISH_scRNA...</td>\n",
       "      <td>L8TX_181211_01_A01</td>\n",
       "      <td>Mouse1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>L5_IT.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>/scratch/groups/horence/rob/data/MERFISH_scRNA...</td>\n",
       "      <td>L8TX_181211_01_A01</td>\n",
       "      <td>Mouse1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>L6_CT.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4835</th>\n",
       "      <td>/scratch/groups/horence/rob/data/MERFISH_scRNA...</td>\n",
       "      <td>L8TX_190430_01_G08</td>\n",
       "      <td>Mouse4</td>\n",
       "      <td>chrY</td>\n",
       "      <td>Sncg.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4836</th>\n",
       "      <td>/scratch/groups/horence/rob/data/MERFISH_scRNA...</td>\n",
       "      <td>L8TX_190430_01_G08</td>\n",
       "      <td>Mouse4</td>\n",
       "      <td>chrY</td>\n",
       "      <td>Sst.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4837</th>\n",
       "      <td>/scratch/groups/horence/rob/data/MERFISH_scRNA...</td>\n",
       "      <td>L8TX_190430_01_G08</td>\n",
       "      <td>Mouse4</td>\n",
       "      <td>chrY</td>\n",
       "      <td>VLMC.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4838</th>\n",
       "      <td>/scratch/groups/horence/rob/data/MERFISH_scRNA...</td>\n",
       "      <td>L8TX_190430_01_G08</td>\n",
       "      <td>Mouse4</td>\n",
       "      <td>chrY</td>\n",
       "      <td>Vip.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4839</th>\n",
       "      <td>/scratch/groups/horence/rob/data/MERFISH_scRNA...</td>\n",
       "      <td>L8TX_190430_01_G08</td>\n",
       "      <td>Mouse4</td>\n",
       "      <td>chrY</td>\n",
       "      <td>doublet.bam</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4840 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   path              sample  \\\n",
       "0     /scratch/groups/horence/rob/data/MERFISH_scRNA...  L8TX_181211_01_A01   \n",
       "1     /scratch/groups/horence/rob/data/MERFISH_scRNA...  L8TX_181211_01_A01   \n",
       "2     /scratch/groups/horence/rob/data/MERFISH_scRNA...  L8TX_181211_01_A01   \n",
       "3     /scratch/groups/horence/rob/data/MERFISH_scRNA...  L8TX_181211_01_A01   \n",
       "4     /scratch/groups/horence/rob/data/MERFISH_scRNA...  L8TX_181211_01_A01   \n",
       "...                                                 ...                 ...   \n",
       "4835  /scratch/groups/horence/rob/data/MERFISH_scRNA...  L8TX_190430_01_G08   \n",
       "4836  /scratch/groups/horence/rob/data/MERFISH_scRNA...  L8TX_190430_01_G08   \n",
       "4837  /scratch/groups/horence/rob/data/MERFISH_scRNA...  L8TX_190430_01_G08   \n",
       "4838  /scratch/groups/horence/rob/data/MERFISH_scRNA...  L8TX_190430_01_G08   \n",
       "4839  /scratch/groups/horence/rob/data/MERFISH_scRNA...  L8TX_190430_01_G08   \n",
       "\n",
       "       mouse    chr     ontology  \n",
       "0     Mouse1  chr10   L23_IT.bam  \n",
       "1     Mouse1  chr10   L56_NP.bam  \n",
       "2     Mouse1  chr10    L5_ET.bam  \n",
       "3     Mouse1  chr10    L5_IT.bam  \n",
       "4     Mouse1  chr10    L6_CT.bam  \n",
       "...      ...    ...          ...  \n",
       "4835  Mouse4   chrY     Sncg.bam  \n",
       "4836  Mouse4   chrY      Sst.bam  \n",
       "4837  Mouse4   chrY     VLMC.bam  \n",
       "4838  Mouse4   chrY      Vip.bam  \n",
       "4839  Mouse4   chrY  doublet.bam  \n",
       "\n",
       "[4840 rows x 5 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pysam\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import glob\n",
    "import time\n",
    "import os\n",
    "\n",
    "bam_paths = pd.read_csv(\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/split_bams.txt',\n",
    "    header=None,\n",
    "    names=['path'],\n",
    ")\n",
    "\n",
    "sample_to_mouse = {\n",
    "    'L8TX_181211_01_A01':'Mouse1',\n",
    "    'L8TX_181211_01_B01':'Mouse2',\n",
    "    'L8TX_181211_01_C01':'Mouse2',\n",
    "    'L8TX_181211_01_E01':'Mouse2',\n",
    "    'L8TX_181211_01_F01':'Mouse1',\n",
    "    'L8TX_181211_01_G12':'Mouse1',\n",
    "    'L8TX_181211_01_H12':'Mouse1',\n",
    "    'L8TX_190430_01_A08':'Mouse3',\n",
    "    'L8TX_190430_01_B08':'Mouse3',\n",
    "    'L8TX_190430_01_F08':'Mouse4',\n",
    "    'L8TX_190430_01_G08':'Mouse4',\n",
    "}\n",
    "\n",
    "bam_paths['sample'] = bam_paths['path'].map(lambda p: os.path.basename(p).split('_chr')[0])\n",
    "bam_paths['mouse'] = bam_paths['sample'].map(sample_to_mouse)\n",
    "bam_paths['chr'] = 'chr'+bam_paths['path'].map(lambda p: os.path.basename(p).split('_chr')[1].split('_')[0])\n",
    "bam_paths['ontology'] = bam_paths['path'].map(lambda p: '_'.join(os.path.basename(p).split('_chr')[1].split('_')[1:]))\n",
    "\n",
    "\n",
    "#Sam flag meanings from https://broadinstitute.github.io/picard/explain-flags.html\n",
    "bit_meanings = {\n",
    "    'read paired':                               0b000000000001,\n",
    "    'read mapped in proper pair':                0b000000000010,\n",
    "    'read unmapped':                             0b000000000100,\n",
    "    'mate unmapped':                             0b000000001000,\n",
    "    'read reverse strand':                       0b000000010000,\n",
    "    'mate reverse strand':                       0b000000100000,\n",
    "    'first in pair':                             0b000001000000,\n",
    "    'second in pair':                            0b000010000000,\n",
    "    'not primary alignment':                     0b000100000000,\n",
    "    'read fails platform/vendor quality checks': 0b001000000000,\n",
    "    'read is PCR or optical duplicate':          0b010000000000,\n",
    "    'supplementary alignment':                   0b100000000000,\n",
    "}\n",
    "\n",
    "#If any of these bits are set, reject the read\n",
    "#can test if a flag is rejectable if flag&reject_flags != 0\n",
    "reject_flags = (\n",
    "    bit_meanings['read unmapped'] |\n",
    "    bit_meanings['mate unmapped'] |\n",
    "    bit_meanings['second in pair'] |\n",
    "    bit_meanings['not primary alignment'] |\n",
    "    bit_meanings['read fails platform/vendor quality checks'] |\n",
    "    bit_meanings['supplementary alignment'] |\n",
    "    bit_meanings['read is PCR or optical duplicate']\n",
    ")\n",
    "\n",
    "\n",
    "def pass_filter(read):\n",
    "    cigar = read.cigar\n",
    "    mq = read.mapping_quality\n",
    "    read_length = read.query_length\n",
    "    rejectable_flag = read.flag & reject_flags\n",
    "    if cigar == [(0, read_length)] and mq == 255 and not rejectable_flag:\n",
    "        return True\n",
    "\n",
    "out_stem = '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_mouse_celltype/{}_{}'\n",
    "\n",
    "for (m,ont),g in bam_paths.groupby(['mouse','ontology']):\n",
    "    #create a single merged file\n",
    "    merge_file = out_stem.format(m,ont)\n",
    "    pysam.merge('-f', merge_file, *g['path'].values)\n",
    "    time.sleep(5)\n",
    "    pysam.index(merge_file)\n",
    "    \n",
    "    #filter out the bad reads just as ReadZs does (copy and pasted)\n",
    "    in_bam = pysam.AlignmentFile(merge_file)\n",
    "    filt_out = merge_file.replace('.bam','_filtered.bam')\n",
    "    out_bam = pysam.AlignmentFile(filt_out, 'wb', template=in_bam)\n",
    "    \n",
    "    #Iterate through all reads to determine if they pass filters, writing to new file\n",
    "    for read in in_bam.fetch():\n",
    "        if pass_filter(read):\n",
    "            out_bam.write(read)\n",
    "    \n",
    "    #Then finally index the filtered bam\n",
    "    in_bam.close()\n",
    "    out_bam.close()\n",
    "    pysam.index(filt_out)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "bam_paths[['mouse','ontology']].drop_duplicates().to_csv('/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/merged_by_mouse_celltype/mouse_onts.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bam_paths['mouse'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bam_paths['ontology'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5583466</td>\n",
       "      <td>5611131</td>\n",
       "      <td>Oprk1</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12687277</td>\n",
       "      <td>12866192</td>\n",
       "      <td>Sulf1</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12861549</td>\n",
       "      <td>12997650</td>\n",
       "      <td>Slco5a1</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>42690768</td>\n",
       "      <td>42708176</td>\n",
       "      <td>Pou3f3</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>56788981</td>\n",
       "      <td>56983650</td>\n",
       "      <td>Satb2</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #chr     start       end     gene strand\n",
       "0  chr1   5583466   5611131    Oprk1      +\n",
       "1  chr1  12687277  12866192    Sulf1      +\n",
       "2  chr1  12861549  12997650  Slco5a1      -\n",
       "3  chr1  42690768  42708176   Pou3f3      +\n",
       "4  chr1  56788981  56983650    Satb2      -"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(\n",
    "    '/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/plotting/buildup_plots/MERFISH_genes.bed',\n",
    "    sep = ' ',\n",
    ")\n",
    "df['start'] -= 5000\n",
    "df['end'] += 5000\n",
    "df.to_csv(\n",
    "    '/oak/stanford/groups/horence/rob/isoform_localizations/SRRS/plotting/buildup_plots/MERFISH_genes_permissive.bed',\n",
    "    index=None,\n",
    "    sep=' ',\n",
    ")\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import itertools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "bam_paths = [\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam',\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam',\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam',\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam',\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam',\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam',\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam',\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam',\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam',\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam',\n",
    "    '/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam',\n",
    "]\n",
    "chrs = df['#chr'].unique()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "220"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(list(itertools.product(bam_paths,chrs)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr1\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr2\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr3\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr4\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr5\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr6\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr7\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr8\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr9\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr10\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr11\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr12\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr13\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr14\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr15\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr16\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr17\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr18\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chr19\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_A01.bam chrX\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr1\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr2\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr3\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr4\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr5\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr6\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr7\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr8\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr9\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr10\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr11\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr12\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr13\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr14\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr15\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr16\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr17\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr18\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chr19\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_B01.bam chrX\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr1\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr2\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr3\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr4\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr5\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr6\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr7\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr8\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr9\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr10\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr11\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr12\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr13\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr14\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr15\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr16\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr17\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr18\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chr19\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_C01.bam chrX\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr1\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr2\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr3\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr4\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr5\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr6\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr7\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr8\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr9\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr10\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr11\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr12\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr13\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr14\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr15\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr16\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr17\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr18\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chr19\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_E01.bam chrX\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr1\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr2\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr3\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr4\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr5\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr6\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr7\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr8\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr9\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr10\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr11\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr12\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr13\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr14\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr15\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr16\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr17\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr18\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chr19\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_F01.bam chrX\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr1\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr2\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr3\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr4\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr5\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr6\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr7\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr8\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr9\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr10\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr11\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr12\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr13\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr14\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr15\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr16\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr17\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr18\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chr19\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_G12.bam chrX\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr1\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr2\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr3\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr4\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr5\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr6\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr7\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr8\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr9\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr10\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr11\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr12\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr13\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr14\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr15\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr16\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr17\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr18\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chr19\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_181211_01_H12.bam chrX\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr1\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr2\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr3\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr4\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr5\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr6\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr7\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr8\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr9\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr10\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr11\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr12\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr13\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr14\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr15\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr16\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr17\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr18\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chr19\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_A08.bam chrX\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr1\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr2\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr3\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr4\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr5\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr6\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr7\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr8\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr9\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr10\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr11\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr12\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr13\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr14\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr15\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr16\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr17\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr18\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chr19\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_B08.bam chrX\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr1\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr2\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr3\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr4\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr5\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr6\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr7\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr8\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr9\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr10\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr11\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr12\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr13\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr14\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr15\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr16\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr17\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr18\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chr19\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_F08.bam chrX\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr1\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr2\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr3\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr4\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr5\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr6\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr7\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr8\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr9\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr10\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr11\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr12\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr13\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr14\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr15\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr16\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr17\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr18\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chr19\n",
      "/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam chrX\n"
     ]
    }
   ],
   "source": [
    "for p,c in itertools.product(bam_paths,chrs):\n",
    "    print(p,c)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pysam.libcalignmentfile.AlignmentFile at 0x7f7769d81c18>"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bam = pysam.AlignmentFile('/scratch/groups/horence/rob/data/MERFISH_scRNAseq/10X_mapping/permissive_merfish_filt_L8TX_190430_01_G08.bam')\n",
    "bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('chr1',\n",
       " 'chr10',\n",
       " 'chr11',\n",
       " 'chr12',\n",
       " 'chr13',\n",
       " 'chr14',\n",
       " 'chr15',\n",
       " 'chr16',\n",
       " 'chr17',\n",
       " 'chr18',\n",
       " 'chr19',\n",
       " 'chr2',\n",
       " 'chr3',\n",
       " 'chr4',\n",
       " 'chr5',\n",
       " 'chr6',\n",
       " 'chr7',\n",
       " 'chr8',\n",
       " 'chr9',\n",
       " 'chrM',\n",
       " 'chrX',\n",
       " 'chrY',\n",
       " 'JH584299.1',\n",
       " 'GL456233.1',\n",
       " 'JH584301.1',\n",
       " 'GL456211.1',\n",
       " 'GL456350.1',\n",
       " 'JH584293.1',\n",
       " 'GL456221.1',\n",
       " 'JH584297.1',\n",
       " 'JH584296.1',\n",
       " 'GL456354.1',\n",
       " 'JH584294.1',\n",
       " 'JH584298.1',\n",
       " 'JH584300.1',\n",
       " 'GL456219.1',\n",
       " 'GL456210.1',\n",
       " 'JH584303.1',\n",
       " 'JH584302.1',\n",
       " 'GL456212.1',\n",
       " 'JH584304.1',\n",
       " 'GL456379.1',\n",
       " 'GL456216.1',\n",
       " 'GL456393.1',\n",
       " 'GL456366.1',\n",
       " 'GL456367.1',\n",
       " 'GL456239.1',\n",
       " 'GL456213.1',\n",
       " 'GL456383.1',\n",
       " 'GL456385.1',\n",
       " 'GL456360.1',\n",
       " 'GL456378.1',\n",
       " 'GL456389.1',\n",
       " 'GL456372.1',\n",
       " 'GL456370.1',\n",
       " 'GL456381.1',\n",
       " 'GL456387.1',\n",
       " 'GL456390.1',\n",
       " 'GL456394.1',\n",
       " 'GL456392.1',\n",
       " 'GL456382.1',\n",
       " 'GL456359.1',\n",
       " 'GL456396.1',\n",
       " 'GL456368.1',\n",
       " 'JH584292.1',\n",
       " 'JH584295.1')"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bam.references"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
